---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

```{r}
setwd("~/Documents/Bioinformatics/Lakes380/DNA/WestCoast/")
```

```{r}
library(phyloseq)
library(dplyr)
library(ggplot2)
library(ape)
library(picante)
library(reshape2)
library(ecodist)
library(tidyr)
library(circlize)
library(tibble)
library(FSA)
library(iCAMP)
library(viridis)
library(cowplot)
library(stringr)
library(ranacapa)
library(grid)
library(geosphere)
library(circlize)
library(data.table)
library(pairwiseAdonis)
library(caret)
library(ggcorrplot)
library(lares)
library(Hmisc)
```

#Load

```{r}

West.Coast.ss.bact.rare <- readRDS("WestCoast.ss16S.subtract.cleaned.merged.pruned.rare.rds")

West.Coast.ss.euks.rare <- readRDS("WestCoast.ss18S.subtract.cleaned.merged.pruned.rare.rds")

West.Coast.Phytoplankton.cleaned.rare <- readRDS("Lakes380.Phytoplankton.cleaned.ps.pruned.rare.rds")

sample_names(West.Coast.Phytoplankton.cleaned.rare) <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Code

West.Coast.zooplankton.cleaned.rare <- readRDS("Lakes380.zooplankton.cleaned.ps.pruned.rare.rds")

sample_names(West.Coast.zooplankton.cleaned.rare) <- sample_data(West.Coast.zooplankton.cleaned.rare)$Code

West.Coast.bacterioplankton.rare <- readRDS("WestCoast.16S.bacterioplankton.cleaned.ps.merged.pruned.rare.rds")
```

```{r}

tax.clean <- data.frame(tax_table(West.Coast.ss.euks.rare))

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}

tax.clean[is.na(tax.clean)] <- ""

for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    kingdom <- paste("Unclassified_Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:7] <- kingdom
  } else if (tax.clean[i,3] == ""){
    phylum <- paste("Unclassified_Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:7] <- phylum
  } else if (tax.clean[i,4] == ""){
    class <- paste("Unclassified_Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:7] <- class
  } else if (tax.clean[i,5] == ""){
    order <- paste("Unclassified_Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:7] <- order
  } else if (tax.clean[i,6] == ""){
    family <- paste("Unclassified_Family_", tax.clean[i,5], sep = "")
    tax.clean[i, 6:7] <- family
  } else if (tax.clean[i,7] == ""){
    tax.clean$Species[i] <- paste("Unclassified_Genus_",tax.clean$Genus[i], sep = "_")
  } 
}

tax_table(West.Coast.ss.euks.rare) <- as.matrix(tax.clean)
```

```{r}

tax.clean <- data.frame(tax_table(West.Coast.ss.bact.rare))

for (i in 1:6){ tax.clean[,i] <- as.character(tax.clean[,i])}

tax.clean[is.na(tax.clean)] <- ""

for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    kingdom <- paste("Unclassified_Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:6] <- kingdom
  } else if (tax.clean[i,3] == ""){
    phylum <- paste("Unclassified_Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:6] <- phylum
  } else if (tax.clean[i,4] == ""){
    class <- paste("Unclassified_Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:6] <- class
  } else if (tax.clean[i,5] == ""){
    order <- paste("Unclassified_Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:6] <- order
  } else if (tax.clean[i,6] == ""){
    tax.clean$Genus[i] <- paste("Unclassified_Family",tax.clean$Family[i], sep = "_")
  } 
}

tax_table(West.Coast.ss.bact.rare) <- as.matrix(tax.clean)
```

```{r}
ss.bact.sd <- data.frame(sample_data(West.Coast.ss.bact.rare))

ss.bact.sd$Code <- rownames(ss.bact.sd)

ss.bact.sd <- ss.bact.sd %>% select(Lake.Code, Code)

metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

ss.bact.sd <- left_join(ss.bact.sd, metadata, by = "Code")

ss.bact.sd <- left_join(ss.bact.sd, rbrdata, by = "Code")

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

ss.bact.sd.filtered <- ss.bact.sd %>%
  select(WCS.Bul.TN:NaOH.Al)

ss.bact.sdheaders <- as.data.frame(colnames(ss.bact.sd.filtered))

colnames(ss.bact.sdheaders) <- "Var"

ss.bact.sd.cleaned <- 
  ss.bact.sd %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(ss.bact.sdheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

ss.bact.sd.cleaned <- column_to_rownames(ss.bact.sd.cleaned, var="Code")

sample_data(West.Coast.ss.bact.rare) <- ss.bact.sd.cleaned
```

```{r}
ss.euks.sd <- data.frame(sample_data(West.Coast.ss.euks.rare))

ss.euks.sd$Code <- rownames(ss.euks.sd)

ss.euks.sd <- ss.euks.sd %>% select(Lake.Code, Code)

metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

ss.euks.sd <- left_join(ss.euks.sd, metadata, by = "Code")

ss.euks.sd <- left_join(ss.euks.sd, rbrdata, by = "Code")

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

ss.euks.sd.filtered <- ss.euks.sd %>%
  select(WCS.Bul.TN:NaOH.Al)

ss.euks.sdheaders <- as.data.frame(colnames(ss.euks.sd.filtered))

colnames(ss.euks.sdheaders) <- "Var"

ss.euks.sd.cleaned <- 
  ss.euks.sd %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(ss.euks.sdheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

ss.euks.sd.cleaned <- column_to_rownames(ss.euks.sd.cleaned, var="Code")

sample_data(West.Coast.ss.euks.rare) <- ss.euks.sd.cleaned
```

```{r}
phytoplankton.sd <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare))

phytoplankton.sd$Code <- as.character(phytoplankton.sd$Code)

phytoplankton.sd <- phytoplankton.sd %>% select(Code) 

metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

phytoplankton.sd <- left_join(phytoplankton.sd, metadata, by = "Code")

phytoplankton.sd <- phytoplankton.sd %>% unite("Lake.Code", c("Lake", "Code"), remove=FALSE, sep = ".")

phytoplankton.sd <- left_join(phytoplankton.sd, rbrdata, by = "Code")

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

phytoplankton.sd.filtered <- phytoplankton.sd %>%
  select(WCS.Bul.TN:NaOH.Al)

phytoplankton.sdheaders <- as.data.frame(colnames(phytoplankton.sd.filtered))

colnames(phytoplankton.sdheaders) <- "Var"

phytoplankton.sd.cleaned <- 
  phytoplankton.sd %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(phytoplankton.sdheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

phytoplankton.sd.cleaned <- column_to_rownames(phytoplankton.sd.cleaned, var="Code")

sample_data(West.Coast.Phytoplankton.cleaned.rare) <- phytoplankton.sd.cleaned
```

```{r}
zooplankton.sd <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare))

zooplankton.sd$Code <- as.character(zooplankton.sd$Code)

zooplankton.sd <- zooplankton.sd %>% select(Code) 

metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

zooplankton.sd <- left_join(zooplankton.sd, metadata, by = "Code")

zooplankton.sd <- zooplankton.sd %>% unite("Lake.Code", c("Lake", "Code"), remove=FALSE, sep = ".")

zooplankton.sd <- left_join(zooplankton.sd, rbrdata, by = "Code")

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

zooplankton.sd.filtered <- zooplankton.sd %>%
  select(WCS.Bul.TN:NaOH.Al)

zooplankton.sdheaders <- as.data.frame(colnames(zooplankton.sd.filtered))

colnames(zooplankton.sdheaders) <- "Var"

zooplankton.sd.cleaned <- 
  zooplankton.sd %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(zooplankton.sdheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

zooplankton.sd.cleaned <- column_to_rownames(zooplankton.sd.cleaned, var="Code")

sample_data(West.Coast.zooplankton.cleaned.rare) <- zooplankton.sd.cleaned
```

```{r}
new.taxonomy1 <- readRDS("WC.zooplankton.BOLD.NCBI.genus.rds")

asv_headers <- vector(dim(new.taxonomy1)[2], mode="character")

for (i in 1:dim(new.taxonomy1)[1]) {
  asv_headers[i] <- paste(">ASV", i, sep="_")
}

row.names(new.taxonomy1) <- sub(">", "", asv_headers)

new.taxonomy1 <- data.frame(new.taxonomy1)

new.taxonomy1 <- rownames_to_column(new.taxonomy1)

tax.clean <- data.frame(tax_table(West.Coast.zooplankton.cleaned.rare))

tax.clean <- rownames_to_column(tax.clean)

tax.clean  <- tax.clean %>% select(rowname)

tax.clean <- left_join(tax.clean, new.taxonomy1, by  = "rowname")

tax.clean <- column_to_rownames(tax.clean, var = "rowname")

for (i in 1:6){ tax.clean[,i] <- as.character(tax.clean[,i])}

tax.clean[is.na(tax.clean)] <- ""

for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    kingdom <- paste("Unclassified_Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:6] <- kingdom
  } else if (tax.clean[i,3] == ""){
    phylum <- paste("Unclassified_Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:6] <- phylum
  } else if (tax.clean[i,4] == ""){
    class <- paste("Unclassified_Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:6] <- class
  } else if (tax.clean[i,5] == ""){
    order <- paste("Unclassified_Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:6] <- order
  } else if (tax.clean[i,6] == ""){
    tax.clean$Genus[i] <- paste("Unclassified_Family",tax.clean$Family[i], sep = "_")
  } 
}

tax.clean <- rownames_to_column(tax.clean)

species.list <- read.csv("zooplankton_blast.tax.csv", h=T)

species.list <- species.list %>% filter(PID >= 95)

tax.clean <- left_join(tax.clean, species.list,  by = c("rowname" = "ASV"))

tax.clean <- tax.clean %>% select(-Accession, -PID)

tax.clean <- column_to_rownames(tax.clean, var = "rowname")
tax_table(West.Coast.zooplankton.cleaned.rare) <- as.matrix(tax.clean)

```

```{r}
bacterioplankton.sd <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

bacterioplankton.sd$Code <- rownames(bacterioplankton.sd)

bacterioplankton.sd <- bacterioplankton.sd %>% select(Code)

metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

bacterioplankton.sd <- left_join(bacterioplankton.sd, metadata, by = "Code")

bacterioplankton.sd <- bacterioplankton.sd %>% unite("Lake.Code", c("Lake", "Code"), remove=FALSE, sep = ".")

bacterioplankton.sd <- left_join(bacterioplankton.sd, rbrdata, by = "Code")

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

bacterioplankton.sd.filtered <- bacterioplankton.sd %>%
  select(WCS.Bul.TN:NaOH.Al)

bacterioplankton.sdheaders <- as.data.frame(colnames(bacterioplankton.sd.filtered))

colnames(bacterioplankton.sdheaders) <- "Var"

bacterioplankton.sd.cleaned <- 
  bacterioplankton.sd %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(bacterioplankton.sdheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

bacterioplankton.sd.cleaned <- column_to_rownames(bacterioplankton.sd.cleaned, var="Code")

sample_data(West.Coast.bacterioplankton.rare) <- bacterioplankton.sd.cleaned
```

#Map

```{r}
access <- read.csv("WestCoastAccess.csv", h=T)

access <- access %>% select(Lake, Code, No.access, Access.Pressure, Access.Pressure2, AccessPressure3)

access$No.access[access$No.access == 1] <- "No"
access$No.access[access$No.access == 0] <- "Yes"
access$Code <- as.character(access$Code)
```

```{r}
sd.bact.mv <- data.frame(sample_data(West.Coast.bacterioplankton.rare))
sd.bact.mv$Code <- rownames(sd.bact.mv)

sd.bact.mv$Code <- as.character(sd.bact.mv$Code)

sd.bact.mv <- left_join(sd.bact.mv, access, by="Code")

rownames(sd.bact.mv) <- sd.bact.mv$Code
sample_data(West.Coast.bacterioplankton.rare) <- sd.bact.mv
```

```{r}
sd.Bacterioplankton <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

library(marmap)
library(ggsn)
library(ggrepel)

getNOAA.bathy(lon1=165.5,lon2=179,lat1=-48,lat2=-34, resolution=1) -> NZ

getNOAA.bathy(lon1=168,lon2=172.2,lat1=-45,lat2=-42.3, resolution=1) -> West.Coast

trace(autoplot.bathy, edit=TRUE) ### Change coastline parameters colour = "gray20", linetype = "solid", size = 0.2, breaks = 0, alpha = 1)

NZ.p <- autoplot(NZ, geom=c("contour", "raster"), colour="white", size=0.1, show.legend = FALSE) +
  scale_fill_gradient2(low="#FFFFFF", mid="#FFFFFF", high="#a19f9c") +
  theme(axis.title=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank()) +
  annotate("rect", xmin = 168, xmax = 172.2, ymin = -45, ymax = -42.3, alpha = 0, color="black")

West.Coast.p <- autoplot(West.Coast, geom=c("contour", "raster"), colour="white", size=0.1, show.legend = FALSE) +
  scale_fill_gradient2(low="#FFFFFF", mid="#FFFFFF", high="#a19f9c") +
  theme(axis.title=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank())

sd.Bacterioplankton$Latitude <- as.numeric(sd.Bacterioplankton$Latitude)

sd.Bacterioplankton$Longitude <- as.numeric(sd.Bacterioplankton$Longitude)

sd.Bacterioplankton$lat <- sd.Bacterioplankton$Latitude
sd.Bacterioplankton$long<- sd.Bacterioplankton$Longitude

sd.Bacterioplankton$lat <- as.numeric(sd.Bacterioplankton$lat)

sd.Bacterioplankton$long <- as.numeric(sd.Bacterioplankton$long)

sd.Bacterioplankton$AccessPressure3<- factor(sd.Bacterioplankton$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.p1 <- West.Coast.p + geom_point(data=sd.Bacterioplankton, aes(y=Latitude, x=Longitude, colour=AccessPressure3), size=2) + 
  theme_bw() +
  xlab("Longitude") + 
  ylab("Latitude") +
  theme(axis.text=element_text(size = 12, color="black")) + 
  theme(axis.title=element_text(size = 12, color="black")) +
  theme(legend.background = element_rect(fill="transparent", colour=NA),
        legend.key = element_rect(colour = NA, fill=NA)) +
  theme(legend.text=element_text(size = 10, color="black")) + 
  theme(legend.title=element_text(size = 10, color="black")) + 
  scale_colour_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  guides(colour = guide_legend(override.aes = list(size=3), title="Ease of \nhuman access")) + 
  theme(legend.position = c(0.9,0.14)) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  geom_text_repel(data=sd.Bacterioplankton, aes(Longitude, Latitude, label=as.character(Lake.y)), size=2.5, segment.size = 0.1) +
  north(x.min = 168, x.max = 172.2, y.min = -42.3, y.max = -45, scale= 0.1, symbol=3, anchor =c(x= 170.0, y=-42.35)) +
  ggsn::scalebar(sd.Bacterioplankton, dist = 25, dist_unit = "km",
             transform = TRUE, model = "WGS84", height = 0.01, st.size = 3, border.size = 0.5, anchor =c(x= 170.8, y=-42.5)) 

West.Coast.p2 <- West.Coast.p1 + annotation_custom(grob=ggplotGrob(NZ.p), xmin = 168.2, xmax = 169.4, ymin = -42.3, ymax = -43.5) +
  annotate("rect", xmin = 168.2, xmax = 169.4, ymin = -42.37, ymax = -43.4, alpha = 0, color="black")

cairo_ps(file="West.Coast.map.eps", width=6, height=6, bg="transparent")
West.Coast.p2
dev.off()
```

#Rename_sequences

```{r}
asv_headers1 <- vector(dim(data.frame(taxa_names(West.Coast.ss.bact.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.ss.bact.rare)))[1]) {
  asv_headers1[i] <- paste("Surf_bact", i, sep="_")
}

taxa_names(West.Coast.ss.bact.rare) <-asv_headers1

asv_headers1 <- vector(dim(data.frame(taxa_names(West.Coast.ss.euks.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.ss.euks.rare)))[1]) {
  asv_headers1[i] <- paste("Surf_euks", i, sep="_")
}

taxa_names(West.Coast.ss.euks.rare) <-asv_headers1

asv_headers1 <- vector(dim(data.frame(taxa_names(West.Coast.Phytoplankton.cleaned.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.Phytoplankton.cleaned.rare)))[1]) {
  asv_headers1[i] <- paste("Phyto", i, sep="_")
}

taxa_names(West.Coast.Phytoplankton.cleaned.rare) <-asv_headers1


asv_headers1 <- vector(dim(data.frame(taxa_names(West.Coast.bacterioplankton.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.bacterioplankton.rare)))[1]) {
  asv_headers1[i] <- paste("Bacterioplankton", i, sep="_")
}

taxa_names(West.Coast.bacterioplankton.rare) <-asv_headers1
```

#Align

```{r}

asv_seqs <- taxa_names(West.Coast.ss.bact.rare)

asv_headers <- vector(dim(data.frame(taxa_names(West.Coast.ss.bact.rare)))[1], mode="character")


for (i in 1:dim(data.frame(taxa_names(West.Coast.ss.bact.rare)))[1]) {
  asv_headers[i] <- paste(">Surf_bact", i, sep="_")
}

  # making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "West.Coast.Surf_bact.ASVs.fa")

asv_seqs <- taxa_names(West.Coast.ss.euks.rare)

asv_headers <- vector(dim(data.frame(taxa_names(West.Coast.ss.euks.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.ss.euks.rare)))[1]) {
  asv_headers[i] <- paste(">Surf_euks", i, sep="_")
}

  # making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "West.Coast.Surf_euks.ASVs.fa")

asv_seqs <- taxa_names(West.Coast.Phytoplankton.cleaned.rare)

asv_headers <- vector(dim(data.frame(taxa_names(West.Coast.Phytoplankton.cleaned.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.Phytoplankton.cleaned.rare)))[1]) {
  asv_headers[i] <- paste(">Phyto", i, sep="_")
}

  # making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "West.Coast.phytoplankton.ASVs.fa")

write.csv(data.frame(taxa_names(West.Coast.zooplankton.cleaned.rare)), "Zooplankton_names.csv")


asv_seqs <- taxa_names(West.Coast.bacterioplankton.rare)

asv_headers <- vector(dim(data.frame(taxa_names(West.Coast.bacterioplankton.rare)))[1], mode="character")

for (i in 1:dim(data.frame(taxa_names(West.Coast.bacterioplankton.rare)))[1]) {
  asv_headers[i] <- paste(">Bacterioplankton", i, sep="_")
}

  # making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "West.Coast.bacterioplankton.ASVs.fa")

file.zoo <- readDNAStringSet("Zooplankton_filtered_ASVs.fasta")

align.zoo<- msa(file.zoo, method="ClustalW", verbose=T)

cv.align.zoo <- msaConvert(align.zoo, type=c("bios2mds::align"))

export.fasta(cv.align.zoo, outfile = "zooplankton_aligned.fas", ncol(align.zoo), open = "w")

file.ssbact <- readDNAStringSet("West.Coast.Surf_bact.ASVs.fa")

align.ssbact <- msa(file.ssbact, method="ClustalW", verbose=T)

cv.align.ssbact <- msaConvert(align.ssbact, type=c("bios2mds::align"))

export.fasta(cv.align.ssbact, outfile = "surface_sediment_bact_aligned.fas", ncol(align.ssbact), open = "w")

file.sseuks <- readDNAStringSet("West.Coast.Surf_euks.ASVs.fa")

align.sseuks <- msa(file.sseuks, method="ClustalW", verbose=T)

cv.align.seuks <- msaConvert(align.sseuks, type=c("bios2mds::align"))

export.fasta(cv.align.sseuks, outfile = "surface_sediment_euks_aligned.fas", ncol(align.sseuks), open = "w")

file.phyto <- readDNAStringSet("West.Coast.phytoplankton.ASVs.fa")

align.phyto<- msa(file.phyto, method="ClustalW", verbose=T)

cv.align.phyto <- msaConvert(align.phyto, type=c("bios2mds::align"))

export.fasta(cv.align.phyto, outfile = "phytoplankton_aligned.fas", ncol(align.phyto), open = "w")

file.bacterioplankton <- readDNAStringSet("West.Coast.bacterioplankton.ASVs.fa")

align.bacterioplankton <- msa(file.bacterioplankton, method="ClustalW", verbose=T)

cv.align.bacterioplankton <- msaConvert(align.bacterioplankton, type=c("bios2mds::align"))

export.fasta(cv.align.bacterioplankton, outfile = "bacterioplankton_aligned.fas", ncol(align.bacterioplankton), open = "w")
```

#Rarefaction investigation

```{r warning=FALSE eval=FALSE}
library(ranacapa)

ss.bact.rarefaction <- ggrare(West.Coast.ss.bact.rare, step=500) + 
  ggplot2::theme_bw() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="bottom") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  guides(color=guide_legend(override.aes=list(fill=NA, size = 5))) 

ss.euks.rarefaction <- ggrare(West.Coast.ss.euks.rare, step=500) + 
  ggplot2::theme_bw() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="bottom") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  guides(color=guide_legend(override.aes=list(fill=NA, size = 5))) 

phyto.rarefaction <- ggrare(West.Coast.Phytoplankton.cleaned.rare, step=1000) + 
  ggplot2::theme_bw() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="bottom") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  guides(color=guide_legend(override.aes=list(fill=NA, size = 5))) 

zoo.rarefaction <- ggrare(West.Coast.zooplankton.cleaned.rare, step=500) + 
  ggplot2::theme_bw() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="bottom") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  guides(color=guide_legend(override.aes=list(fill=NA, size = 5))) 

bacterioplankton.rarefaction <- ggrare(West.Coast.bacterioplankton.rare, step=500) + 
  ggplot2::theme_bw() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="bottom") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  guides(color=guide_legend(override.aes=list(fill=NA, size = 5))) 

```

```{r}
metadata <- read.csv("~/Documents/Bioinformatics/Lakes380/metadata_2021.csv", h=T)

rbrdata <- read.csv("~/Documents/Bioinformatics/Lakes380/rbrfiles/rbrfiles.exports/Lake.summary.data.rbr.csv", h=T)

map <- left_join(metadata, rbrdata, by = "Code")
```

```{r}

h_detect_lim <- function(var) {
  var <- as.character(var)
  isLimit <- grepl("<", var)
  var[isLimit] <- as.numeric(gsub("<(.*)", "\\1", var[isLimit])) / 2
  as.numeric(var)
}

map.filtered <- map %>%
  select(WCS.Bul.TN:NaOH.Al)

mapheaders <- as.data.frame(colnames(map.filtered))

colnames(mapheaders) <- "Var"

map.cleaned <- 
  map %>%
  mutate(across(Max.Depth:Phytoplankton.filtered, ~ as.numeric(str_replace_all(., "[^0-9.-]", "")))) %>% 
  mutate(across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "No results", "")), # replace "No results" 
         across(WCS.Bul.TN:last_col(), ~ str_replace_all(., "Not tested", ""))) %>% # replace "Not tested" 
  mutate_at(vars(mapheaders$Var, -Sys.for.Densities, -Shallow.Deep), h_detect_lim) %>% #TN.Median -  check saved as number (, in number) and RSE.NaOH.rP.BD.P - No DIV.
  mutate(across(TLI3:Bloom , ~as.numeric(.)))

```

#Alpha diveristy

```{r}
sample_data(West.Coast.ss.bact.rare)$Type <- "Sediment Bacteria"

West.Coast.ss.bact.rare.merged <- merge_samples(West.Coast.ss.bact.rare, "Type")

sample_data(West.Coast.ss.bact.rare.merged)$Type <- "Surface sediment Bacteria"

ss.bact.rarefaction <- ggrare(West.Coast.ss.bact.rare.merged, step=25000, parallel=TRUE, color = "Type") + 
  ggplot2::theme_classic() +
  ggplot2::xlab("# Reads") +
  ggplot2::ylab("# ASVs") +
  scale_color_manual(values= "#2e382e") +
  scale_fill_manual(values= "#2e382e") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Surface sediment bacteria") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))

```

```{r}
sample_data(West.Coast.ss.euks.rare)$Type <- "Sediment Eukaryotes"

West.Coast.ss.euks.rare.merged <- merge_samples(West.Coast.ss.euks.rare, "Type")

sample_data(West.Coast.ss.euks.rare.merged)$Type <- "Surface sediment eukaryotes"

ss.euks.rarefaction <- ggrare(West.Coast.ss.euks.rare.merged, step=25000, parallel=TRUE, color = "Type") + 
  ggplot2::theme_classic() +
  ggplot2::xlab("") +
  ggplot2::ylab("# ASVs") +
  scale_color_manual(values= "#50c9ce") +
  scale_fill_manual(values= "#50c9ce") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Surface sediment eukaryotes") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))

```

```{r}
sample_data(West.Coast.Phytoplankton.cleaned.rare)$Type <- "Small Planktonic Eukaryotes"

West.Coast.Phytoplankton.cleaned.rare.merged <- merge_samples(West.Coast.Phytoplankton.cleaned.rare, "Type")

sample_data(West.Coast.Phytoplankton.cleaned.rare.merged)$Type <- "Small planktonic eukaryotes"

Phytoplankton.rarefaction <- ggrare(West.Coast.Phytoplankton.cleaned.rare.merged, step=25000, parallel=TRUE, color = "Type") + 
  ggplot2::theme_classic() +
  ggplot2::xlab("") +
  ggplot2::ylab("# ASVs") +
  scale_color_manual(values= "#72a1e5") +
  scale_fill_manual(values= "#72a1e5") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Small planktonic eukaryotes") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))
```

```{r}
sample_data(West.Coast.zooplankton.cleaned.rare)$Type <- "Large Planktonic Eukaryotes"

West.Coast.zooplankton.cleaned.rare.merged <- merge_samples(West.Coast.zooplankton.cleaned.rare, "Type")

sample_data(West.Coast.zooplankton.cleaned.rare.merged)$Type <- "Large planktonic eukaryotes"

zooplankton.rarefaction <- ggrare(West.Coast.zooplankton.cleaned.rare.merged, step=25000, parallel=TRUE, color = "Type") + 
  ggplot2::theme_classic() +
  ggplot2::xlab("") +
  ggplot2::ylab("# ASVs") +
  scale_color_manual(values= "#9883e5") +
  scale_fill_manual(values= "#9883e5") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Large planktonic eukaryotes") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))
```

```{r}
sample_data(West.Coast.bacterioplankton.rare)$Type <- "Bacterioplankton"

West.Coast.bacterioplankton.rare.merged <- merge_samples(West.Coast.bacterioplankton.rare, "Type")

sample_data(West.Coast.bacterioplankton.rare.merged)$Type <- "Bacterioplankton"

bacterioplankton.rarefaction <- ggrare(West.Coast.bacterioplankton.rare.merged, step=25000, parallel=TRUE, color = "Type") + 
  ggplot2::theme_classic() +
  ggplot2::xlab("") +
  ggplot2::ylab("# ASVs") +
  scale_color_manual(values= "#fcd3de") +
  scale_fill_manual(values= "#fcd3de") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) + 
  theme(plot.title=element_text(size=14, color="black",  family="sans")) + 
  theme(legend.position="") +
  theme(legend.title = element_blank()) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  ggtitle("Bacterioplankton") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))
```

```{r}
cairo_ps(file="West.Coast.rarefaction.eps", width=4, height=10, bg="transparent")
gl<-grid.layout(nrow=5, ncol=1, heights=unit(c(20,20,20,20,20), "null"), widths=unit(c(20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.3<-viewport(layout.pos.col=1, layout.pos.row=3)
vp.4<-viewport(layout.pos.col=1, layout.pos.row=4)
vp.5<-viewport(layout.pos.col=1, layout.pos.row=5)
pushViewport(viewport(layout=gl))
print(bacterioplankton.rarefaction, vp=vp.1)
print(zooplankton.rarefaction, vp=vp.2)
print(Phytoplankton.rarefaction, vp=vp.3)
print(ss.euks.rarefaction, vp=vp.4)
print(ss.bact.rarefaction, vp=vp.5)
dev.off()
```

```{r}

ss.bact.richness <- estimate_richness(West.Coast.ss.bact.rare, measures=c("Observed", "Shannon"))

mean(ss.bact.richness$Observed)

max(ss.bact.richness$Observed)
min(ss.bact.richness$Observed)

ss.bact.richness$Type <- "Sediment Bacteria"

ss.bact.richness.plot <- ggplot(ss.bact.richness, aes(x = Type, y=Observed, fill=Type)) + 
  geom_violin() +
  theme_classic() +
  scale_fill_manual(values= "#2e382e") +
  theme(axis.title.x = element_blank(),
        legend.position = "",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylab("") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) 
  
ss.bact.richness.plot

ss.bact.richness <- rownames_to_column(ss.bact.richness)

ss.bact.richness$rowname <- gsub("X", "", ss.bact.richness$rowname)

sd.bact.richness <- data.frame(sample_data(West.Coast.ss.bact.rare))

sd.bact.richness  <- rownames_to_column(sd.bact.richness)

sd.bact.richness <- sd.bact.richness %>% select(rowname, Altitude)

ss.bact.richness.alt <- left_join(ss.bact.richness, sd.bact.richness, by  = "rowname")

bact.alt <- ggplot(ss.bact.richness.alt, aes(y=Observed, x = Altitude)) + geom_point() + theme_bw() + geom_smooth(method = "lm")

```

```{r}

ss.euks.richness <- estimate_richness(West.Coast.ss.euks.rare, measures=c("Observed", "Shannon"))

mean(ss.euks.richness$Observed)

max(ss.euks.richness$Observed)
min(ss.euks.richness$Observed)

ss.euks.richness$Type <- "Sediment Eukaryotes"

ss.euks.richness.plot <- ggplot(ss.euks.richness, aes(x = Type, y=Observed, fill=Type)) + 
  geom_violin() +
  theme_classic() +
  scale_fill_manual(values= "#50c9ce") +
  theme(axis.title.x = element_blank(),
        legend.position = "",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylab("") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) 
  
ss.euks.richness.plot

ss.euks.richness <- rownames_to_column(ss.euks.richness)

ss.euks.richness$rowname <- gsub("X", "", ss.euks.richness$rowname)

sd.euks.richness <- data.frame(sample_data(West.Coast.ss.euks.rare))

sd.euks.richness  <- rownames_to_column(sd.euks.richness)

sd.euks.richness <- sd.euks.richness %>% select(rowname, Altitude)

ss.euks.richness.alt <- left_join(ss.euks.richness, sd.euks.richness, by  = "rowname")

euks.alt <- ggplot(ss.euks.richness.alt, aes(y=Observed, x = Altitude)) + geom_point() + theme_bw() + geom_smooth(method = "lm")
```

```{r}

Phytoplankton.richness <- estimate_richness(West.Coast.Phytoplankton.cleaned.rare, measures=c("Observed", "Shannon"))

mean(Phytoplankton.richness$Observed)

max(Phytoplankton.richness$Observed)
min(Phytoplankton.richness$Observed)

Phytoplankton.richness$Type <- "Phytoplankton"

Phytoplankton.richness.plot <- ggplot(Phytoplankton.richness, aes(x = Type, y=Observed, fill=Type)) + 
  geom_violin() +
  theme_classic() +
  scale_fill_manual(values= "#72a1e5") +
  theme(axis.title.x = element_blank(),
        legend.position = "",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylab("") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) 
  
Phytoplankton.richness <- rownames_to_column(Phytoplankton.richness)

Phytoplankton.richness$rowname <- gsub("X", "", Phytoplankton.richness$rowname)

sd.Phytoplankton.richness <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare))

sd.Phytoplankton.richness  <- rownames_to_column(sd.Phytoplankton.richness)

sd.Phytoplankton.richness <- sd.Phytoplankton.richness %>% select(rowname, Altitude)

sd.Phytoplankton.richness$rowname <- gsub(" ", ".", sd.Phytoplankton.richness$rowname)
sd.Phytoplankton.richness$rowname <- gsub("-", ".", sd.Phytoplankton.richness$rowname)

Phytoplankton.richness.alt <- left_join(Phytoplankton.richness, sd.Phytoplankton.richness, by  = "rowname")

Phytoplankton.alt <- ggplot(Phytoplankton.richness.alt, aes(y=Observed, x = Altitude)) + geom_point() + theme_bw() + geom_smooth(method = "lm")
```

```{r}
zooplankton.richness <- estimate_richness(West.Coast.zooplankton.cleaned.rare, measures=c("Observed", "Shannon"))

mean(zooplankton.richness$Observed)

max(zooplankton.richness$Observed)
min(zooplankton.richness$Observed)

zooplankton.richness$Type <- "Zooplankton"

zooplankton.richness.plot <- ggplot(zooplankton.richness, aes(x = Type, y=Observed, fill=Type)) + 
  geom_violin() +
  theme_classic() +
  scale_fill_manual(values= "#9883e5") +
  theme(axis.title.x = element_blank(),
        legend.position = "",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylab("") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) 
  
zooplankton.richness <- rownames_to_column(zooplankton.richness)

zooplankton.richness$rowname <- gsub("X", "", zooplankton.richness$rowname)

sd.zooplankton.richness <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare))

sd.zooplankton.richness  <- rownames_to_column(sd.zooplankton.richness)

sd.zooplankton.richness <- sd.zooplankton.richness %>% select(rowname, Altitude)

sd.zooplankton.richness$rowname <- gsub(" ", ".", sd.zooplankton.richness$rowname)
sd.zooplankton.richness$rowname <- gsub("-", ".", sd.zooplankton.richness$rowname)

zooplankton.richness.alt <- left_join(zooplankton.richness, sd.zooplankton.richness, by  = "rowname")

zooplankton.alt <- ggplot(zooplankton.richness.alt, aes(y=Observed, x = Altitude)) + geom_point() + theme_bw() + geom_smooth(method = "lm")
```

```{r}
Bacterioplankton.richness <- estimate_richness(West.Coast.bacterioplankton.rare, measures=c("Observed", "Shannon"))

mean(Bacterioplankton.richness$Observed)

max(Bacterioplankton.richness$Observed)
min(Bacterioplankton.richness$Observed)

Bacterioplankton.richness$Type <- "Bacterioplankton"

Bacterioplankton.richness.plot <- ggplot(Bacterioplankton.richness, aes(x = Type, y=Observed, fill=Type)) + 
  geom_violin() +
  theme_classic() +
  scale_fill_manual(values= "#fcd3de") +
  theme(axis.title.x = element_blank(),
        legend.position = "",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylab("") +
  theme(axis.title=element_text(size=12, color="black",  family="sans")) + 
 theme(plot.title=element_text(size=16, color="black",  family="sans")) + 
  theme(axis.text=element_text(size=11, color="black",  family="sans")) +
  ggtitle("Violin Plots")

Bacterioplankton.richness <- rownames_to_column(Bacterioplankton.richness)

Bacterioplankton.richness$rowname <- gsub("X", "", Bacterioplankton.richness$rowname)

sd.Bacterioplankton.richness <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

sd.Bacterioplankton.richness  <- rownames_to_column(sd.Bacterioplankton.richness)

sd.Bacterioplankton.richness <- sd.Bacterioplankton.richness %>% select(rowname, Altitude)

sd.Bacterioplankton.richness$rowname <- gsub(" ", ".", sd.Bacterioplankton.richness$rowname)
sd.Bacterioplankton.richness$rowname <- gsub("-", ".", sd.Bacterioplankton.richness$rowname)

Bacterioplankton.richness.alt <- left_join(Bacterioplankton.richness, sd.Bacterioplankton.richness, by  = "rowname")

Bacterioplankton.alt <- ggplot(Bacterioplankton.richness.alt, aes(y=Observed, x = Altitude)) + geom_point() + theme_bw() + geom_smooth(method = "lm")
```

```{r}
library(cowplot)

bacterioplankton.merged <- plot_grid(bacterioplankton.rarefaction, Bacterioplankton.richness.plot, align = "h", rel_widths =c(2,1))
zooplankton.merged <- plot_grid(zooplankton.rarefaction, zooplankton.richness.plot, align = "h", rel_widths =c(2,1))
Phytoplankton.merged <- plot_grid(Phytoplankton.rarefaction, Phytoplankton.richness.plot, align = "h", rel_widths =c(2,1))
ss.euks.merged <- plot_grid(ss.euks.rarefaction, ss.euks.richness.plot, align = "h", rel_widths =c(2,1))
ss.bact.merged <- plot_grid(ss.bact.rarefaction, ss.bact.richness.plot, align = "h", rel_widths =c(2,1))
```

```{r}
cairo_ps(file="West.Coast.richness.eps", width=8, height=10, bg="transparent")
gl<-grid.layout(nrow=5, ncol=2, heights=unit(c(22,20,20,20,20), "null"), widths=unit(c(3,20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=2, layout.pos.row=1)
vp.3<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.4<-viewport(layout.pos.col=2, layout.pos.row=2)
vp.5<-viewport(layout.pos.col=1, layout.pos.row=3)
vp.6<-viewport(layout.pos.col=2, layout.pos.row=3)
vp.7<-viewport(layout.pos.col=1, layout.pos.row=4)
vp.8<-viewport(layout.pos.col=2, layout.pos.row=4)
vp.9<-viewport(layout.pos.col=1, layout.pos.row=5)
vp.10<-viewport(layout.pos.col=2, layout.pos.row=5)
pushViewport(viewport(layout=gl))
pushViewport(vp.1)
grid.text("Bacterioplankton", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.3)
grid.text("Large Planktonic \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.5)
grid.text("Small Planktonic \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.7)
grid.text("Sediment \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.9)
grid.text("Sediment \nBacteria", gp = gpar(fontsize = 16), rot = 90)
popViewport()
print(bacterioplankton.merged, vp=vp.2)
print(zooplankton.merged, vp=vp.4)
print(Phytoplankton.merged, vp=vp.6)
print(ss.euks.merged, vp=vp.8)
print(ss.bact.merged, vp=vp.10)
dev.off()
```

```{r}
cairo_ps(file="West.Coast.richness.alt.eps", width=4, height=10, bg="transparent")
gl<-grid.layout(nrow=5, ncol=2, heights=unit(c(22,20,20,20,20), "null"), widths=unit(c(3,20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=2, layout.pos.row=1)
vp.3<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.4<-viewport(layout.pos.col=2, layout.pos.row=2)
vp.5<-viewport(layout.pos.col=1, layout.pos.row=3)
vp.6<-viewport(layout.pos.col=2, layout.pos.row=3)
vp.7<-viewport(layout.pos.col=1, layout.pos.row=4)
vp.8<-viewport(layout.pos.col=2, layout.pos.row=4)
vp.9<-viewport(layout.pos.col=1, layout.pos.row=5)
vp.10<-viewport(layout.pos.col=2, layout.pos.row=5)
pushViewport(viewport(layout=gl))
pushViewport(vp.1)
grid.text("Bacterioplankton", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.3)
grid.text("Large Planktonic \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.5)
grid.text("Small Planktonic \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.7)
grid.text("Sediment \nEukaryotes", gp = gpar(fontsize = 16), rot = 90)
popViewport()
pushViewport(vp.9)
grid.text("Sediment \nBacteria", gp = gpar(fontsize = 16), rot = 90)
popViewport()
print(Bacterioplankton.alt, vp=vp.2)
print(zooplankton.alt, vp=vp.4)
print(Phytoplankton.alt, vp=vp.6)
print(euks.alt, vp=vp.8)
print(bact.alt, vp=vp.10)
dev.off()
```

#Shared

```{r}
sample_data(West.Coast.ss.bact.rare)$Type <- "Sediment Bacteria"

West.Coast.ss.bact.rare.merged <- merge_samples(West.Coast.ss.bact.rare, "Type")

West.Coast.ss.bact.rare.merged.df <- data.frame(t(otu_table(West.Coast.ss.bact.rare.merged)))
West.Coast.ss.bact.rare.merged.df <- rownames_to_column(West.Coast.ss.bact.rare.merged.df)

ss.bact.input.fa <- read.fasta("West.Coast.Surf_bact.ASVs.fa", as.string = TRUE)

ss.bact.seqs <- data.frame(taxonomy = rep(names(ss.bact.input.fa), sapply(ss.bact.input.fa, length)),
                 Seqs = unlist(ss.bact.input.fa))

West.Coast.ss.bact.rare.merged.df <- left_join(West.Coast.ss.bact.rare.merged.df, ss.bact.seqs, 
                                                        by= c("rowname" = "taxonomy"))

sample_data(West.Coast.bacterioplankton.rare)$Type <- "Bacterioplankton"

West.Coast.bacterioplankton.rare.merged <- merge_samples(West.Coast.bacterioplankton.rare, "Type")

West.Coast.bacterioplankton.rare.merged.df <- data.frame(t(otu_table(West.Coast.bacterioplankton.rare.merged)))
West.Coast.bacterioplankton.rare.merged.df <- rownames_to_column(West.Coast.bacterioplankton.rare.merged.df)

bacterioplankton.input.fa <- read.fasta("West.Coast.bacterioplankton.ASVs.fa", as.string = TRUE)

bacterioplankton.seqs <- data.frame(taxonomy = rep(names(bacterioplankton.input.fa), sapply(bacterioplankton.input.fa, length)),
                 Seqs = unlist(bacterioplankton.input.fa))

West.Coast.bacterioplankton.rare.merged.df <- left_join(West.Coast.bacterioplankton.rare.merged.df, bacterioplankton.seqs, 
                                                        by= c("rowname" = "taxonomy"))

West.Coast.Bacteria <- full_join(West.Coast.ss.bact.rare.merged.df, West.Coast.bacterioplankton.rare.merged.df, by="Seqs")

West.Coast.Bacteria.Shared <- West.Coast.Bacteria %>% filter(Sediment.Bacteria > 0) %>% filter(Bacterioplankton > 0)

(sum(West.Coast.Bacteria.Shared$Sediment.Bacteria, na.rm = TRUE) + sum(West.Coast.Bacteria.Shared$Bacterioplankton, na.rm = TRUE))/(sum(West.Coast.Bacteria$Sediment.Bacteria, na.rm = TRUE) + sum(West.Coast.Bacteria$Bacterioplankton, na.rm = TRUE))*100

shared.bacteria <- ggplot(West.Coast.Bacteria.Shared, aes(x=Sediment.Bacteria, y=Bacterioplankton)) + 
  geom_point() +
  theme_classic() +
  xlab("# reads in sediment") +
  ylab("# reads in plankton") +
  ggtitle("A") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))

```

```{r}
sample_data(West.Coast.ss.euks.rare)$Type <- "Sediment Eukaryotes"

West.Coast.ss.euks.rare.merged <- merge_samples(West.Coast.ss.euks.rare, "Type")

West.Coast.ss.euks.rare.merged.df <- data.frame(t(otu_table(West.Coast.ss.euks.rare.merged)))
West.Coast.ss.euks.rare.merged.df <- rownames_to_column(West.Coast.ss.euks.rare.merged.df)

ss.euks.input.fa <- read.fasta("West.Coast.Surf_euks.ASVs.fa", as.string = TRUE)

ss.euks.seqs <- data.frame(taxonomy = rep(names(ss.euks.input.fa), sapply(ss.euks.input.fa, length)),
                 Seqs = unlist(ss.euks.input.fa))

West.Coast.ss.euks.rare.merged.df <- left_join(West.Coast.ss.euks.rare.merged.df, ss.euks.seqs, 
                                                        by= c("rowname" = "taxonomy"))

sample_data(West.Coast.Phytoplankton.cleaned.rare)$Type <- "Phytoplankton"

West.Coast.Phytoplankton.cleaned.rare.merged <- merge_samples(West.Coast.Phytoplankton.cleaned.rare, "Type")

West.Coast.Phytoplankton.cleaned.rare.merged.df <- data.frame(t(otu_table(West.Coast.Phytoplankton.cleaned.rare.merged)))
West.Coast.Phytoplankton.cleaned.rare.merged.df <- rownames_to_column(West.Coast.Phytoplankton.cleaned.rare.merged.df)

phytoplankton.input.fa <- read.fasta("West.Coast.phytoplankton.ASVs.fa", as.string = TRUE)

phytoplankton.seqs <- data.frame(taxonomy = rep(names(phytoplankton.input.fa), sapply(phytoplankton.input.fa, length)),
                 Seqs = unlist(phytoplankton.input.fa))

West.Coast.Phytoplankton.cleaned.rare.merged.df <- left_join(West.Coast.Phytoplankton.cleaned.rare.merged.df, phytoplankton.seqs, 
                                                        by= c("rowname" = "taxonomy"))

West.Coast.Eukaryotes <- full_join(West.Coast.ss.euks.rare.merged.df, West.Coast.Phytoplankton.cleaned.rare.merged.df, by="Seqs")

West.Coast.Eukaryotes.Shared <- West.Coast.Eukaryotes %>% filter(Sediment.Eukaryotes > 0) %>% filter(Phytoplankton > 0)

(sum(West.Coast.Eukaryotes.Shared$Sediment.Eukaryotes, na.rm = TRUE) + sum(West.Coast.Eukaryotes.Shared$Phytoplankton, na.rm = TRUE))/(sum(West.Coast.Eukaryotes$Sediment.Eukaryotes, na.rm = TRUE) + sum(West.Coast.Eukaryotes$Phytoplankton, na.rm = TRUE))*100

shared.eukaryotes <- ggplot(West.Coast.Eukaryotes.Shared, aes(x=Sediment.Eukaryotes, y=Phytoplankton)) + 
  geom_point() +
  theme_classic() +
  xlab("# reads in sediment") +
  ylab("# reads in plankton") +
  ggtitle("B") +
  theme(plot.margin = unit(c(0,1,0,0), "lines"))

```

```{r}
combined <- plot_grid(shared.bacteria, shared.eukaryotes, align = "hv")

cairo_ps(file="Shared.eps", width=8, height=4, bg="transparent")
combined
dev.off()
```

#Composition

```{r}

West.Coast.bacterioplankton.rare.family <- tax_glom(West.Coast.bacterioplankton.rare, "Family")

West.Coast.bacterioplankton.rare.perc = transform_sample_counts(West.Coast.bacterioplankton.rare.family, function(OTU) OTU/sum(OTU)*100)

top10 <- names(sort(taxa_sums(West.Coast.bacterioplankton.rare.perc), decreasing=TRUE))[1:11]

West.Coast.bacterioplankton.rare.perc.top10 <- prune_taxa(top10, West.Coast.bacterioplankton.rare.perc)

West.Coast.bacterioplankton.rare.perc_df <- psmelt(West.Coast.bacterioplankton.rare.perc.top10)

West.Coast.bacterioplankton.rare.perc_df.wider <- West.Coast.bacterioplankton.rare.perc_df %>%
  ungroup() %>%
  dplyr::select(Phylum, Class, Family, Sample, Abundance) %>%
  pivot_wider(names_from = Sample, values_from = Abundance) %>%
  filter(Family != "Unclassified_Class_Verrucomicrobiae") %>% 
  filter(Family != "Unclassified_Class_Acidimicrobiia") %>% 
  select(Phylum, Class, Family, '47198', '39050', '46723', '38955', '47193', '46954', '46212', 
                                                     '46775', '54655', '47491', "47252", '38664', '46725', '39225', '46484', 
                                                     '46496', '46493', '39191', '54720', '45434', '46959', '46774', '47225', 
                                                     '45686', '46138', '41679', '45688', '45472', '8138', '46494', '39621', 
                                                     '38421', '54177', '38665')

colnames(West.Coast.bacterioplankton.rare.perc_df.wider) <- c("Phylum", "Class", "Family", "L47198", "L39050", "L46723", "L38955", "L47193", "L46954", "L46212", 
                                                     "L46775", "L54655", "L47491", "L47252", "L38664", "L46725", "L39225", "L46484", 
                                                     "L46496", "L46493", "L39191", "L54720", "L45434", "L46959", "L46774", "L47225", 
                                                     "L45686", "L46138", "L41679", "L45688", "L45472", "L8138", "L46494", "L39621", 
                                                     "L38421", "L54177", "L38665")

West.Coast.bacterioplankton.rare.perc_df.wider1 <- West.Coast.bacterioplankton.rare.perc_df.wider %>%
  dplyr::add_row(Phylum = "Other", Class = "Other",Family = "Other",
    L47198 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L47198), 
    L39050 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L39050), 
    L46723 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46723), 
    L38955 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L38955), 
    L47193 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L47193), 
    L46954 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46954), 
    L46212 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46212), 
    L46775 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46775), 
    L54655 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L54655), 
    L47491 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L47491), 
    L47252 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L47252), 
    L38664 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L38664), 
    L46725 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46725), 
    L39225 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L39225), 
    L46484 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46484), 
    L46496 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46496), 
    L46493 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46493), 
    L39191 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L39191), 
    L54720 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L54720), 
    L45434 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L45434), 
    L46959 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46959), 
    L46774 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46774), 
    L47225 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L47225), 
    L45686 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L45686), 
    L46138 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46138), 
    L41679 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L41679), 
    L45688 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L45688), 
    L45472 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L45472),
    L8138 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L8138),
    L46494 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L46494),
    L39621 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L39621),
    L38421 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L38421),
    L54177 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L54177),
    L38665 = 100-sum(West.Coast.bacterioplankton.rare.perc_df.wider$L38665))

West.Coast.bacterioplankton.rare.perc_df.wider2 <- pivot_longer(West.Coast.bacterioplankton.rare.perc_df.wider1, cols=L47198:L38665, names_to = "Samples", values_to = "Abundance")

West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L47198", "47198", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L39050", "39050", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46723", "46723", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L38955", "38955", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L47193", "47193", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46954", "46954", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46212", "46212", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46775", "46775", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L54655", "54655", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L47491", "47491", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L47252", "47252", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L38664", "38664", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46725", "46725", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L39225", "39225", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46484", "46484", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46496", "46496", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46493", "46493", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L39191", "39191", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L54720", "54720", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L45434", "45434", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46959", "46959", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46774", "46774", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L47225", "47225", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L45686", "45686", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46138", "46138", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L41679", "41679", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L45688", "45688", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L45472", "45472", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L8138",  "8138", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L46494", "46494", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L39621", "39621", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L38421", "38421", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples) 
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L54177", "54177", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)
West.Coast.bacterioplankton.rare.perc_df.wider2$Samples <- gsub("L38665", "38665", West.Coast.bacterioplankton.rare.perc_df.wider2$Samples)

map.ssbact <- data.frame(sample_data(West.Coast.bacterioplankton.rare))
map.ssbact$Code <- rownames(map.ssbact)
map.ssbact <- map.ssbact %>% select(Lake, Code, Longitude, Latitude) 

West.Coast.bacterioplankton.rare.perc_df.wider3 <- left_join(West.Coast.bacterioplankton.rare.perc_df.wider2, map.ssbact, by=c("Samples" = "Code"))

West.Coast.bacterioplankton.rare.perc_df.wider3 <- West.Coast.bacterioplankton.rare.perc_df.wider3 %>% filter(Family != "Other")

library(forcats)

compositionplot.ss.bacterioplankton <- West.Coast.bacterioplankton.rare.perc_df.wider3 %>%
  mutate(Samples = fct_reorder(Samples, Latitude)) %>%
  ggplot(aes(x=Samples, y=Family, size=Abundance, color=Abundance)) + 
  geom_point() + 
   # scale_color_continuous(name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25),
   #                    type = "viridis") +
  scale_color_gradient2(midpoint=50, low="blue", mid = "red", high="yellow", name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25)) + 
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(size=14, angle = 90, hjust = 1, vjust=0.4)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_text(size=8)) + 
  theme(legend.title=element_text(size=16, color="black")) + 
  theme(legend.text=element_text(size=14, color="black")) + 
  theme(plot.title=element_text(size=14, color="black")) + 
  theme(axis.text.y= element_text(size=12, color="black")) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  ggtitle("Bacterioplankton") + 
  scale_size(name="Proportional Abundance",range = c(0, 5), limits = c(0, 100), breaks=seq(0, 100, by=25)) +
  theme(legend.position = "")

bacterioplankton.composition.summary <- West.Coast.bacterioplankton.rare.perc_df.wider3 %>% 
                                              group_by(Family) %>% 
                                              summarise(mean = mean(Abundance),
                                                        sd = sd(Abundance))

compositionplot.bacterioplankton.summary <- bacterioplankton.composition.summary %>%
  mutate(Family = fct_reorder(Family, -mean)) %>%
  ggplot(aes(x=Family, y=mean)) + 
  geom_bar(stat="identity", fill="grey65") + 
  geom_errorbar(aes(ymin = pmax(mean-sd,0), ymax = mean+sd), position = "dodge", width = 0.1) +
  theme_bw() +
  theme(axis.text.x = element_text(size=14, angle = 90, hjust=1, vjust=0.4),
        axis.text.y = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=14),
        panel.grid = element_blank()) +
  ylab("Mean relative proportion (%)") +
  ggtitle("Bacterioplankton")
```

```{r}
West.Coast.zooplankton.cleaned.ps.pruned.rare.class <- tax_glom(West.Coast.zooplankton.cleaned.rare, "Family")

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc = transform_sample_counts(West.Coast.zooplankton.cleaned.ps.pruned.rare.class, function(OTU) OTU/sum(OTU)*100)

top10 <- names(sort(taxa_sums(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc), decreasing=TRUE))[1:19]

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc.top10 <- prune_taxa(top10, West.Coast.zooplankton.cleaned.ps.pruned.rare.perc)


West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df <- psmelt(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc.top10)

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider <- West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df %>%
  ungroup() %>%
  dplyr::select(Phylum, Class, Family, Sample, Abundance) %>%
  pivot_wider(names_from = Sample, values_from = Abundance) %>%
  filter(Family != "Unclassified_Phylum_Arthropoda") %>%
  filter(Family != "Unclassified_Order_Diplostraca") %>%
  filter(Family != "Unclassified_Kingdom_Eukaryota") %>%
  filter(Family != "Unclassified_Class_Hexanauplia") %>%
  filter(Family != "Unclassified_Order_Calanoida") %>% 
  filter(Family != "Unclassified_Order_Cyclopoida") %>%
  filter(Family != "Unclassified_Order_Harpacticoida") %>% 
  filter(Family != "Unclass_Harpacticoida") %>% 
  filter(Family != "Unclassified_Class_Magnoliopsida") %>% 
  select(Phylum, Class, Family, '47198', '39050', '46723', '38955', '47193', '46954', '46212', 
                                                     '46775', '54655', '47491', "47252", '38664', '46725', '39225', '46484', 
                                                     '46496', '46493', '39191', '54720', '45434', '46959', '46774', '47225', 
                                                     '45686', '46138', '41679', '45688', '45472', '8138', '46494', '39621', 
                                                     '38421', '54177', '38665')


colnames(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider) <- c("Phylum", "Class", "Family", "L47198", "L39050", "L46723", "L38955", 
                                                                           "L47193", "L46954", "L46212", 
                                                     "L46775", "L54655", "L47491", "L47252", "L38664", "L46725", "L39225", "L46484", 
                                                     "L46496", "L46493", "L39191", "L54720", "L45434", "L46959", "L46774", "L47225", 
                                                     "L45686", "L46138", "L41679", "L45688", "L45472", "L8138", "L46494", "L39621", 
                                                     "L38421", "L54177", "L38665")


West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider1 <- West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider %>%
  dplyr::add_row(Phylum = "Other", Class = "Other",Family = "Other",
    L47198 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L47198), 
    L39050 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L39050), 
    L46723 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46723), 
    L38955 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L38955), 
    L47193 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L47193), 
    L46954 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46954), 
    L46212 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46212), 
    L46775 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46775), 
    L54655 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L54655), 
    L47491 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L47491), 
    L47252 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L47252), 
    L38664 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L38664), 
    L46725 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46725), 
    L39225 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L39225), 
    L46484 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46484), 
    L46496 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46496), 
    L46493 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46493), 
    L39191 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L39191), 
    L54720 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L54720), 
    L45434 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L45434), 
    L46959 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46959), 
    L46774 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46774), 
    L47225 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L47225), 
    L45686 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L45686), 
    L46138 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46138), 
    L41679 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L41679), 
    L45688 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L45688), 
    L45472 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L45472),
    L8138 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L8138),
    L46494 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L46494),
    L39621 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L39621),
    L38421 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L38421),
    L54177 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L54177),
    L38665 = 100-sum(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider$L38665))

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2 <- pivot_longer(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider1, cols=L47198:L38665, names_to = "Samples", values_to = "Abundance")

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L47198", "47198", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L39050", "39050", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46723", "46723", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L38955", "38955", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L47193", "47193", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46954", "46954", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46212", "46212", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46775", "46775", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L54655", "54655", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L47491", "47491", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L47252", "47252", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L38664", "38664", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46725", "46725", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L39225", "39225", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46484", "46484", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46496", "46496", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46493", "46493", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L39191", "39191", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L54720", "54720", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L45434", "45434", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46959", "46959", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46774", "46774", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L47225", "47225", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L45686", "45686", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46138", "46138", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L41679", "41679", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L45688", "45688", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L45472", "45472", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L8138",  "8138", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L46494", "46494", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L39621", "39621", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L38421", "38421", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples) 
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L54177", "54177", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)
West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples <- gsub("L38665", "38665", West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2$Samples)


map.zoo <- read.csv("mapzooplankton.csv")

map.zoo <- map.zoo %>% filter(SampleID != "IANTHE") %>% select(Lake, Code, Longitude, Latitude) 

map.zoo$Code <- as.character(map.zoo$Code)

West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3 <- left_join(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider2, map.zoo, by=c("Samples" = "Code"))


West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3 <- West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>% filter(Family != "Other")


West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3$Samples <- as.character(West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3$Samples)

compositionplot.zooplankton <- West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>%
  mutate(Samples = fct_reorder(Samples, Latitude)) %>%
  ggplot(aes(x=Samples, y=Family, size=Abundance, color=Abundance)) + 
  geom_point() + 
  #scale_color_continuous(name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25),
  #                     type = "viridis") +
  scale_color_gradient2(midpoint=50, low="blue", mid = "red", high="yellow", name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25)) + 
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(size=14, angle = 90, hjust = 1, vjust=0.4)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_text(size=8)) + 
  theme(legend.title=element_text(size=16, color="black")) + 
  theme(legend.text=element_text(size=14, color="black")) + 
  theme(plot.title=element_text(size=14, color="black")) + 
  theme(axis.text.y= element_text(size=12, color="black")) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  scale_size(name="Proportional Abundance",range = c(0, 5), limits = c(0, 100), breaks=seq(0, 100, by=25)) + 
  ggtitle("Large Planktonic Eukaryotes") + 
  theme(legend.position = "")

zooplankton.composition.summary <- West.Coast.zooplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>% 
                                              group_by(Family) %>% 
                                              summarise(mean = mean(Abundance),
                                                        sd = sd(Abundance))

compositionplot.zooplankton.summary <- zooplankton.composition.summary %>%
  mutate(Family = fct_reorder(Family, -mean)) %>%
  ggplot(aes(x=Family, y=mean)) + 
  geom_bar(stat="identity", fill="grey65") + 
  geom_errorbar(aes(ymin = pmax(mean-sd,0), ymax = mean+sd), position = "dodge", width = 0.1) +
  theme_bw() +
  theme(axis.text.x = element_text(size=14, angle = 90, hjust=1, vjust=0.4),
        axis.text.y = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=14),
        panel.grid = element_blank()) +
  ylab("Mean relative proportion (%)") +
  ggtitle("Large Planktonic Eukaryotes")
  

```

```{r}
West.Coast.Phytoplankton.cleaned.rare.class <- tax_glom(West.Coast.Phytoplankton.cleaned.rare, "Family")

West.Coast.Phytoplankton.cleaned.rare.perc = transform_sample_counts(West.Coast.Phytoplankton.cleaned.rare.class, function(OTU) OTU/sum(OTU)*100)

top10 <- names(sort(taxa_sums(West.Coast.Phytoplankton.cleaned.rare.perc), decreasing=TRUE))[1:15]

West.Coast.Phytoplankton.cleaned.rare.perc.top10 <- prune_taxa(top10, West.Coast.Phytoplankton.cleaned.rare.perc)


West.Coast.Phytoplankton.cleaned.rare.perc_df <- psmelt(West.Coast.Phytoplankton.cleaned.rare.perc.top10)

West.Coast.Phytoplankton.cleaned.rare.perc_df.wider <- West.Coast.Phytoplankton.cleaned.rare.perc_df %>%
  ungroup() %>%
  dplyr::select(Phylum, Class, Family, Sample, Abundance) %>%
  pivot_wider(names_from = Sample, values_from = Abundance) %>%
  filter(Family != "Unclassified_Class_Dinophyceae") %>%
  filter(Family != "Unclassified_Order_Flosculariaceae") %>%
  filter(Family != "Zygnemophyceae_XX") %>%
  filter(Family != "Unclassified_Order_Ploima") %>%
  filter(Family != "Chlamydomonadales_X") %>% 
  select(Phylum, Class, Family, '47198', '39050', '46723', '38955', '47193', '46954', '46212', 
                                                     #'46775', 
         '54655', '47491', "47252", '38664', '46725', '39225', '46484', 
                                                     '46496', '46493', '39191', '54720', '45434', '46959', '46774', '47225', 
                                                     '45686', '46138', '41679', '45688', '45472', '8138', '46494', '39621', 
                                                     '38421', '54177', '38665')


colnames(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider) <- c("Phylum", "Class", "Family", "L47198", "L39050", "L46723", "L38955", 
                                                                           "L47193", "L46954", "L46212", 
                                                    # "L46775", 
                                                     "L54655", "L47491", "L47252", "L38664", "L46725", "L39225", "L46484", 
                                                     "L46496", "L46493", "L39191", "L54720", "L45434", "L46959", "L46774", "L47225", 
                                                     "L45686", "L46138", "L41679", "L45688", "L45472", "L8138", "L46494", "L39621", 
                                                     "L38421", "L54177", "L38665")


West.Coast.Phytoplankton.cleaned.rare.perc_df.wider1 <- West.Coast.Phytoplankton.cleaned.rare.perc_df.wider %>%
  dplyr::add_row(Phylum = "Other", Class = "Other",Family = "Other",
    L47198 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L47198), 
    L39050 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L39050), 
    L46723 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46723), 
    L38955 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L38955), 
    L47193 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L47193), 
    L46954 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46954), 
    L46212 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46212), 
    #L46775 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46775), 
    L54655 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L54655), 
    L47491 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L47491), 
    L47252 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L47252), 
    L38664 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L38664), 
    L46725 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46725), 
    L39225 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L39225), 
    L46484 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46484), 
    L46496 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46496), 
    L46493 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46493), 
    L39191 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L39191), 
    L54720 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L54720), 
    L45434 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L45434), 
    L46959 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46959), 
    L46774 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46774), 
    L47225 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L47225), 
    L45686 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L45686), 
    L46138 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46138), 
    L41679 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L41679), 
    L45688 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L45688), 
    L45472 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L45472),
    L8138 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L8138),
    L46494 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L46494),
    L39621 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L39621),
    L38421 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L38421),
    L54177 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L54177),
    L38665 = 100-sum(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider$L38665))


West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2 <- pivot_longer(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider1, cols=L47198:L38665, names_to = "Samples", values_to = "Abundance")

West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L47198", "47198", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L39050", "39050", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46723", "46723", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L38955", "38955", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L47193", "47193", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46954", "46954", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46212", "46212", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
#West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46775", "46775", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L54655", "54655", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L47491", "47491", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L47252", "47252", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L38664", "38664", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46725", "46725", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L39225", "39225", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46484", "46484", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46496", "46496", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46493", "46493", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L39191", "39191", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L54720", "54720", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L45434", "45434", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46959", "46959", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46774", "46774", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L47225", "47225", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L45686", "45686", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46138", "46138", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L41679", "41679", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L45688", "45688", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L45472", "45472", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L8138",  "8138", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L46494", "46494", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L39621", "39621", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L38421", "38421", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples) 
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L54177", "54177", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)
West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples <- gsub("L38665", "38665", West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2$Samples)


map.phyto <- read.csv("mapphytoplankton.csv")

map.phyto <- map.phyto %>% filter(Run != "C") %>% filter(Run != "D") %>% select(Lake, Code, Longitude, Latitude) 

map.phyto$Code <- as.character(map.phyto$Code)

West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider3 <- left_join(West.Coast.Phytoplankton.cleaned.rare.perc_df.wider2, map.phyto, by=c("Samples" = "Code"))

West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider3 <- West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>% filter(Family != "Other")

West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider4 <- West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>%
                                                      dplyr::add_row(Samples = "46775", 
                                                                     Abundance = NA,
                                                                     Lake = "Rototekoiti",
                                                                     Longitude = 169.7684,
                                                                     Latitude = -43.64052,
                                                                     Family = "Ceratiaceae", 
                                                                     Class = "Dinophyceae", 
                                                                     Phylum = "Dinoflagellata")


West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider4$Samples <- as.character(West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider4$Samples)

compositionplot.Phytoplankton <- West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider4 %>%
  mutate(Samples = fct_reorder(Samples, Latitude)) %>%
  ggplot(aes(x=Samples, y=Family, size=Abundance, color=Abundance)) + 
  geom_point() + 
  #scale_color_continuous(name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25),
  #                     type = "viridis") +
  scale_color_gradient2(midpoint=50, low="blue", mid = "red", high="yellow", name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25)) + 
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(size=14, angle = 90, hjust = 1, vjust=0.4)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_text(size=8)) + 
  theme(legend.title=element_text(size=16, color="black")) + 
  theme(legend.text=element_text(size=14, color="black")) + 
  theme(plot.title=element_text(size=14, color="black")) + 
  theme(axis.text.y= element_text(size=12, color="black")) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  ggtitle("Small Planktonic Eukaryotes") + 
  scale_size(name="Proportional Abundance",range = c(0, 5), limits = c(0, 100), breaks=seq(0, 100, by=25)) +
  theme(legend.position = "")

Phytoplankton.composition.summary <- West.Coast.Phytoplankton.cleaned.ps.pruned.rare.perc_df.wider3 %>% 
                                              group_by(Family) %>% 
                                              summarise(mean = mean(Abundance),
                                                        sd = sd(Abundance))

compositionplot.Phytoplankton.summary <- Phytoplankton.composition.summary %>%
  mutate(Family = fct_reorder(Family, -mean)) %>%
  ggplot(aes(x=Family, y=mean)) + 
  geom_bar(stat="identity", fill="grey65") + 
  geom_errorbar(aes(ymin = pmax(mean-sd,0), ymax = mean+sd), position = "dodge", width = 0.1) +
  theme_bw() +
  theme(axis.text.x = element_text(size=14, angle = 90, hjust=1, vjust=0.4),
        axis.text.y = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=14),
        panel.grid = element_blank()) +
  ylab("Mean relative proportion (%)") +
  ggtitle("Small Planktonic Eukaryotes")
```


```{r}
West.Coast.ss.euks.rare.family <- tax_glom(West.Coast.ss.euks.rare, "Family")

West.Coast.ss.euks.rare.perc = transform_sample_counts(West.Coast.ss.euks.rare.family, function(OTU) OTU/sum(OTU)*100)

top10 <- names(sort(taxa_sums(West.Coast.ss.euks.rare.perc), decreasing=TRUE))[1:16]

West.Coast.ss.euks.rare.perc.top10 <- prune_taxa(top10, West.Coast.ss.euks.rare.perc)

West.Coast.ss.euks.rare.perc_df <- psmelt(West.Coast.ss.euks.rare.perc.top10)

West.Coast.ss.euks.rare.perc_df.wider <- West.Coast.ss.euks.rare.perc_df %>%
  ungroup() %>%
  dplyr::select(Phylum, Class, Family, Sample, Abundance) %>%
  pivot_wider(names_from = Sample, values_from = Abundance) %>%
  filter(Family != "Unclassified_Phylum_Fungi") %>%
  filter(Family != "Unclassified_Phylum_Cercozoa") %>%
  filter(Family != "Pansomonadida_X") %>%
  filter(Family != "NA")%>%
  filter(Family != "Chlorellales_X")%>%
  filter(Family != "Embryophyceae_XX")%>%
  filter(Family != "Cryptomycotina_X")%>%
  filter(Family != "Strombidiida_A_X")%>%
  filter(Family != "Unclassified_Kingdom_Eukaryota")%>%
  filter(Family != "Unclassified_Order_Podocopida")%>%
  filter(Family != "Sphaeropleales_X")%>%
  filter(Family != "Unclassified_Class_Dinophyceae")%>%
  filter(Family != "Unclassified_Order_Ploima")


colnames(West.Coast.ss.euks.rare.perc_df.wider) <- c("Phylum", "Class", "Family", "L47198", "L39050", "L46723", "L38955", "L47193", "L46954", "L46212", 
                                                     "L46775", "L54655", "L47491", "L47252", "L38664", "L46725", "L39225", "L46484", 
                                                     "L46496", "L46493", "L39191", "L54720", "L45434", "L46959", "L46774", "L47225", 
                                                     "L45686", "L46138", "L41679", "L45688", "L45472", "L8138", "L46494", "L39621", 
                                                     "L38421", "L54177", "L38665")


West.Coast.ss.euks.rare.perc_df.wider1 <- West.Coast.ss.euks.rare.perc_df.wider %>%
  dplyr::add_row(Phylum = "Other", Class = "Other",Family = "Other",
    L47198 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L47198), 
    L39050 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L39050), 
    L46723 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46723), 
    L38955 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L38955), 
    L47193 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L47193), 
    L46954 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46954), 
    L46212 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46212), 
    L46775 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46775), 
    L54655 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L54655), 
    L47491 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L47491), 
    L47252 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L47252), 
    L38664 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L38664), 
    L46725 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46725), 
    L39225 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L39225), 
    L46484 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46484), 
    L46496 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46496), 
    L46493 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46493), 
    L39191 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L39191), 
    L54720 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L54720), 
    L45434 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L45434), 
    L46959 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46959), 
    L46774 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46774), 
    L47225 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L47225), 
    L45686 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L45686), 
    L46138 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46138), 
    L41679 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L41679), 
    L45688 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L45688), 
    L45472 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L45472),
    L8138 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L8138),
    L46494 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L46494),
    L39621 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L39621),
    L38421 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L38421),
    L54177 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L54177),
    L38665 = 100-sum(West.Coast.ss.euks.rare.perc_df.wider$L38665))


West.Coast.ss.euks.rare.perc_df.wider2 <- pivot_longer(West.Coast.ss.euks.rare.perc_df.wider1, cols=L47198:L38665, names_to = "Samples", values_to = "Abundance")


West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L47198", "47198", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L39050", "39050", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46723", "46723", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L38955", "38955", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L47193", "47193", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46954", "46954", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46212", "46212", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46775", "46775", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L54655", "54655", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L47491", "47491", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L47252", "47252", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L38664", "38664", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46725", "46725", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L39225", "39225", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46484", "46484", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46496", "46496", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46493", "46493", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L39191", "39191", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L54720", "54720", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L45434", "45434", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46959", "46959", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46774", "46774", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L47225", "47225", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L45686", "45686", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46138", "46138", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L41679", "41679", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L45688", "45688", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L45472", "45472", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L8138",  "8138", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L46494", "46494", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L39621", "39621", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L38421", "38421", West.Coast.ss.euks.rare.perc_df.wider2$Samples) 
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L54177", "54177", West.Coast.ss.euks.rare.perc_df.wider2$Samples)
West.Coast.ss.euks.rare.perc_df.wider2$Samples <- gsub("L38665", "38665", West.Coast.ss.euks.rare.perc_df.wider2$Samples)

map.sseuks <- data.frame(sample_data(West.Coast.ss.euks.rare))
map.sseuks$Code <- rownames(map.sseuks)
map.sseuks <- map.sseuks %>% select(Lake, Code, Longitude, Latitude) 

West.Coast.ss.euks.rare.perc_df.wider3 <- left_join(West.Coast.ss.euks.rare.perc_df.wider2, map.sseuks, by=c("Samples" = "Code"))

West.Coast.ss.euks.rare.perc_df.wider3 <- West.Coast.ss.euks.rare.perc_df.wider3 %>% filter(Family != "Other")

library(forcats)

compositionplot.ss.euks <- West.Coast.ss.euks.rare.perc_df.wider3 %>%
  mutate(Samples = fct_reorder(Samples, Latitude)) %>%
  ggplot(aes(x=Samples, y=Family, size=Abundance, color=Abundance)) + 
  geom_point() + 
  #scale_color_continuous(name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25),
  #                     type = "viridis") +
  scale_color_gradient2(midpoint=50, low="blue", mid = "red", high="yellow", name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25)) + 
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(size=14, angle = 90, hjust = 1, vjust=0.4)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_text(size=8)) + 
  theme(legend.title=element_text(size=16, color="black")) + 
  theme(legend.text=element_text(size=14, color="black")) + 
  theme(plot.title=element_text(size=14, color="black")) + 
  theme(axis.text.y= element_text(size=12, color="black")) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  ggtitle("Surface Sediment Eukaryotes") + 
  scale_size(name="Proportional Abundance",range = c(0, 5), limits = c(0, 100), breaks=seq(0, 100, by=25)) +
  theme(legend.position = "")

ss_euks.composition.summary <- West.Coast.ss.euks.rare.perc_df.wider3 %>% 
                                              group_by(Family) %>% 
                                              summarise(mean = mean(Abundance),
                                                        sd = sd(Abundance))

compositionplot.ss.euks.summary <- ss_euks.composition.summary %>%
  mutate(Family = fct_reorder(Family, -mean)) %>%
  ggplot(aes(x=Family, y=mean)) + 
  geom_bar(stat="identity", fill="grey65") + 
  geom_errorbar(aes(ymin = pmax(mean-sd,0), ymax = mean+sd), position = "dodge", width = 0.1) +
  theme_bw() +
  theme(axis.text.x = element_text(size=14, angle = 90, hjust=1, vjust=0.4),
        axis.text.y = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=14),
        panel.grid = element_blank()) +
  ylab("Mean relative proportion (%)") +
  ggtitle("Surface Sediment Eukaryotes")
```

```{r}
West.Coast.ss.bact.rare.family <- tax_glom(West.Coast.ss.bact.rare, "Family")

West.Coast.ss.bact.rare.perc = transform_sample_counts(West.Coast.ss.bact.rare.family, function(OTU) OTU/sum(OTU)*100)

top10 <- names(sort(taxa_sums(West.Coast.ss.bact.rare.perc), decreasing=TRUE))[1:11]

West.Coast.ss.bact.rare.perc.top10 <- prune_taxa(top10, West.Coast.ss.bact.rare.perc)

West.Coast.ss.bact.rare.perc_df <- psmelt(West.Coast.ss.bact.rare.perc.top10)

West.Coast.ss.bact.rare.perc_df.wider <- West.Coast.ss.bact.rare.perc_df %>%
  ungroup() %>%
  dplyr::select(Phylum, Class, Family, Sample, Abundance) %>%
  pivot_wider(names_from = Sample, values_from = Abundance) %>%
  filter(Family != "Acidobacteriae_fa") %>% 
  filter(Family != "Unclassified_Phylum_MBNT15") %>% 
  filter(Family != "Unclassified_Class_Thermodesulfovibrionia") %>% 
  filter(Family != "Unclassified_Order_Vicinamibacterales") %>% 
  filter(Family != "Unclassified_Order_JG36-TzT-191") %>% 
  filter(Family != "Unclassified_Class_KD4-96") %>% 
  select(Phylum, Class, Family, '47198', '39050', '46723', '38955', '47193', '46954', '46212', 
                                                     '46775', '54655', '47491', '38664', '46725', '39225', '46484', 
                                                     '46496', '46493', '39191', '54720', '45434', '46959', '46774', '47225', 
                                                     '45686', '46138', '41679', '45688', '45472', '8138', '46494', '39621', 
                                                     '38421', '54177', '38665')

colnames(West.Coast.ss.bact.rare.perc_df.wider) <- c("Phylum", "Class", "Family", "L47198", "L39050", "L46723", "L38955", "L47193", "L46954", "L46212", 
                                                     "L46775", "L54655", "L47491", "L38664", "L46725", "L39225", "L46484", 
                                                     "L46496", "L46493", "L39191", "L54720", "L45434", "L46959", "L46774", "L47225", 
                                                     "L45686", "L46138", "L41679", "L45688", "L45472", "L8138", "L46494", "L39621", 
                                                     "L38421", "L54177", "L38665")

West.Coast.ss.bact.rare.perc_df.wider1 <- West.Coast.ss.bact.rare.perc_df.wider %>%
  dplyr::add_row(Phylum = "Other", Class = "Other",Family = "Other",
    L47198 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L47198), 
    L39050 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L39050), 
    L46723 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46723), 
    L38955 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L38955), 
    L47193 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L47193), 
    L46954 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46954), 
    L46212 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46212), 
    L46775 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46775), 
    L54655 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L54655), 
    L47491 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L47491), 
    L38664 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L38664), 
    L46725 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46725), 
    L39225 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L39225), 
    L46484 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46484), 
    L46496 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46496), 
    L46493 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46493), 
    L39191 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L39191), 
    L54720 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L54720), 
    L45434 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L45434), 
    L46959 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46959), 
    L46774 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46774), 
    L47225 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L47225), 
    L45686 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L45686), 
    L46138 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46138), 
    L41679 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L41679), 
    L45688 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L45688), 
    L45472 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L45472),
    L8138 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L8138),
    L46494 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L46494),
    L39621 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L39621),
    L38421 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L38421),
    L54177 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L54177),
    L38665 = 100-sum(West.Coast.ss.bact.rare.perc_df.wider$L38665))

West.Coast.ss.bact.rare.perc_df.wider2 <- pivot_longer(West.Coast.ss.bact.rare.perc_df.wider1, cols=L47198:L38665, names_to = "Samples", values_to = "Abundance")

West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L47198", "47198", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L39050", "39050", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46723", "46723", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L38955", "38955", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L47193", "47193", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46954", "46954", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46212", "46212", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46775", "46775", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L54655", "54655", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L47491", "47491", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L38664", "38664", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46725", "46725", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L39225", "39225", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46484", "46484", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46496", "46496", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46493", "46493", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L39191", "39191", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L54720", "54720", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L45434", "45434", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46959", "46959", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46774", "46774", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L47225", "47225", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L45686", "45686", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46138", "46138", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L41679", "41679", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L45688", "45688", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L45472", "45472", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L8138",  "8138", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L46494", "46494", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L39621", "39621", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L38421", "38421", West.Coast.ss.bact.rare.perc_df.wider2$Samples) 
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L54177", "54177", West.Coast.ss.bact.rare.perc_df.wider2$Samples)
West.Coast.ss.bact.rare.perc_df.wider2$Samples <- gsub("L38665", "38665", West.Coast.ss.bact.rare.perc_df.wider2$Samples)

map.ssbact <- data.frame(sample_data(West.Coast.ss.bact.rare))
map.ssbact$Code <- rownames(map.ssbact)
map.ssbact <- map.ssbact %>% select(Lake, Code, Longitude, Latitude) 

West.Coast.ss.bact.rare.perc_df.wider3 <- left_join(West.Coast.ss.bact.rare.perc_df.wider2, map.ssbact, by=c("Samples" = "Code"))

West.Coast.ss.bact.rare.perc_df.wider3 <- West.Coast.ss.bact.rare.perc_df.wider3 %>% filter(Family != "Other")


West.Coast.ss.bact.rare.perc_df.wider3 <- West.Coast.ss.bact.rare.perc_df.wider3 %>%
                                                      dplyr::add_row(Lake = "Stoney Tarn", 
                                                                     Abundance = NA,
                                                                     Samples = "47252",
                                                                     Longitude = 170.4458,
                                                                     Latitude = -43.87433,
                                                                     Family = "Pedosphaeraceae", 
                                                                     Class = "Verrucomicrobiae", 
                                                                     Phylum = "Verrucomicrobiota")

library(forcats)

compositionplot.ss.bact <- West.Coast.ss.bact.rare.perc_df.wider3 %>%
  mutate(Lake = fct_reorder(Lake, Latitude)) %>%
  ggplot(aes(x=Lake, y=Family, size=Abundance, color=Abundance)) + 
  geom_point() + 
  #scale_color_continuous(name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25),
  #                     type = "viridis") +
  scale_color_gradient2(midpoint=50, low="blue", mid = "red", high="yellow", name="Proportional Abundance",guide = "legend", limits=c(0,100), breaks=seq(0, 100, by=25)) + 
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(size=14, angle = 90, hjust = 1, vjust=0.4)) +
  #theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_text(size=8)) + 
  theme(legend.title=element_text(size=16, color="black")) + 
  theme(legend.text=element_text(size=14, color="black")) + 
  theme(plot.title=element_text(size=14, color="black")) + 
  theme(axis.text.y= element_text(size=12, color="black")) +
  theme(panel.background = element_rect(fill="transparent"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_rect(fill="transparent", colour=NA),
        legend.background = element_rect(fill="transparent"),
        legend.key = element_rect(fill = NA)) + 
  ggtitle("Surface Sediment Bacteria") + 
  scale_size(name="Proportional Abundance",range = c(0, 5), limits = c(0, 100), breaks=seq(0, 100, by=25)) +
  theme(legend.position = "bottom") 


ss_bact.composition.summary <- West.Coast.ss.bact.rare.perc_df.wider3 %>% 
                                              filter(Lake != "Stoney Tarn") %>%
                                              group_by(Family) %>% 
                                              summarise(mean = mean(Abundance),
                                                        sd = sd(Abundance))


compositionplot.ss.bact.summary <- ss_bact.composition.summary %>%
  mutate(Family = fct_reorder(Family, -mean)) %>%
  ggplot(aes(x=Family, y=mean)) + 
  geom_bar(stat="identity", fill="grey65") + 
  geom_errorbar(aes(ymin = pmax(mean-sd,0), ymax = mean+sd), position = "dodge", width = 0.1) +
  theme_bw() +
  theme(axis.text.x = element_text(size=14, angle = 90, hjust=1, vjust=0.4),
        axis.text.y = element_text(size=14),
        axis.title = element_text(size=14),
        panel.grid = element_blank()) +
  ylab("Mean relative proportion (%)") +
  ggtitle("Surface Sediment Bacteria")
```

```{r}
combined.plot <- plot_grid(compositionplot.ss.bacterioplankton, compositionplot.zooplankton,
          compositionplot.Phytoplankton, compositionplot.ss.euks, 
          compositionplot.ss.bact, ncol = 1, align = "v", axis = "lr", rel_heights = c(1, 1,1,1,1.8))

cairo_ps(file="WestCoast.Composition.eps", width=16, height=18)
combined.plot
dev.off()
```

```{r}
combined.plot.summary <- plot_grid(compositionplot.bacterioplankton.summary, compositionplot.zooplankton.summary,
          compositionplot.Phytoplankton.summary, compositionplot.ss.euks.summary, 
          compositionplot.ss.bact.summary, ncol = 1, align = "v", axis = "lrtb", rel_heights = c(1, 0.95,1,1,1.15))

cairo_ps(file="WestCoast.Composition_summary.eps", width=16, height=22)
combined.plot.summary
dev.off()
```


#Assembly Processing

```{r}

write.csv(otu_table(West.Coast.ss.bact.rare), "SS_bact_comm.csv")
write.csv(sample_data(West.Coast.ss.bact.rare), "SS_bact_comm.sd.csv")

tree.SS_bact <- read.nexus(file="SS_bact.nexus")

SS_bact.comm <-read.csv("SS_bact_comm.csv", h=T, row.names=1)

comm_SS_bact_r <- SS_bact.comm/rowSums(SS_bact.comm)

comm.SS_bact.r.p <- match.phylo.comm(tree.SS_bact, comm_SS_bact_r)$comm
tree.SS_bact.r.p <- match.phylo.comm(tree.SS_bact, comm_SS_bact_r)$phy

write.csv(t(comm.SS_bact.r.p), "comm.SS_bact.r.p.csv")

phydist_SS_bact <- cophenetic(tree.SS_bact.r.p)

phydist_SS_bact_hel <- decostand(phydist_SS_bact, method="hellinger")

table_w <- read.delim2("comm.SS_bact.r.p_TS.txt",header=FALSE, row.names=1, dec='.') #get table

table <- t(table_w)

headTable <- colnames(table)
x <- headTable[1] # The first column is a constant Eh_mV and is being compared to OTU

# Create the final_table empty

m <- matrix(0, nrow = 0, ncol = 3) 
final_table <- data.frame(m)
colnames(final_table)  <- c('ymax','d50', 'xmax') 

# the for loop, to get all otus 1 by 1 compared to Eh_mV

for(i in seq(2,length(headTable),1)) #start at 2 because the 1 is the Eh_mv
{
  y <- headTable[i] # get otu name
  tablei <-  table[,c(x,y)] # extract 2 columns to create a new table
  
  head  <- colnames(tablei)
  
  ymax = which.max(tablei[,2]) # get the max relative abundance of the OTU
  
  MaxRow <- tablei[ymax,] # create a table with the line of the max OTU value
  #MaxRow <-as.table(MaxRow)
  
  #tableiInf <- tablei[tablei[,1] <= MaxRow[1],] #create two tables greater and lower than the x coord
  #tableiSup <- tablei[tablei[,1] >= MaxRow[1],] #of the max
  
  tableiInf <- subset(tablei,tablei[,1]<= MaxRow[1]) #create two tables greater and lower than the x coord
  tableiSup <- subset(tablei,tablei[,1]>= MaxRow[1]) #of the max
  
  tableiInfno0 <- subset(tableiInf,tableiInf[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  tableiSupno0 <- subset(tableiSup,tableiSup[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  
  #Determine the lowest and highest Eh value where you find the OTU
  
  xInf <- min(tableiInfno0[,1])
  xSup <- max(tableiSupno0[,1])
  
  # calculate the d50
  
  d50  <- 0.68*abs(xSup-xInf)/2
  
  # create the final table for the otu in the loop
  
  tablei_fin  <- matrix(c(MaxRow[2],d50,MaxRow[1]),ncol=3)
  rownames(tablei_fin)  <- c(head[2])
  colnames(tablei_fin)  <- c('ymax','d50', 'xmax')
  
  # add to the final_table the values for each otu 
  final_table <- rbind(final_table,tablei_fin)
  
  
}

write.table(final_table_noInf_T, "niche_SS_bact_TS.txt", sep="\t")

optima_dist_T_SS_bact <- vegdist(final_table$xmax, method="euclidean")

mantel_correlog_SS_bact_T <- mantel.correlog(optima_dist_T_SS_bact, phydist_SS_bact_hel, nperm=99) 

pdf(file="mantel_correl_SS_bact_TS.pdf")
plot(mantel_correlog_SS_bact_T)
dev.off()

```

```{r}

tree.SS_bact <- read.nexus(file="SS_bact.nexus")

SS_bact.comm <-read.csv("SS_bact_comm.csv", h=T, row.names=1)

comm.SS_bact_log <- log(SS_bact.comm + 1)

comm.SS_bact_log.p <- match.phylo.comm(tree.SS_bact, comm.SS_bact_log)$comm
tree.SS_bact_log.p <- match.phylo.comm(tree.SS_bact, comm.SS_bact_log)$phy

otu <- comm.SS_bact_log.p; ##set your OTU table to the object 'otu' if you have it imported already##
dim(otu); # this gives the dimensions
otu[1:5,1:5]; # this gives a look at the first 5 rows and columns

## read in the phylogeny

phylo <- tree.SS_bact_log.p
phylo; # a summary of the phylogeny
#plot.phylo(phylo,typ="fan"); # a quick plot

## make sure the names on the phylogeny are ordered the same as the names in otu table

match.phylo.otu = match.phylo.data(phylo, t(otu));
str(match.phylo.otu);

## calculate empirical betaMNTD

beta.mntd.weighted = as.matrix(comdistnt(t(match.phylo.otu$data),cophenetic(match.phylo.otu$phy),abundance.weighted=T));
dim(beta.mntd.weighted);
beta.mntd.weighted[1:5,1:5];
write.csv(beta.mntd.weighted,'betaMNTD_weighted_SS_bact.csv',quote=F);

identical(colnames(match.phylo.otu$data),colnames(beta.mntd.weighted)); # just a check, should be TRUE
identical(colnames(match.phylo.otu$data),rownames(beta.mntd.weighted)); # just a check, should be TRUE

# calculate randomized betaMNTD

beta.reps = 999; # number of randomizations

rand.weighted.bMNTD.comp = array(c(-999),dim=c(ncol(match.phylo.otu$data),ncol(match.phylo.otu$data),beta.reps));
dim(rand.weighted.bMNTD.comp);

for (rep in 1:beta.reps) {
  
  rand.weighted.bMNTD.comp[,,rep] = as.matrix(comdistnt(t(match.phylo.otu$data),taxaShuffle(cophenetic(match.phylo.otu$phy)),abundance.weighted=T,exclude.conspecifics = F));
  
  print(c(date(),rep));
  
}

weighted.bNTI = matrix(c(NA),nrow=ncol(match.phylo.otu$data),ncol=ncol(match.phylo.otu$data));
dim(weighted.bNTI);

for (columns in 1:(ncol(match.phylo.otu$data)-1)) {
  for (rows in (columns+1):ncol(match.phylo.otu$data)) {
    
    rand.vals = rand.weighted.bMNTD.comp[rows,columns,];
    weighted.bNTI[rows,columns] = (beta.mntd.weighted[rows,columns] - mean(rand.vals)) / sd(rand.vals);
    rm("rand.vals");
    
  };
};

rownames(weighted.bNTI) = colnames(match.phylo.otu$data);
colnames(weighted.bNTI) = colnames(match.phylo.otu$data);
weighted.bNTI;
write.csv(weighted.bNTI,"weighted_bNTI_SS_bact.csv",quote=F);

weighted.bNTI.melt <- melt(weighted.bNTI)
write.csv(weighted.bNTI.melt,"weighted_bNTI.melt_SS_bact.csv",quote=F, row.names = FALSE)

```

```{r}

tree.SS_bact <- read.nexus(file="SS_bact.nexus")

SS_bact.comm <-read.csv("SS_bact_comm.csv", h=T, row.names=1)


comm.SS_bact_log <- log(SS_bact.comm + 1)

comm.SS_bact_log.p <- match.phylo.comm(tree.SS_bact, comm.SS_bact_log)$comm
tree.SS_bact_log.p <- match.phylo.comm(tree.SS_bact, comm.SS_bact_log)$phy

rand.time=999 # usually use 1000 for real data.
nworker=6 # parallel computing thread number
set.seed(201)
RaupCrick_results=RC.pc(comm=comm.SS_bact_log.p, rand = rand.time,
         nworker = nworker, weighted = TRUE,
         sig.index="RC")

RaupCrick_results.mat <- as.matrix(RaupCrick_results$index)
RaupCrick_results.mat[!lower.tri(RaupCrick_results.mat)] <- NA

RaupCrick_results.melt <- melt(RaupCrick_results.mat)

write.csv(RaupCrick_results.mat, "RaupCrick_results_SS_bact.csv")
write.csv(RaupCrick_results.melt, "RaupCrick_results_SS_bact.melt.csv", row.names = FALSE)
```

```{r}
write.csv(otu_table(West.Coast.ss.euks.rare), "SS_euks_comm.csv")
write.csv(sample_data(West.Coast.ss.euks.rare), "SS_euks_comm.sd.csv")

tree.SS_euks <- read.nexus(file="SS_euks.nexus")

SS_euks.comm <-read.csv("SS_euks_comm.csv", h=T, row.names=1)

comm_SS_euks_r <- SS_euks.comm/rowSums(SS_euks.comm)

comm.SS_euks.r.p <- match.phylo.comm(tree.SS_euks, comm_SS_euks_r)$comm
tree.SS_euks.r.p <- match.phylo.comm(tree.SS_euks, comm_SS_euks_r)$phy

write.csv(t(comm.SS_euks.r.p), "comm.SS_euks.r.p.csv")

phydist_SS_euks <- cophenetic(tree.SS_euks.r.p)

phydist_SS_euks_hel <- decostand(phydist_SS_euks, method="hellinger")

table_w <- read.delim2("comm.SS_euks.r.p_TS.txt",header=FALSE, row.names=1, dec='.') #get table

table <- t(table_w)

headTable <- colnames(table)
x <- headTable[1] # The first column is a constant Eh_mV and is being compared to OTU

# Create the final_table empty

m <- matrix(0, nrow = 0, ncol = 3) 
final_table <- data.frame(m)
colnames(final_table)  <- c('ymax','d50', 'xmax') 

# the for loop, to get all otus 1 by 1 compared to Eh_mV

for(i in seq(2,length(headTable),1)) #start at 2 because the 1 is the Eh_mv
{
  y <- headTable[i] # get otu name
  tablei <-  table[,c(x,y)] # extract 2 columns to create a new table
  
  head  <- colnames(tablei)
  
  ymax = which.max(tablei[,2]) # get the max relative abundance of the OTU
  
  MaxRow <- tablei[ymax,] # create a table with the line of the max OTU value
  #MaxRow <-as.table(MaxRow)
  
  #tableiInf <- tablei[tablei[,1] <= MaxRow[1],] #create two tables greater and lower than the x coord
  #tableiSup <- tablei[tablei[,1] >= MaxRow[1],] #of the max
  
  tableiInf <- subset(tablei,tablei[,1]<= MaxRow[1]) #create two tables greater and lower than the x coord
  tableiSup <- subset(tablei,tablei[,1]>= MaxRow[1]) #of the max
  
  tableiInfno0 <- subset(tableiInf,tableiInf[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  tableiSupno0 <- subset(tableiSup,tableiSup[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0

  #Determine the lowest and highest Eh value where you find the OTU
  
  xInf <- min(tableiInfno0[,1])
  xSup <- max(tableiSupno0[,1])
  
  # calculate the d50
  
  d50  <- 0.68*abs(xSup-xInf)/2
  
  # create the final table for the otu in the loop
  
  tablei_fin  <- matrix(c(MaxRow[2],d50,MaxRow[1]),ncol=3)
  rownames(tablei_fin)  <- c(head[2])
  colnames(tablei_fin)  <- c('ymax','d50', 'xmax')
  
  # add to the final_table the values for each otu 
  final_table <- rbind(final_table,tablei_fin)
  
  
}

write.table(final_table_noInf_T, "niche_SS_euks_TS.txt", sep="\t")

optima_dist_T_SS_euks <- vegdist(final_table$xmax, method="euclidean")

mantel_correlog_SS_euks_T <- mantel.correlog(optima_dist_T_SS_euks, phydist_SS_euks_hel, nperm=99) 

pdf(file="mantel_correl_SS_euks_TS.pdf")
plot(mantel_correlog_SS_euks_T)
dev.off()

```

```{r}
tree.SS_euks <- read.nexus(file="SS_euks.nexus")

SS_euks.comm <-read.csv("SS_euks_comm.csv", h=T, row.names=1)

comm.SS_euks_log <- log(SS_euks.comm + 1)

comm.SS_euks_log.p <- match.phylo.comm(tree.SS_euks, comm.SS_euks_log)$comm
tree.SS_euks_log.p <- match.phylo.comm(tree.SS_euks, comm.SS_euks_log)$phy

#write.csv(comm.ARMS_log.p, "comm.ARMS_log.p.csv")

otu <- comm.SS_euks_log.p; ##set your OTU table to the object 'otu' if you have it imported already##
dim(otu); # this gives the dimensions
otu[1:5,1:5]; # this gives a look at the first 5 rows and columns

## read in the phylogeny

phylo <- tree.SS_euks_log.p
phylo; # a summary of the phylogeny
#plot.phylo(phylo,typ="fan"); # a quick plot

## make sure the names on the phylogeny are ordered the same as the names in otu table

match.phylo.otu = match.phylo.data(phylo, t(otu));
str(match.phylo.otu);

## calculate empirical betaMNTD

beta.mntd.weighted = as.matrix(comdistnt(t(match.phylo.otu$data),cophenetic(match.phylo.otu$phy),abundance.weighted=T));
dim(beta.mntd.weighted);
beta.mntd.weighted[1:5,1:5];
write.csv(beta.mntd.weighted,'betaMNTD_weighted_SS_euks.csv',quote=F);

identical(colnames(match.phylo.otu$data),colnames(beta.mntd.weighted)); # just a check, should be TRUE
identical(colnames(match.phylo.otu$data),rownames(beta.mntd.weighted)); # just a check, should be TRUE

# calculate randomized betaMNTD

beta.reps = 999; # number of randomizations

rand.weighted.bMNTD.comp = array(c(-999),dim=c(ncol(match.phylo.otu$data),ncol(match.phylo.otu$data),beta.reps));
dim(rand.weighted.bMNTD.comp);

for (rep in 1:beta.reps) {
  
  rand.weighted.bMNTD.comp[,,rep] = as.matrix(comdistnt(t(match.phylo.otu$data),taxaShuffle(cophenetic(match.phylo.otu$phy)),abundance.weighted=T,exclude.conspecifics = F));
  
  print(c(date(),rep));
  
}

weighted.bNTI = matrix(c(NA),nrow=ncol(match.phylo.otu$data),ncol=ncol(match.phylo.otu$data));
dim(weighted.bNTI);

for (columns in 1:(ncol(match.phylo.otu$data)-1)) {
  for (rows in (columns+1):ncol(match.phylo.otu$data)) {
    
    rand.vals = rand.weighted.bMNTD.comp[rows,columns,];
    weighted.bNTI[rows,columns] = (beta.mntd.weighted[rows,columns] - mean(rand.vals)) / sd(rand.vals);
    rm("rand.vals");
    
  };
};

rownames(weighted.bNTI) = colnames(match.phylo.otu$data);
colnames(weighted.bNTI) = colnames(match.phylo.otu$data);
weighted.bNTI;
write.csv(weighted.bNTI,"weighted_bNTI_SS_euks.csv",quote=F);

weighted.bNTI.melt <- melt(weighted.bNTI)
write.csv(weighted.bNTI.melt,"weighted_bNTI.melt_SS_euks.csv",quote=F, row.names = FALSE)

```


```{r}
tree.SS_euks <- read.nexus(file="SS_euks.nexus")

SS_euks.comm <-read.csv("SS_euks_comm.csv", h=T, row.names=1)


comm.SS_euks_log <- log(SS_euks.comm + 1)

comm.SS_euks_log.p <- match.phylo.comm(tree.SS_euks, comm.SS_euks_log)$comm
tree.SS_euks_log.p <- match.phylo.comm(tree.SS_euks, comm.SS_euks_log)$phy

set.seed(201)
RaupCrick_results=RC.pc(comm.SS_euks_log.p, rand = rand.time,
         nworker = nworker, weighted = TRUE,
         sig.index="RC")

RaupCrick_results.mat <- as.matrix(RaupCrick_results$index)
RaupCrick_results.mat[!lower.tri(RaupCrick_results.mat)] <- NA

RaupCrick_results.melt <- melt(RaupCrick_results.mat)

#RaupCrick_results.melt <- RaupCrick_results.melt %>% select(-X)

write.csv(RaupCrick_results.mat, "RaupCrick_results_SS_euks.csv")
write.csv(RaupCrick_results.melt, "RaupCrick_results_SS_euks.melt.csv", row.names = FALSE)
```

```{r}

write.csv(otu_table(West.Coast.Phytoplankton.cleaned.rare), "Phyto_comm.csv")
write.csv(sample_data(West.Coast.Phytoplankton.cleaned.rare), "Phyto_comm.sd.csv")

tree.phyto <- read.nexus(file="phyto.tree")

Phyto.comm <-read.csv("Phyto_comm.csv", h=T, row.names=1)

comm_Phyto_r <- Phyto.comm/rowSums(Phyto.comm)

comm.Phyto.r.p <- match.phylo.comm(tree.phyto, comm_Phyto_r)$comm
tree.Phyto.r.p <- match.phylo.comm(tree.phyto, comm_Phyto_r)$phy

write.csv(t(comm.Phyto.r.p), "comm.Phyto.r.p.csv")

phydist_Phyto <- cophenetic(tree.Phyto.r.p)

phydist_Phyto_hel <- decostand(phydist_Phyto, method="hellinger")

table_w <- read.delim2("comm.Phyto.r.p_TS.txt",header=FALSE, row.names=1, dec='.') #get table

table <- t(table_w)

headTable <- colnames(table)
x <- headTable[1] # The first column is a constant Eh_mV and is being compared to OTU

# Create the final_table empty

m <- matrix(0, nrow = 0, ncol = 3) 
final_table <- data.frame(m)
colnames(final_table)  <- c('ymax','d50', 'xmax') 

# the for loop, to get all otus 1 by 1 compared to Eh_mV

for(i in seq(2,length(headTable),1)) #start at 2 because the 1 is the Eh_mv
{
  y <- headTable[i] # get otu name
  tablei <-  table[,c(x,y)] # extract 2 columns to create a new table
  
  head  <- colnames(tablei)
  
  ymax = which.max(tablei[,2]) # get the max relative abundance of the OTU
  
  MaxRow <- tablei[ymax,] # create a table with the line of the max OTU value
  #MaxRow <-as.table(MaxRow)
  
  #tableiInf <- tablei[tablei[,1] <= MaxRow[1],] #create two tables greater and lower than the x coord
  #tableiSup <- tablei[tablei[,1] >= MaxRow[1],] #of the max
  
  tableiInf <- subset(tablei,tablei[,1]<= MaxRow[1]) #create two tables greater and lower than the x coord
  tableiSup <- subset(tablei,tablei[,1]>= MaxRow[1]) #of the max
  
  tableiInfno0 <- subset(tableiInf,tableiInf[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  tableiSupno0 <- subset(tableiSup,tableiSup[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  
  #Determine the lowest and highest Eh value where you find the OTU
  
  xInf <- min(tableiInfno0[,1])
  xSup <- max(tableiSupno0[,1])
  
  # calculate the d50
  
  d50  <- 0.68*abs(xSup-xInf)/2
  
  # create the final table for the otu in the loop
  
  tablei_fin  <- matrix(c(MaxRow[2],d50,MaxRow[1]),ncol=3)
  rownames(tablei_fin)  <- c(head[2])
  colnames(tablei_fin)  <- c('ymax','d50', 'xmax')
  
  # add to the final_table the values for each otu 
  final_table <- rbind(final_table,tablei_fin)
  
  
}

write.table(final_table_noInf_T, "niche_phyto_TS.txt", sep="\t")

optima_dist_T_phyto <- vegdist(final_table$xmax, method="euclidean")

mantel_correlog_phyto_T <- mantel.correlog(optima_dist_T_phyto, phydist_Phyto_hel, nperm=99) 

pdf(file="mantel_correl_phyto_TS.pdf")
plot(mantel_correlog_phyto_T)
dev.off()

```

```{r}

tree.phyto <- read.nexus(file="phytoplankton_tree.nexus")

Phyto.comm <-read.csv("Phyto_comm.csv", h=T, row.names=1)

comm.Phyto_log <- log(Phyto.comm + 1)

comm.Phyto_log.p <- match.phylo.comm(tree.phyto, comm.Phyto_log)$comm
tree.Phyto_log.p <- match.phylo.comm(tree.phyto, comm.Phyto_log)$phy

#write.csv(comm.ARMS_log.p, "comm.ARMS_log.p.csv")

otu <- comm.Phyto_log.p; ##set your OTU table to the object 'otu' if you have it imported already##
dim(otu); # this gives the dimensions
otu[1:5,1:5]; # this gives a look at the first 5 rows and columns

## read in the phylogeny

phylo <- tree.Phyto_log.p
phylo; # a summary of the phylogeny
#plot.phylo(phylo,typ="fan"); # a quick plot

## make sure the names on the phylogeny are ordered the same as the names in otu table

match.phylo.otu = match.phylo.data(phylo, t(otu));
str(match.phylo.otu);

## calculate empirical betaMNTD

beta.mntd.weighted = as.matrix(comdistnt(t(match.phylo.otu$data),cophenetic(match.phylo.otu$phy),abundance.weighted=T));
dim(beta.mntd.weighted);
beta.mntd.weighted[1:5,1:5];
write.csv(beta.mntd.weighted,'betaMNTD_weighted_Phyto.csv',quote=F);

identical(colnames(match.phylo.otu$data),colnames(beta.mntd.weighted)); # just a check, should be TRUE
identical(colnames(match.phylo.otu$data),rownames(beta.mntd.weighted)); # just a check, should be TRUE

# calculate randomized betaMNTD

beta.reps = 999; # number of randomizations

rand.weighted.bMNTD.comp = array(c(-999),dim=c(ncol(match.phylo.otu$data),ncol(match.phylo.otu$data),beta.reps));
dim(rand.weighted.bMNTD.comp);

for (rep in 1:beta.reps) {
  
  rand.weighted.bMNTD.comp[,,rep] = as.matrix(comdistnt(t(match.phylo.otu$data),taxaShuffle(cophenetic(match.phylo.otu$phy)),abundance.weighted=T,exclude.conspecifics = F));
  
  print(c(date(),rep));
  
}

weighted.bNTI = matrix(c(NA),nrow=ncol(match.phylo.otu$data),ncol=ncol(match.phylo.otu$data));
dim(weighted.bNTI);

for (columns in 1:(ncol(match.phylo.otu$data)-1)) {
  for (rows in (columns+1):ncol(match.phylo.otu$data)) {
    
    rand.vals = rand.weighted.bMNTD.comp[rows,columns,];
    weighted.bNTI[rows,columns] = (beta.mntd.weighted[rows,columns] - mean(rand.vals)) / sd(rand.vals);
    rm("rand.vals");
    
  };
};

rownames(weighted.bNTI) = colnames(match.phylo.otu$data);
colnames(weighted.bNTI) = colnames(match.phylo.otu$data);
weighted.bNTI;
write.csv(weighted.bNTI,"weighted_bNTI_Phyto.csv",quote=F);

weighted.bNTI.melt <- melt(weighted.bNTI)
write.csv(weighted.bNTI.melt,"weighted_bNTI.melt_Phyto.csv",quote=F, row.names = FALSE)

```

```{r}
tree.phyto <- read.nexus(file="phytoplankton_tree.nexus")

Phyto.comm <-read.csv("Phyto_comm.csv", h=T, row.names=1)

comm.Phyto_log <- log(Phyto.comm + 1)

comm.Phyto_log.p <- match.phylo.comm(tree.phyto, comm.Phyto_log)$comm
tree.Phyto_log.p <- match.phylo.comm(tree.phyto, comm.Phyto_log)$phy

set.seed(201)
RaupCrick_results=RC.pc(comm.Phyto_log.p, rand = rand.time,
         nworker = nworker, weighted = TRUE,
         sig.index="RC")

RaupCrick_results.mat <- as.matrix(RaupCrick_results$index)
RaupCrick_results.mat[!lower.tri(RaupCrick_results.mat)] <- NA

RaupCrick_results.melt <- melt(RaupCrick_results.mat)

write.csv(RaupCrick_results.mat, "RaupCrick_results_Phyto.csv")
write.csv(RaupCrick_results.melt, "RaupCrick_results_Phyto.melt.csv", row.names = FALSE)
```

```{r}
write.csv(otu_table(West.Coast.zooplankton.cleaned.rare), "Zoo_comm.csv")
write.csv(sample_data(West.Coast.zooplankton.cleaned.rare), "Zoo_comm.sd.csv")

tree.Zoo <- read.nexus(file="zooplankton_tree.nexus")

Zoo.comm <-read.csv("Zoo_comm.csv", h=T, row.names=1)

comm_Zoo_r <- Zoo.comm/rowSums(Zoo.comm)

comm.Zoo.r.p <- match.phylo.comm(tree.Zoo, comm_Zoo_r)$comm
tree.Zoo.r.p <- match.phylo.comm(tree.Zoo, comm_Zoo_r)$phy

write.csv(t(comm.Zoo.r.p), "comm.Zoo.r.p.csv")

phydist_Zoo <- cophenetic(tree.Zoo.r.p)

phydist_Zoo_hel <- decostand(phydist_Zoo, method="hellinger")

table_w <- read.delim2("comm.Zoo.r.p_TS.txt",header=FALSE, row.names=1, dec='.') #get table

table <- t(table_w)

headTable <- colnames(table)
x <- headTable[1] # The first column is a constant Eh_mV and is being compared to OTU

# Create the final_table empty

m <- matrix(0, nrow = 0, ncol = 3) 
final_table <- data.frame(m)
colnames(final_table)  <- c('ymax','d50', 'xmax') 

# the for loop, to get all otus 1 by 1 compared to Eh_mV

for(i in seq(2,length(headTable),1)) #start at 2 because the 1 is the Eh_mv
{
  y <- headTable[i] # get otu name
  tablei <-  table[,c(x,y)] # extract 2 columns to create a new table
  
  head  <- colnames(tablei)
  
  ymax = which.max(tablei[,2]) # get the max relative abundance of the OTU
  
  MaxRow <- tablei[ymax,] # create a table with the line of the max OTU value
  #MaxRow <-as.table(MaxRow)
  
  #tableiInf <- tablei[tablei[,1] <= MaxRow[1],] #create two tables greater and lower than the x coord
  #tableiSup <- tablei[tablei[,1] >= MaxRow[1],] #of the max
  
  tableiInf <- subset(tablei,tablei[,1]<= MaxRow[1]) #create two tables greater and lower than the x coord
  tableiSup <- subset(tablei,tablei[,1]>= MaxRow[1]) #of the max
  
  tableiInfno0 <- subset(tableiInf,tableiInf[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  tableiSupno0 <- subset(tableiSup,tableiSup[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  
  
  #Determine the lowest and highest Eh value where you find the OTU
  
  xInf <- min(tableiInfno0[,1])
  xSup <- max(tableiSupno0[,1])
  
  # calculate the d50
  
  d50  <- 0.68*abs(xSup-xInf)/2
  
  # create the final table for the otu in the loop
  
  tablei_fin  <- matrix(c(MaxRow[2],d50,MaxRow[1]),ncol=3)
  rownames(tablei_fin)  <- c(head[2])
  colnames(tablei_fin)  <- c('ymax','d50', 'xmax')
  
  # add to the final_table the values for each otu 
  final_table <- rbind(final_table,tablei_fin)
}

write.table(final_table_noInf_T, "niche_Zoo_TS.txt", sep="\t")

optima_dist_T_Zoo <- vegdist(final_table$xmax, method="euclidean")

mantel_correlog_Zoo_T <- mantel.correlog(optima_dist_T_Zoo, phydist_Zoo_hel, nperm=99) 

pdf(file="mantel_correl_Zoo_TS.pdf")
plot(mantel_correlog_Zoo_T)
dev.off()

```

```{r}
tree.Zoo <- read.nexus(file="zooplankton_tree.nexus")

Zoo.comm <-read.csv("Zoo_comm.csv", h=T, row.names=1)

comm.Zoo_log <- log(Zoo.comm + 1)

comm.Zoo_log.p <- match.phylo.comm(tree.Zoo, comm.Zoo_log)$comm
tree.Zoo_log.p <- match.phylo.comm(tree.Zoo, comm.Zoo_log)$phy

#write.csv(comm.ARMS_log.p, "comm.ARMS_log.p.csv")

otu <- comm.Zoo_log.p; ##set your OTU table to the object 'otu' if you have it imported already##
dim(otu); # this gives the dimensions
otu[1:5,1:5]; # this gives a look at the first 5 rows and columns

## read in the phylogeny

phylo <- tree.Zoo_log.p
phylo; # a summary of the phylogeny
#plot.phylo(phylo,typ="fan"); # a quick plot

## make sure the names on the phylogeny are ordered the same as the names in otu table

match.phylo.otu = match.phylo.data(phylo, t(otu));
str(match.phylo.otu);

## calculate empirical betaMNTD

beta.mntd.weighted = as.matrix(comdistnt(t(match.phylo.otu$data),cophenetic(match.phylo.otu$phy),abundance.weighted=T));
dim(beta.mntd.weighted);
beta.mntd.weighted[1:5,1:5];
write.csv(beta.mntd.weighted,'betaMNTD_weighted_Zoo.csv',quote=F);

identical(colnames(match.phylo.otu$data),colnames(beta.mntd.weighted)); # just a check, should be TRUE
identical(colnames(match.phylo.otu$data),rownames(beta.mntd.weighted)); # just a check, should be TRUE

# calculate randomized betaMNTD

beta.reps = 999; # number of randomizations

rand.weighted.bMNTD.comp = array(c(-999),dim=c(ncol(match.phylo.otu$data),ncol(match.phylo.otu$data),beta.reps));
dim(rand.weighted.bMNTD.comp);

for (rep in 1:beta.reps) {
  
  rand.weighted.bMNTD.comp[,,rep] = as.matrix(comdistnt(t(match.phylo.otu$data),taxaShuffle(cophenetic(match.phylo.otu$phy)),abundance.weighted=T,exclude.conspecifics = F));
  
  print(c(date(),rep));
  
}

weighted.bNTI = matrix(c(NA),nrow=ncol(match.phylo.otu$data),ncol=ncol(match.phylo.otu$data));
dim(weighted.bNTI);

for (columns in 1:(ncol(match.phylo.otu$data)-1)) {
  for (rows in (columns+1):ncol(match.phylo.otu$data)) {
    
    rand.vals = rand.weighted.bMNTD.comp[rows,columns,];
    weighted.bNTI[rows,columns] = (beta.mntd.weighted[rows,columns] - mean(rand.vals)) / sd(rand.vals);
    rm("rand.vals");
    
  };
};

rownames(weighted.bNTI) = colnames(match.phylo.otu$data);
colnames(weighted.bNTI) = colnames(match.phylo.otu$data);
weighted.bNTI;
write.csv(weighted.bNTI,"weighted_bNTI_Zoo.csv",quote=F);

weighted.bNTI.melt <- melt(weighted.bNTI)
write.csv(weighted.bNTI.melt,"weighted_bNTI.melt_Zoo.csv",quote=F, row.names = FALSE)

```


```{r}
tree.Zoo <- read.nexus(file="zooplankton_tree.nexus")

Zoo.comm <-read.csv("Zoo_comm.csv", h=T, row.names=1)

comm.Zoo_log <- log(Zoo.comm + 1)

comm.Zoo_log.p <- match.phylo.comm(tree.Zoo, comm.Zoo_log)$comm
tree.Zoo_log.p <- match.phylo.comm(tree.Zoo, comm.Zoo_log)$phy

set.seed(205)
RaupCrick_results=RC.pc(comm.Zoo_log.p, rand = rand.time,
         nworker = nworker, weighted = TRUE,
         sig.index="RC")

RaupCrick_results.mat <- as.matrix(RaupCrick_results$index)
RaupCrick_results.mat[!lower.tri(RaupCrick_results.mat)] <- NA

RaupCrick_results.melt <- melt(RaupCrick_results.mat)

write.csv(RaupCrick_results.mat, "RaupCrick_results_Zoo.csv")
write.csv(RaupCrick_results.melt, "RaupCrick_results_Zoo.melt.csv", row.names = FALSE)
```

```{r}
write.csv(otu_table(West.Coast.bacterioplankton.rare), "Bacterioplankton_comm.csv")
write.csv(sample_data(West.Coast.bacterioplankton.rare), "Bacterioplankton_comm.sd.csv")

tree.Bacterioplankton <- read.nexus(file="bacterioplankton_tree.nexus")

Bacterioplankton.comm <-read.csv("Bacterioplankton_comm.csv", h=T, row.names=1)

comm_Bacterioplankton_r <- Bacterioplankton.comm/rowSums(Bacterioplankton.comm)

comm.Bacterioplankton.r.p <- match.phylo.comm(tree.Bacterioplankton, comm_Bacterioplankton_r)$comm
tree.Bacterioplankton.r.p <- match.phylo.comm(tree.Bacterioplankton, comm_Bacterioplankton_r)$phy

write.csv(t(comm.Bacterioplankton.r.p), "comm.Bacterioplankton.r.p.csv")

phydist_Bacterioplankton <- cophenetic(tree.Bacterioplankton.r.p)

phydist_Bacterioplankton_hel <- decostand(phydist_Bacterioplankton, method="hellinger")

table_w <- read.delim2("comm.Bacterioplankton.r.p_TS.txt",header=FALSE, row.names=1, dec='.') #get table

table <- t(table_w)

headTable <- colnames(table)
x <- headTable[1] # The first column is a constant Eh_mV and is being compared to OTU

# Create the final_table empty

m <- matrix(0, nrow = 0, ncol = 3) 
final_table <- data.frame(m)
colnames(final_table)  <- c('ymax','d50', 'xmax') 

# the for loop, to get all otus 1 by 1 compared to Eh_mV

for(i in seq(2,length(headTable),1)) #start at 2 because the 1 is the Eh_mv
{
  y <- headTable[i] # get otu name
  tablei <-  table[,c(x,y)] # extract 2 columns to create a new table
  
  head  <- colnames(tablei)
  
  ymax = which.max(tablei[,2]) # get the max relative abundance of the OTU
  
  MaxRow <- tablei[ymax,] # create a table with the line of the max OTU value
  #MaxRow <-as.table(MaxRow)
  
  #tableiInf <- tablei[tablei[,1] <= MaxRow[1],] #create two tables greater and lower than the x coord
  #tableiSup <- tablei[tablei[,1] >= MaxRow[1],] #of the max
  
  tableiInf <- subset(tablei,tablei[,1]<= MaxRow[1]) #create two tables greater and lower than the x coord
  tableiSup <- subset(tablei,tablei[,1]>= MaxRow[1]) #of the max
  
  tableiInfno0 <- subset(tableiInf,tableiInf[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  tableiSupno0 <- subset(tableiSup,tableiSup[,2]>(MaxRow[2]/20)) #remove the observations where the relative OTU abundance is 0
  
  
  #Determine the lowest and highest Eh value where you find the OTU
  
  xInf <- min(tableiInfno0[,1])
  xSup <- max(tableiSupno0[,1])
  
  # calculate the d50
  
  d50  <- 0.68*abs(xSup-xInf)/2
  
  # create the final table for the otu in the loop
  
  tablei_fin  <- matrix(c(MaxRow[2],d50,MaxRow[1]),ncol=3)
  rownames(tablei_fin)  <- c(head[2])
  colnames(tablei_fin)  <- c('ymax','d50', 'xmax')
  
  # add to the final_table the values for each otu 
  final_table <- rbind(final_table,tablei_fin)
  
  
}

write.table(final_table_noInf_T, "niche_Bacterioplankton_TS.txt", sep="\t")

optima_dist_T_Bacterioplankton <- vegdist(final_table$xmax, method="euclidean")

mantel_correlog_Bacterioplankton_T <- mantel.correlog(optima_dist_T_Bacterioplankton, phydist_Bacterioplankton_hel, nperm=99) 

pdf(file="mantel_correl_Bacterioplankton_TS.pdf")
plot(mantel_correlog_Bacterioplankton_T)
dev.off()
```

```{r}
tree.Bacterioplankton <- read.nexus(file="bacterioplankton_tree.nexus")

Bacterioplankton.comm <-read.csv("Bacterioplankton_comm.csv", h=T, row.names=1)

comm.Bacterioplankton_log <- log(Bacterioplankton.comm + 1)

comm.Bacterioplankton_log.p <- match.phylo.comm(tree.Bacterioplankton, comm.Bacterioplankton_log)$comm
tree.Bacterioplankton_log.p <- match.phylo.comm(tree.Bacterioplankton, comm.Bacterioplankton_log)$phy

#write.csv(comm.ARMS_log.p, "comm.ARMS_log.p.csv")

otu <- comm.Bacterioplankton_log.p; ##set your OTU table to the object 'otu' if you have it imported already##
dim(otu); # this gives the dimensions
otu[1:5,1:5]; # this gives a look at the first 5 rows and columns

## read in the phylogeny

phylo <- tree.Bacterioplankton_log.p
phylo; # a summary of the phylogeny
#plot.phylo(phylo,typ="fan"); # a quick plot

## make sure the names on the phylogeny are ordered the same as the names in otu table

match.phylo.otu = match.phylo.data(phylo, t(otu));
str(match.phylo.otu);

## calculate empirical betaMNTD

beta.mntd.weighted = as.matrix(comdistnt(t(match.phylo.otu$data),cophenetic(match.phylo.otu$phy),abundance.weighted=T));
dim(beta.mntd.weighted);
beta.mntd.weighted[1:5,1:5];
write.csv(beta.mntd.weighted,'betaMNTD_weighted_Bacterioplankton.csv',quote=F);

identical(colnames(match.phylo.otu$data),colnames(beta.mntd.weighted)); # just a check, should be TRUE
identical(colnames(match.phylo.otu$data),rownames(beta.mntd.weighted)); # just a check, should be TRUE

# calculate randomized betaMNTD

beta.reps = 999; # number of randomizations

rand.weighted.bMNTD.comp = array(c(-999),dim=c(ncol(match.phylo.otu$data),ncol(match.phylo.otu$data),beta.reps));
dim(rand.weighted.bMNTD.comp);

for (rep in 1:beta.reps) {
  
  rand.weighted.bMNTD.comp[,,rep] = as.matrix(comdistnt(t(match.phylo.otu$data),taxaShuffle(cophenetic(match.phylo.otu$phy)),abundance.weighted=T,exclude.conspecifics = F));
  
  print(c(date(),rep));
  
}

weighted.bNTI = matrix(c(NA),nrow=ncol(match.phylo.otu$data),ncol=ncol(match.phylo.otu$data));
dim(weighted.bNTI);

for (columns in 1:(ncol(match.phylo.otu$data)-1)) {
  for (rows in (columns+1):ncol(match.phylo.otu$data)) {
    
    rand.vals = rand.weighted.bMNTD.comp[rows,columns,];
    weighted.bNTI[rows,columns] = (beta.mntd.weighted[rows,columns] - mean(rand.vals)) / sd(rand.vals);
    rm("rand.vals");
    
  };
};

rownames(weighted.bNTI) = colnames(match.phylo.otu$data);
colnames(weighted.bNTI) = colnames(match.phylo.otu$data);
weighted.bNTI;
write.csv(weighted.bNTI,"weighted_bNTI_Bacterioplankton.csv",quote=F);

weighted.bNTI.melt <- melt(weighted.bNTI)
write.csv(weighted.bNTI.melt,"weighted_bNTI.melt_Bacterioplankton.csv",quote=F, row.names = FALSE)
```


```{r}
tree.Bacterioplankton <- read.nexus(file="bacterioplankton_tree.nexus")

Bacterioplankton.comm <-read.csv("Bacterioplankton_comm.csv", h=T, row.names=1)

comm.Bacterioplankton_log <- log(Bacterioplankton.comm + 1)

comm.Bacterioplankton_log.p <- match.phylo.comm(tree.Bacterioplankton, comm.Bacterioplankton_log)$comm
tree.Bacterioplankton_log.p <- match.phylo.comm(tree.Bacterioplankton, comm.Bacterioplankton_log)$phy

set.seed(201)
RaupCrick_results=RC.pc(comm.Bacterioplankton_log.p, rand = rand.time,
         nworker = nworker, weighted = TRUE,
         sig.index="RC")

RaupCrick_results.mat <- as.matrix(RaupCrick_results$index)
RaupCrick_results.mat[!lower.tri(RaupCrick_results.mat)] <- NA

RaupCrick_results.melt <- melt(RaupCrick_results.mat)

write.csv(RaupCrick_results.mat, "RaupCrick_results_Bacterioplankton.csv")
write.csv(RaupCrick_results.melt, "RaupCrick_results_Bacterioplankton.melt.csv", row.names = FALSE)
```

#Assembly stats

```{r}
RaupCrick_results.melt.SS_bact <- read.csv("RaupCrick_results_SS_bact.melt.csv")
BNTI_results.melt.SS_bact <- read.csv("weighted_bNTI.melt_SS_bact.csv")

RaupCrick_results.melt.SS_bact1 <- RaupCrick_results.melt.SS_bact %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(RaupCrick_results.melt.SS_bact1)[2] <- "RC"

BNTI_results.melt.SS_bact1 <- BNTI_results.melt.SS_bact %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(BNTI_results.melt.SS_bact1)[2] <- "BNTI"

Combined.results.SS_bact <- left_join(BNTI_results.melt.SS_bact1, RaupCrick_results.melt.SS_bact1, by = "Comparison")

Combined.results.SS_bact$Type <- "NDP"

Combined.results.SS_bact$Type[Combined.results.SS_bact$RC > 0.95] <- "Dispersal Limitation"
Combined.results.SS_bact$Type[Combined.results.SS_bact$RC < -0.95] <- "Homogenizing Dispersal"

Combined.results.SS_bact$Type[Combined.results.SS_bact$BNTI > 2] <- "Variable Selection"
Combined.results.SS_bact$Type[Combined.results.SS_bact$BNTI < -2] <- "Homogeneous Selection"

write.csv(Combined.results.SS_bact, "Combined.results.SS_bact.csv")

summary.SS_bact <- Combined.results.SS_bact  %>% group_by(Type) %>% tally()

summary.SS_bact$Proportion <- summary.SS_bact$n/sum(summary.SS_bact$n)*100
```

```{r}
RaupCrick_results.melt.SS_euks <- read.csv("RaupCrick_results_SS_euks.melt.csv")
BNTI_results.melt.SS_euks <- read.csv("weighted_bNTI.melt_SS_euks.csv")

RaupCrick_results.melt.SS_euks1 <- RaupCrick_results.melt.SS_euks %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(RaupCrick_results.melt.SS_euks1)[2] <- "RC"

BNTI_results.melt.SS_euks1 <- BNTI_results.melt.SS_euks %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(BNTI_results.melt.SS_euks1)[2] <- "BNTI"

Combined.results.SS_euks <- left_join(BNTI_results.melt.SS_euks1, RaupCrick_results.melt.SS_euks1, by = "Comparison")

Combined.results.SS_euks$Type <- "NDP"

Combined.results.SS_euks$Type[Combined.results.SS_euks$RC > 0.95] <- "Dispersal Limitation"
Combined.results.SS_euks$Type[Combined.results.SS_euks$RC < -0.95] <- "Homogenizing Dispersal"

Combined.results.SS_euks$Type[Combined.results.SS_euks$BNTI > 2] <- "Variable Selection"
Combined.results.SS_euks$Type[Combined.results.SS_euks$BNTI < -2] <- "Homogeneous Selection"

write.csv(Combined.results.SS_euks, "Combined.results.SS_euks.csv")

summary.SS_euks <- Combined.results.SS_euks  %>% group_by(Type) %>% tally()

summary.SS_euks$Proportion <- summary.SS_euks$n/sum(summary.SS_euks$n)*100
```

```{r}
RaupCrick_results.melt.Phyto <- read.csv("RaupCrick_results_Phyto.melt.csv")
BNTI_results.melt.Phyto <- read.csv("weighted_bNTI.melt_Phyto.csv")

RaupCrick_results.melt.Phyto1 <- RaupCrick_results.melt.Phyto %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(RaupCrick_results.melt.Phyto1)[2] <- "RC"

BNTI_results.melt.Phyto1 <- BNTI_results.melt.Phyto %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(BNTI_results.melt.Phyto1)[2] <- "BNTI"

Combined.results.Phyto <- left_join(BNTI_results.melt.Phyto1, RaupCrick_results.melt.Phyto1, by = "Comparison")

Combined.results.Phyto$Type <- "NDP"

Combined.results.Phyto$Type[Combined.results.Phyto$RC > 0.95] <- "Dispersal Limitation"
Combined.results.Phyto$Type[Combined.results.Phyto$RC < -0.95] <- "Homogenizing Dispersal"

Combined.results.Phyto$Type[Combined.results.Phyto$BNTI > 2] <- "Variable Selection"
Combined.results.Phyto$Type[Combined.results.Phyto$BNTI < -2] <- "Homogeneous Selection"

write.csv(Combined.results.Phyto, "Combined.results.phyto.csv")

summary.Phyto <- Combined.results.Phyto  %>% group_by(Type) %>% tally()

summary.Phyto$Proportion <- summary.Phyto$n/sum(summary.Phyto$n)*100
```

```{r}
RaupCrick_results.melt.Zoo <- read.csv("RaupCrick_results_Zoo.melt.csv")
BNTI_results.melt.Zoo <- read.csv("weighted_bNTI.melt_Zoo.csv")

RaupCrick_results.melt.Zoo1 <- RaupCrick_results.melt.Zoo %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(RaupCrick_results.melt.Zoo1)[2] <- "RC"

BNTI_results.melt.Zoo1 <- BNTI_results.melt.Zoo %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(BNTI_results.melt.Zoo1)[2] <- "BNTI"

Combined.results.Zoo <- left_join(BNTI_results.melt.Zoo1, RaupCrick_results.melt.Zoo1, by = "Comparison")

Combined.results.Zoo$Type <- "NDP"

Combined.results.Zoo$Type[Combined.results.Zoo$RC > 0.95] <- "Dispersal Limitation"
Combined.results.Zoo$Type[Combined.results.Zoo$RC < -0.95] <- "Homogenizing Dispersal"

Combined.results.Zoo$Type[Combined.results.Zoo$BNTI > 2] <- "Variable Selection"
Combined.results.Zoo$Type[Combined.results.Zoo$BNTI < -2] <- "Homogeneous Selection"

write.csv(Combined.results.Zoo, "Combined.results.Zoo.csv")

summary.Zoo <- Combined.results.Zoo  %>% group_by(Type) %>% tally()

summary.Zoo$Proportion <- summary.Zoo$n/sum(summary.Zoo$n)*100
```

```{r}
RaupCrick_results.melt.Bacterioplankton <- read.csv("RaupCrick_results_Bacterioplankton.melt.csv")
BNTI_results.melt.Bacterioplankton <- read.csv("weighted_bNTI.melt_Bacterioplankton.csv")

RaupCrick_results.melt.Bacterioplankton1 <- RaupCrick_results.melt.Bacterioplankton %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(RaupCrick_results.melt.Bacterioplankton1)[2] <- "RC"

BNTI_results.melt.Bacterioplankton1 <- BNTI_results.melt.Bacterioplankton %>% filter(!is.na(value)) %>% unite("Comparison", Var1:Var2, remove = TRUE)

colnames(BNTI_results.melt.Bacterioplankton1)[2] <- "BNTI"

Combined.results.Bacterioplankton <- left_join(BNTI_results.melt.Bacterioplankton1, RaupCrick_results.melt.Bacterioplankton1, by = "Comparison")

Combined.results.Bacterioplankton$Type <- "NDP"

Combined.results.Bacterioplankton$Type[Combined.results.Bacterioplankton$RC > 0.95] <- "Dispersal Limitation"
Combined.results.Bacterioplankton$Type[Combined.results.Bacterioplankton$RC < -0.95] <- "Homogenizing Dispersal"

Combined.results.Bacterioplankton$Type[Combined.results.Bacterioplankton$BNTI > 2] <- "Variable Selection"
Combined.results.Bacterioplankton$Type[Combined.results.Bacterioplankton$BNTI < -2] <- "Homogeneous Selection"

write.csv(Combined.results.Bacterioplankton, "Combined.results.Bacterioplankton.csv")

summary.Bacterioplankton <- Combined.results.Bacterioplankton  %>% group_by(Type) %>% tally()

summary.Bacterioplankton$Proportion <- summary.Bacterioplankton$n/sum(summary.Bacterioplankton$n)*100
```

#Stochastic vs Distance

```{r}

sd.Bacterioplankton <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

sd.Bacterioplankton1 <- sd.Bacterioplankton %>% select(Lake.Code, Latitude, Longitude) %>% separate("Lake.Code", c("Lake", "X", "Code"), sep="\\.") %>% select(-X)


Combined.results.Bacterioplankton1 <- Combined.results.Bacterioplankton %>% separate("Comparison", c("Lake1", "Lake2"), sep = "_")


Combined.results.Bacterioplankton2 <- left_join(Combined.results.Bacterioplankton1, sd.Bacterioplankton1, by=c("Lake1" = "Code"))

colnames(Combined.results.Bacterioplankton2) <- c("Lake1", "Lake2","BNTI", "RC", "Type","Lake1.Name", "Lake1.Latitude", "Lake1.Longitude")


Combined.results.Bacterioplankton3 <- left_join(Combined.results.Bacterioplankton2, sd.Bacterioplankton1, by=c("Lake2" = "Code"))

colnames(Combined.results.Bacterioplankton3) <- c("Lake1", "Lake2","BNTI", "RC", "Type",
                                                  "Lake1.Name", "Lake1.Latitude", "Lake1.Longitude",
                                                  "Lake2.Name", "Lake2.Latitude", "Lake2.Longitude")

Combined.results.Bacterioplankton3.filtered <- Combined.results.Bacterioplankton3 %>% 
                                              filter(Type != "NDP") %>%
                                              filter(Type != "Variable Selection") %>%
                                              filter(Type != "Homogeneous Selection")

coord.lake1 <- Combined.results.Bacterioplankton3.filtered %>% select(Lake1.Longitude, Lake1.Latitude)

colnames(coord.lake1) <- c("Longitude", "Latitude")

coord.lake2 <- Combined.results.Bacterioplankton3.filtered %>% select(Lake2.Longitude, Lake2.Latitude)

colnames(coord.lake2) <- c("Longitude", "Latitude")


Combined.results.Bacterioplankton3.filtered$Distance <- distHaversine(coord.lake1, coord.lake2)


Combined.results.Bacterioplankton3.filtered.dist.sum <- Combined.results.Bacterioplankton3.filtered %>% select(Type, Distance) %>% group_by(Type) %>% summarise(Mean.Dist =mean(Distance))

Combined.results.Bacterioplankton3.filtered.dist <- Combined.results.Bacterioplankton3.filtered %>% select(Type, Distance)

kruskal.test(Distance ~ Type, data = Combined.results.Bacterioplankton3.filtered.dist)

#dunnTest(Distance ~ Type, data = Combined.results.Bacterioplankton3.filtered.dist, method="bh") 
```

```{r}
sd.Zooplankton <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare))

sd.Zooplankton1 <- sd.Zooplankton %>% select(Code, Latitude, Longitude)

sd.Zooplankton1 <- rownames_to_column(sd.Zooplankton1)

Combined.results.Zooplankton1 <- Combined.results.Zoo %>% separate("Comparison", c("Lake1", "Lake2"), sep = "_")


Combined.results.Zooplankton2 <- left_join(Combined.results.Zooplankton1, sd.Zooplankton1, by=c("Lake1" = "rowname"))

colnames(Combined.results.Zooplankton2) <- c("Lake1", "Lake2","BNTI", "RC", "Type","Lake1.Name", "Lake1.Latitude", "Lake1.Longitude")


Combined.results.Zooplankton3 <- left_join(Combined.results.Zooplankton2, sd.Zooplankton1, by=c("Lake2" = "rowname"))

colnames(Combined.results.Zooplankton3) <- c("Lake1", "Lake2","BNTI", "RC", "Type",
                                                  "Lake1.Name", "Lake1.Latitude", "Lake1.Longitude",
                                                  "Lake2.Name", "Lake2.Latitude", "Lake2.Longitude")


Combined.results.Zooplankton3.filtered <- Combined.results.Zooplankton3 %>% 
                                              filter(Type != "NDP") %>%
                                              filter(Type != "Variable Selection") %>%
                                              filter(Type != "Homogeneous Selection")


coord.lake1 <- Combined.results.Zooplankton3.filtered %>% select(Lake1.Longitude, Lake1.Latitude)

colnames(coord.lake1) <- c("Longitude", "Latitude")

coord.lake2 <- Combined.results.Zooplankton3.filtered %>% select(Lake2.Longitude, Lake2.Latitude)

colnames(coord.lake2) <- c("Longitude", "Latitude")



Combined.results.Zooplankton3.filtered$Distance <- distHaversine(coord.lake1, coord.lake2)


Combined.results.Zooplankton3.filtered.dist.sum <- Combined.results.Zooplankton3.filtered %>% select(Type, Distance) %>% group_by(Type) %>% summarise(Mean.Dist =mean(Distance))

Combined.results.Zooplankton3.filtered.dist <- Combined.results.Zooplankton3.filtered %>% select(Type, Distance)

kruskal.test(Distance ~ Type, data = Combined.results.Zooplankton3.filtered.dist)

#dunnTest(Distance ~ Type, data = Combined.results.Zooplankton3.filtered.dist, method="bh") 
```

```{r}
sd.Phytoplankton <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare ))

sd.Phytoplankton1 <- sd.Phytoplankton %>% select(Code, Latitude, Longitude)

sd.Phytoplankton1 <- rownames_to_column(sd.Phytoplankton1)

Combined.results.Phytoplankton1 <- Combined.results.Phyto %>% separate("Comparison", c("Lake1", "Lake2"), sep = "_")


Combined.results.Phytoplankton2 <- left_join(Combined.results.Phytoplankton1, sd.Phytoplankton1, by=c("Lake1" = "rowname"))

colnames(Combined.results.Phytoplankton2) <- c("Lake1", "Lake2","BNTI", "RC", "Type","Lake1.Name", "Lake1.Latitude", "Lake1.Longitude")


Combined.results.Phytoplankton3 <- left_join(Combined.results.Phytoplankton2, sd.Phytoplankton1, by=c("Lake2" = "rowname"))

colnames(Combined.results.Phytoplankton3) <- c("Lake1", "Lake2","BNTI", "RC", "Type",
                                                  "Lake1.Name", "Lake1.Latitude", "Lake1.Longitude",
                                                  "Lake2.Name", "Lake2.Latitude", "Lake2.Longitude")


Combined.results.Phytoplankton3.filtered <- Combined.results.Phytoplankton3 %>% 
                                              filter(Type != "NDP") %>%
                                              filter(Type != "Variable Selection") %>%
                                              filter(Type != "Homogeneous Selection")

coord.lake1 <- Combined.results.Phytoplankton3.filtered %>% select(Lake1.Longitude, Lake1.Latitude)

colnames(coord.lake1) <- c("Longitude", "Latitude")

coord.lake2 <- Combined.results.Phytoplankton3.filtered %>% select(Lake2.Longitude, Lake2.Latitude)

colnames(coord.lake2) <- c("Longitude", "Latitude")

Combined.results.Phytoplankton3.filtered$Distance <- distHaversine(coord.lake1, coord.lake2)


Combined.results.Phytoplankton3.filtered.dist.sum <- Combined.results.Phytoplankton3.filtered %>% select(Type, Distance) %>% group_by(Type) %>% summarise(Mean.Dist =mean(Distance))

Combined.results.Phytoplankton3.filtered.dist <- Combined.results.Phytoplankton3.filtered %>% select(Type, Distance)

kruskal.test(Distance ~ Type, data = Combined.results.Phytoplankton3.filtered.dist)

#dunnTest(Distance ~ Type, data = Combined.results.Phytoplankton3.filtered.dist, method="bh") 
```

```{r}
sd.SS_euks <- data.frame(sample_data(West.Coast.ss.euks.rare))

sd.SS_euks1 <- sd.SS_euks %>% select(Lake.Code, Latitude, Longitude) %>% separate("Lake.Code", c("Lake", "Code"), sep="\\.")

Combined.results.SS_euks1 <- Combined.results.SS_euks %>% separate("Comparison", c("Lake1", "Lake2"), sep = "_")


Combined.results.SS_euks2 <- left_join(Combined.results.SS_euks1, sd.SS_euks1, by=c("Lake1" = "Code"))

colnames(Combined.results.SS_euks2) <- c("Lake1", "Lake2","BNTI", "RC", "Type","Lake1.Name", "Lake1.Latitude", "Lake1.Longitude")


Combined.results.SS_euks3 <- left_join(Combined.results.SS_euks2, sd.SS_euks1, by=c("Lake2" = "Code"))

colnames(Combined.results.SS_euks3) <- c("Lake1", "Lake2","BNTI", "RC", "Type",
                                                  "Lake1.Name", "Lake1.Latitude", "Lake1.Longitude",
                                                  "Lake2.Name", "Lake2.Latitude", "Lake2.Longitude")


Combined.results.SS_euks3.filtered <- Combined.results.SS_euks3 %>% 
                                              filter(Type != "NDP") %>%
                                              filter(Type != "Variable Selection") %>%
                                              filter(Type != "Homogeneous Selection")

coord.lake1 <- Combined.results.SS_euks3.filtered %>% select(Lake1.Longitude, Lake1.Latitude)

colnames(coord.lake1) <- c("Longitude", "Latitude")

coord.lake2 <- Combined.results.SS_euks3.filtered %>% select(Lake2.Longitude, Lake2.Latitude)

colnames(coord.lake2) <- c("Longitude", "Latitude")



Combined.results.SS_euks3.filtered$Distance <- distHaversine(coord.lake1, coord.lake2)


Combined.results.SS_euks3.filtered.dist.sum <- Combined.results.SS_euks3.filtered %>% select(Type, Distance) %>% group_by(Type) %>% summarise(Mean.Dist =mean(Distance))

Combined.results.SS_euks3.filtered.dist <- Combined.results.SS_euks3.filtered %>% select(Type, Distance)

kruskal.test(Distance ~ Type, data = Combined.results.SS_euks3.filtered.dist)

#dunnTest(Distance ~ Type, data = Combined.results.SS_euks3.filtered.dist, method="bh") 
```

```{r}
sd.SS_bact <- data.frame(sample_data(West.Coast.ss.bact.rare))

sd.SS_bact1 <- sd.SS_bact %>% select(Lake.Code, Latitude, Longitude) %>% separate("Lake.Code", c("Lake", "Code"), sep="\\.")

Combined.results.SS_bact1 <- Combined.results.SS_bact %>% separate("Comparison", c("Lake1", "Lake2"), sep = "_")


Combined.results.SS_bact2 <- left_join(Combined.results.SS_bact1, sd.SS_bact1, by=c("Lake1" = "Code"))

colnames(Combined.results.SS_bact2) <- c("Lake1", "Lake2","BNTI", "RC", "Type","Lake1.Name", "Lake1.Latitude", "Lake1.Longitude")


Combined.results.SS_bact3 <- left_join(Combined.results.SS_bact2, sd.SS_bact1, by=c("Lake2" = "Code"))

colnames(Combined.results.SS_bact3) <- c("Lake1", "Lake2","BNTI", "RC", "Type",
                                                  "Lake1.Name", "Lake1.Latitude", "Lake1.Longitude",
                                                  "Lake2.Name", "Lake2.Latitude", "Lake2.Longitude")


Combined.results.SS_bact3.filtered <- Combined.results.SS_bact3  %>% 
                                              filter(Type != "NDP") %>%
                                              filter(Type != "Variable Selection") %>%
                                              filter(Type != "Homogeneous Selection")



coord.lake1 <- Combined.results.SS_bact3.filtered %>% select(Lake1.Longitude, Lake1.Latitude)

colnames(coord.lake1) <- c("Longitude", "Latitude")

coord.lake2 <- Combined.results.SS_bact3.filtered %>% select(Lake2.Longitude, Lake2.Latitude)

colnames(coord.lake2) <- c("Longitude", "Latitude")



Combined.results.SS_bact3.filtered$Distance <- distHaversine(coord.lake1, coord.lake2)


Combined.results.SS_bact3.filtered.dist.sum <- Combined.results.SS_bact3.filtered %>% select(Type, Distance) %>% group_by(Type) %>% summarise(Mean.Dist =mean(Distance))

Combined.results.SS_bact3.filtered.dist <- Combined.results.SS_bact3.filtered %>% select(Type, Distance)

kruskal.test(Distance ~ Type, data = Combined.results.SS_bact3.filtered.dist)

#dunnTest(Distance ~ Type, data = Combined.results.SS_bact3.filtered.dist, method="bh") 
```

#Asssembly Plot

```{r}
circostest<-read.csv("Assembly_process_circlize.csv", h=T)

grid.col = c(DL = "#006c18", HD = "#26c2ff", VS = "#d7ae2d", HS = "#c82db1", NDP = "#bebebe",
             SSbact = "#2e382e", SSeuks = "#50c9ce", SPeuks = "#72a1e5", LPeuks = "#9883e5", Pbact = "#fcd3de")

cairo_ps(file="Assembly_processes_summary.eps", width=5, height=5, bg="transparent")
chordDiagram(circostest, grid.col = grid.col, annotationTrack = c("name","grid"))
dev.off()

```

#Env Distance Sim

```{r}
sample_data(West.Coast.ss.bact.rare)$Forestry <- sample_data(West.Coast.ss.bact.rare)$For.Harv + sample_data(West.Coast.ss.bact.rare)$Exot.For
 
sample_data(West.Coast.ss.bact.rare)$Native <- sample_data(West.Coast.ss.bact.rare)$Alpine.Grass + 
                                               sample_data(West.Coast.ss.bact.rare)$Indi.Hard + 
                                               sample_data(West.Coast.ss.bact.rare)$Fernland +
                                               sample_data(West.Coast.ss.bact.rare)$Flaxland + 
                                               sample_data(West.Coast.ss.bact.rare)$Indi.For + 
                                               sample_data(West.Coast.ss.bact.rare)$Manuka +
                                               sample_data(West.Coast.ss.bact.rare)$Matagouri + 
                                               sample_data(West.Coast.ss.bact.rare)$Sub.Alp.Shrub + 
                                               sample_data(West.Coast.ss.bact.rare)$Tussock

sample_data(West.Coast.ss.bact.rare)$Urban <- sample_data(West.Coast.ss.bact.rare)$Built.up + 
                                               sample_data(West.Coast.ss.bact.rare)$Trans.Infra + 
                                               sample_data(West.Coast.ss.bact.rare)$Parkland 
                                               
sample_data(West.Coast.ss.bact.rare)$Non.Native <- sample_data(West.Coast.ss.bact.rare)$Decid.Hard + 
                                               sample_data(West.Coast.ss.bact.rare)$Dep.Grass + 
                                               sample_data(West.Coast.ss.bact.rare)$Gorse +
                                               sample_data(West.Coast.ss.bact.rare)$Mixed.Exo.Shrub + 
                                               sample_data(West.Coast.ss.bact.rare)$Orchard + 
                                               sample_data(West.Coast.ss.bact.rare)$Short.rot.Crop

sample_data(West.Coast.ss.bact.rare)$Water <- sample_data(West.Coast.ss.bact.rare)$Estu.Wat + 
                                               sample_data(West.Coast.ss.bact.rare)$River

sample_data(West.Coast.ss.bact.rare)$Other <- sample_data(West.Coast.ss.bact.rare)$Gravel + 
                                               sample_data(West.Coast.ss.bact.rare)$Herb.FW.Veg + 
                                               sample_data(West.Coast.ss.bact.rare)$Herb.Sal.Veg +
                                               sample_data(West.Coast.ss.bact.rare)$Landslide + 
                                               sample_data(West.Coast.ss.bact.rare)$Per.Snow + 
                                               sample_data(West.Coast.ss.bact.rare)$Sand +
                                               sample_data(West.Coast.ss.bact.rare)$Mine
```

```{r}

ss.bact.sd <- data.frame(sample_data(West.Coast.ss.bact.rare))

ss.bact.sd.filtered <- ss.bact.sd %>% select(BulS.TOC, BulS.TN, BulS.Phosphorus, WCS.Chla, Secchi.Disk, maxDepthTemp, maxDepthDO, 
                                                      Altitude, Max.Depth, Forestry, Native, Non.Native, Other, Low.Prod.Grass)

ss.bact.sd.filtered$BulS.TOC   <- as.numeric(ss.bact.sd.filtered$BulS.TOC)
ss.bact.sd.filtered$BulS.TN   <- as.numeric(ss.bact.sd.filtered$BulS.TN)
ss.bact.sd.filtered$BulS.Phosphorus   <- as.numeric(ss.bact.sd.filtered$BulS.Phosphorus)
ss.bact.sd.filtered$WCS.Chla   <- as.numeric(ss.bact.sd.filtered$WCS.Chla)
ss.bact.sd.filtered$Secchi.Disk   <- as.numeric(ss.bact.sd.filtered$Secchi.Disk)
ss.bact.sd.filtered$maxDepthTemp   <- as.numeric(ss.bact.sd.filtered$maxDepthTemp)
ss.bact.sd.filtered$maxDepthDO   <- as.numeric(ss.bact.sd.filtered$maxDepthDO)
ss.bact.sd.filtered$Altitude   <- as.numeric(ss.bact.sd.filtered$Altitude)
ss.bact.sd.filtered$Max.Depth   <- as.numeric(ss.bact.sd.filtered$Max.Depth)
ss.bact.sd.filtered$Forestry   <- as.numeric(ss.bact.sd.filtered$Forestry)
ss.bact.sd.filtered$Native   <- as.numeric(ss.bact.sd.filtered$Native)
ss.bact.sd.filtered$Non.Native   <- as.numeric(ss.bact.sd.filtered$Non.Native)
ss.bact.sd.filtered$Other   <- as.numeric(ss.bact.sd.filtered$Other)
ss.bact.sd.filtered$Low.Prod.Grass   <- as.numeric(ss.bact.sd.filtered$Low.Prod.Grass)

ss.bact.sd.filtered2 <-
  preProcess(ss.bact.sd.filtered,
             method = c("center", "scale", "YeoJohnson", "corr", "bagImpute"))

ss.bact.sd.filtered3 <- predict(ss.bact.sd.filtered2, ss.bact.sd.filtered)
```

```{r}
West.Coast.ss.bact.rare.comp.df <- data.frame(t(otu_table(West.Coast.ss.bact.rare)), check.names = FALSE)
West.Coast.ss.bact.rare.comp.mat <- as.matrix(t(West.Coast.ss.bact.rare.comp.df))

West.Coast.ss.bact.rare.bray <- vegdist(West.Coast.ss.bact.rare.comp.mat, method="bray")
```

```{r}
ss.bact.sd.filtered.mat <- as.matrix(ss.bact.sd.filtered3)

ss.bact.sd.filtered.dist <- vegdist(ss.bact.sd.filtered.mat, method="euclidean")
```

```{r}
West.Coast.ss.bact.rare.bray_simmat <- 1-(as.matrix(West.Coast.ss.bact.rare.bray))

SS.Bact_bray_simmat_melt <- reshape2::melt(West.Coast.ss.bact.rare.bray_simmat)[reshape2::melt(upper.tri(West.Coast.ss.bact.rare.bray_simmat))$value,]
names(SS.Bact_bray_simmat_melt) <- c("A", "B", "similarity")

env_dist_melt <- reshape2::melt(as.matrix(ss.bact.sd.filtered.dist))[reshape2::melt(upper.tri(as.matrix(ss.bact.sd.filtered.dist)))$value,]
names(env_dist_melt) <- c("A", "B", "distance")

SS.Bact_bray_simmat_melt["distance"] <- NA
SS.Bact_bray_simmat_melt$distance <- env_dist_melt$distance

SS.Bact_bray_simmat_melt["Group"] <- "SS.Bact"
```

```{r}
library(ggpmisc)
library(broom)

SS.Bact_bray_simmat_plot <- ggplot(SS.Bact_bray_simmat_melt, aes(x=distance, y=similarity)) + 
  geom_point(shape=20, size=0.5) + 
  geom_smooth(method = "lm",
              se = T,
              formula = y ~ x) +
 ggpmisc::stat_fit_glance(
    method = "lm",
    label.y = "top",
    label.x = "left",
    method.args = list(formula = y ~ x),
    #mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"="~%.2g',
    mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"<0.001"',
                                  after_stat(r.squared), after_stat(p.value))),
                    parse = TRUE) +
 # geom_smooth(method=lm, se=FALSE, formula=y ~ poly(x, 3, raw=TRUE)) +
  theme(panel.grid.minor = element_blank(), 
        panel.grid.major = element_blank(),
        panel.background = element_blank(), 
        plot.background = element_blank()) + 
  theme_bw() + 
  theme(plot.title=element_text(size=14)) + 
  theme(axis.text=element_text(size=10, color="black")) + 
  theme(legend.background=element_rect(color="black")) + 
  theme(legend.title=element_text(size=10)) + 
  theme (axis.title=element_text(size=10)) + 
  theme(legend.text=element_text(size=10)) + 
  xlab("Environmental Distance") + 
  ylab("") +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) +  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Surface sediment bacteria")

index.ss_bact = lm(similarity ~ distance, SS.Bact_bray_simmat_melt)

index.ss_bact

summary(index.ss_bact)
```

```{r}
sample_data(West.Coast.ss.euks.rare)$Forestry <- sample_data(West.Coast.ss.euks.rare)$For.Harv + sample_data(West.Coast.ss.euks.rare)$Exot.For
 
sample_data(West.Coast.ss.euks.rare)$Native <- sample_data(West.Coast.ss.euks.rare)$Alpine.Grass + 
                                               sample_data(West.Coast.ss.euks.rare)$Indi.Hard + 
                                               sample_data(West.Coast.ss.euks.rare)$Fernland +
                                               sample_data(West.Coast.ss.euks.rare)$Flaxland + 
                                               sample_data(West.Coast.ss.euks.rare)$Indi.For + 
                                               sample_data(West.Coast.ss.euks.rare)$Manuka +
                                               sample_data(West.Coast.ss.euks.rare)$Matagouri + 
                                               sample_data(West.Coast.ss.euks.rare)$Sub.Alp.Shrub + 
                                               sample_data(West.Coast.ss.euks.rare)$Tussock

sample_data(West.Coast.ss.euks.rare)$Urban <- sample_data(West.Coast.ss.euks.rare)$Built.up + 
                                               sample_data(West.Coast.ss.euks.rare)$Trans.Infra + 
                                               sample_data(West.Coast.ss.euks.rare)$Parkland 
                                               
sample_data(West.Coast.ss.euks.rare)$Non.Native <- sample_data(West.Coast.ss.euks.rare)$Decid.Hard + 
                                               sample_data(West.Coast.ss.euks.rare)$Dep.Grass + 
                                               sample_data(West.Coast.ss.euks.rare)$Gorse +
                                               sample_data(West.Coast.ss.euks.rare)$Mixed.Exo.Shrub + 
                                               sample_data(West.Coast.ss.euks.rare)$Orchard + 
                                               sample_data(West.Coast.ss.euks.rare)$Short.rot.Crop

sample_data(West.Coast.ss.euks.rare)$Water <- sample_data(West.Coast.ss.euks.rare)$Estu.Wat + 
                                               sample_data(West.Coast.ss.euks.rare)$River

sample_data(West.Coast.ss.euks.rare)$Other <- sample_data(West.Coast.ss.euks.rare)$Gravel + 
                                               sample_data(West.Coast.ss.euks.rare)$Herb.FW.Veg + 
                                               sample_data(West.Coast.ss.euks.rare)$Herb.Sal.Veg +
                                               sample_data(West.Coast.ss.euks.rare)$Landslide + 
                                               sample_data(West.Coast.ss.euks.rare)$Per.Snow + 
                                               sample_data(West.Coast.ss.euks.rare)$Sand +
                                               sample_data(West.Coast.ss.euks.rare)$Mine
```

```{r}

ss.euks.sd <- data.frame(sample_data(West.Coast.ss.euks.rare))

ss.euks.sd.filtered <- ss.euks.sd %>% select(BulS.TOC, BulS.TN, BulS.Phosphorus, WCS.Chla, Secchi.Disk, maxDepthTemp, maxDepthDO, 
                                                      Altitude, Max.Depth, Forestry, Native, Non.Native, Other, Low.Prod.Grass)

ss.euks.sd.filtered$BulS.TOC   <- as.numeric(ss.euks.sd.filtered$BulS.TOC)
ss.euks.sd.filtered$BulS.TN   <- as.numeric(ss.euks.sd.filtered$BulS.TN)
ss.euks.sd.filtered$BulS.Phosphorus   <- as.numeric(ss.euks.sd.filtered$BulS.Phosphorus)
ss.euks.sd.filtered$WCS.Chla   <- as.numeric(ss.euks.sd.filtered$WCS.Chla)
ss.euks.sd.filtered$Secchi.Disk   <- as.numeric(ss.euks.sd.filtered$Secchi.Disk)
ss.euks.sd.filtered$maxDepthTemp   <- as.numeric(ss.euks.sd.filtered$maxDepthTemp)
ss.euks.sd.filtered$maxDepthDO   <- as.numeric(ss.euks.sd.filtered$maxDepthDO)
ss.euks.sd.filtered$Altitude   <- as.numeric(ss.euks.sd.filtered$Altitude)
ss.euks.sd.filtered$Max.Depth   <- as.numeric(ss.euks.sd.filtered$Max.Depth)
ss.euks.sd.filtered$Forestry   <- as.numeric(ss.euks.sd.filtered$Forestry)
ss.euks.sd.filtered$Native   <- as.numeric(ss.euks.sd.filtered$Native)
ss.euks.sd.filtered$Non.Native   <- as.numeric(ss.euks.sd.filtered$Non.Native)
ss.euks.sd.filtered$Other   <- as.numeric(ss.euks.sd.filtered$Other)
ss.euks.sd.filtered$Low.Prod.Grass   <- as.numeric(ss.euks.sd.filtered$Low.Prod.Grass)

ss.euks.sd.filtered2 <-
  preProcess(ss.euks.sd.filtered,
             method = c("center", "scale", "YeoJohnson", "corr", "bagImpute"))

ss.euks.sd.filtered3 <- predict(ss.euks.sd.filtered2, ss.euks.sd.filtered)
```

```{r}
West.Coast.ss.euks.rare.comp.df <- data.frame(t(otu_table(West.Coast.ss.euks.rare)), check.names = FALSE)
West.Coast.ss.euks.rare.comp.mat <- as.matrix(t(West.Coast.ss.euks.rare.comp.df))

West.Coast.ss.euks.rare.bray <- vegdist(West.Coast.ss.euks.rare.comp.mat, method="bray")
```

```{r}
ss.euks.sd.filtered.mat <- as.matrix(ss.euks.sd.filtered3)

ss.euks.sd.filtered.dist <- vegdist(ss.euks.sd.filtered.mat, method="euclidean")
```

```{r}
West.Coast.ss.euks.rare.bray_simmat <- 1-(as.matrix(West.Coast.ss.euks.rare.bray))

ss.euks_bray_simmat_melt <- reshape2::melt(West.Coast.ss.euks.rare.bray_simmat)[reshape2::melt(upper.tri(West.Coast.ss.euks.rare.bray_simmat))$value,]
names(ss.euks_bray_simmat_melt) <- c("A", "B", "similarity")

env_dist_melt <- reshape2::melt(as.matrix(ss.euks.sd.filtered.dist))[reshape2::melt(upper.tri(as.matrix(ss.euks.sd.filtered.dist)))$value,]
names(env_dist_melt) <- c("A", "B", "distance")

ss.euks_bray_simmat_melt["distance"] <- NA
ss.euks_bray_simmat_melt$distance <- env_dist_melt$distance

ss.euks_bray_simmat_melt["Group"] <- "ss.euks"
```

```{r}
ss_euks_bray_simmat_plot <- ggplot(ss.euks_bray_simmat_melt, aes(x=distance, y=similarity)) + 
  geom_point(shape=20, size=0.5) + 
  geom_smooth(method = "lm",
              se = T,
              formula = y ~ x) +
 ggpmisc::stat_fit_glance(
    method = "lm",
    label.y = "top",
    label.x = "left",
    method.args = list(formula = y ~ x),
    mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"="~%.2g',
                                  after_stat(r.squared), after_stat(p.value))),
                    parse = TRUE) +
 # geom_smooth(method=lm, se=FALSE, formula=y ~ poly(x, 3, raw=TRUE)) +
  theme(panel.grid.minor = element_blank(), 
        panel.grid.major = element_blank(),
        panel.background = element_blank(), 
        plot.background = element_blank()) + 
  theme_bw() + 
  theme(plot.title=element_text(size=14)) + 
  theme(axis.text=element_text(size=10, color="black")) + 
  theme(legend.background=element_rect(color="black")) + 
  theme(legend.title=element_text(size=10)) + 
  theme (axis.title=element_text(size=10)) + 
  theme(legend.text=element_text(size=10)) + 
  xlab(" ") + 
  ylab("Community Similarity (1 - Bray)") +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Surface sediment eukaryotes")

index.ss_euks = lm(similarity ~ distance, ss.euks_bray_simmat_melt)

index.ss_euks

summary(index.ss_euks)
```

```{r}
sample_data(West.Coast.Phytoplankton.cleaned.rare)$Forestry <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$For.Harv + sample_data(West.Coast.Phytoplankton.cleaned.rare)$Exot.For
 
sample_data(West.Coast.Phytoplankton.cleaned.rare)$Native <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Alpine.Grass + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Indi.Hard + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Fernland +
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Flaxland + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Indi.For + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Manuka +
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Matagouri + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Sub.Alp.Shrub + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Tussock

sample_data(West.Coast.Phytoplankton.cleaned.rare)$Urban <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Built.up + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Trans.Infra + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Parkland 

sample_data(West.Coast.Phytoplankton.cleaned.rare)$Non.Native <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Decid.Hard + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Dep.Grass + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Gorse +
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Mixed.Exo.Shrub + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Orchard + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Short.rot.Crop

sample_data(West.Coast.Phytoplankton.cleaned.rare)$Water <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Estu.Wat + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$River

sample_data(West.Coast.Phytoplankton.cleaned.rare)$Other <- sample_data(West.Coast.Phytoplankton.cleaned.rare)$Gravel + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Herb.FW.Veg + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Herb.Sal.Veg +
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Landslide + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Per.Snow + 
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Sand +
                                               sample_data(West.Coast.Phytoplankton.cleaned.rare)$Mine
```

```{r}

phytoplankton.sd <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare))

phytoplankton.sd.filtered <- phytoplankton.sd %>% select(BulS.TOC, BulS.TN, BulS.Phosphorus, WCS.Chla, Secchi.Disk, maxDepthTemp, maxDepthDO, 
                                                      Altitude, Max.Depth, Forestry, Native, Non.Native, Other, Low.Prod.Grass)

phytoplankton.sd.filtered$BulS.TOC   <- as.numeric(phytoplankton.sd.filtered$BulS.TOC)
phytoplankton.sd.filtered$BulS.TN   <- as.numeric(phytoplankton.sd.filtered$BulS.TN)
phytoplankton.sd.filtered$BulS.Phosphorus   <- as.numeric(phytoplankton.sd.filtered$BulS.Phosphorus)
phytoplankton.sd.filtered$WCS.Chla   <- as.numeric(phytoplankton.sd.filtered$WCS.Chla)
phytoplankton.sd.filtered$Secchi.Disk   <- as.numeric(phytoplankton.sd.filtered$Secchi.Disk)
phytoplankton.sd.filtered$maxDepthTemp   <- as.numeric(phytoplankton.sd.filtered$maxDepthTemp)
phytoplankton.sd.filtered$maxDepthDO   <- as.numeric(phytoplankton.sd.filtered$maxDepthDO)
phytoplankton.sd.filtered$Altitude   <- as.numeric(phytoplankton.sd.filtered$Altitude)
phytoplankton.sd.filtered$Max.Depth   <- as.numeric(phytoplankton.sd.filtered$Max.Depth)
phytoplankton.sd.filtered$Forestry   <- as.numeric(phytoplankton.sd.filtered$Forestry)
phytoplankton.sd.filtered$Native   <- as.numeric(phytoplankton.sd.filtered$Native)
phytoplankton.sd.filtered$Non.Native   <- as.numeric(phytoplankton.sd.filtered$Non.Native)
phytoplankton.sd.filtered$Other   <- as.numeric(phytoplankton.sd.filtered$Other)
phytoplankton.sd.filtered$Low.Prod.Grass   <- as.numeric(phytoplankton.sd.filtered$Low.Prod.Grass)

phytoplankton.sd.filtered2 <-
  preProcess(phytoplankton.sd.filtered,
             method = c("center", "scale", "YeoJohnson", "corr", "bagImpute"))

phytoplankton.sd.filtered3 <- predict(phytoplankton.sd.filtered2, phytoplankton.sd.filtered)
```

```{r}
West.Coast.Phytoplankton.cleaned.rare.comp.df <- data.frame(t(otu_table(West.Coast.Phytoplankton.cleaned.rare)), check.names = FALSE)
West.Coast.Phytoplankton.cleaned.rare.comp.mat <- as.matrix(t(West.Coast.Phytoplankton.cleaned.rare.comp.df))


West.Coast.Phytoplankton.cleaned.rare.bray <- vegdist(West.Coast.Phytoplankton.cleaned.rare.comp.mat, method="bray")
```

```{r}
phytoplankton.sd.filtered.mat <- as.matrix(phytoplankton.sd.filtered3)

phytoplankton.sd.filtered.dist <- vegdist(phytoplankton.sd.filtered.mat, method="euclidean")
```

```{r}
West.Coast.Phytoplankton.cleaned.rare.bray_simmat <- 1-(as.matrix(West.Coast.Phytoplankton.cleaned.rare.bray))

phytoplankton_bray_simmat_melt <- reshape2::melt(West.Coast.Phytoplankton.cleaned.rare.bray_simmat)[reshape2::melt(upper.tri(West.Coast.Phytoplankton.cleaned.rare.bray_simmat))$value,]
names(phytoplankton_bray_simmat_melt) <- c("A", "B", "similarity")

env_dist_melt <- reshape2::melt(as.matrix(phytoplankton.sd.filtered.dist))[reshape2::melt(upper.tri(as.matrix(phytoplankton.sd.filtered.dist)))$value,]
names(env_dist_melt) <- c("A", "B", "distance")

phytoplankton_bray_simmat_melt["distance"] <- NA
phytoplankton_bray_simmat_melt$distance <- env_dist_melt$distance

phytoplankton_bray_simmat_melt["Group"] <- "phytoplankton"
```

```{r}
phytoplankton_bray_simmat_plot <- ggplot(phytoplankton_bray_simmat_melt, aes(x=distance, y=similarity)) + 
  geom_point(shape=20, size=0.5) + 
  geom_smooth(method = "lm",
              se = T,
              formula = y ~ x) +
 ggpmisc::stat_fit_glance(
    method = "lm",
    label.y = "top",
    label.x = "left",
    method.args = list(formula = y ~ x),
    mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"="~%.2g',
                                  after_stat(r.squared), after_stat(p.value))),
                    parse = TRUE) +
 # geom_smooth(method=lm, se=FALSE, formula=y ~ poly(x, 3, raw=TRUE)) +
  theme(panel.grid.minor = element_blank(), 
        panel.grid.major = element_blank(),
        panel.background = element_blank(), 
        plot.background = element_blank()) + 
  theme_bw() + 
  theme(plot.title=element_text(size=14)) + 
  theme(axis.text=element_text(size=10, color="black")) + 
  theme(legend.background=element_rect(color="black")) + 
  theme(legend.title=element_text(size=10)) + 
  theme (axis.title=element_text(size=10)) + 
  theme(legend.text=element_text(size=10)) + 
  xlab("") + 
  ylab("") +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  ggtitle("Small planktonic eukaryotes")

index.phytoplankton = lm(similarity ~ distance, phytoplankton_bray_simmat_melt)

index.phytoplankton

summary(index.phytoplankton)
```

```{r}
sample_data(West.Coast.zooplankton.cleaned.rare)$Forestry <- sample_data(West.Coast.zooplankton.cleaned.rare)$For.Harv + sample_data(West.Coast.zooplankton.cleaned.rare)$Exot.For
 
sample_data(West.Coast.zooplankton.cleaned.rare)$Native <- sample_data(West.Coast.zooplankton.cleaned.rare)$Alpine.Grass + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Indi.Hard + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Fernland +
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Flaxland + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Indi.For + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Manuka +
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Matagouri + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Sub.Alp.Shrub + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Tussock

sample_data(West.Coast.zooplankton.cleaned.rare)$Urban <- sample_data(West.Coast.zooplankton.cleaned.rare)$Built.up + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Trans.Infra + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Parkland 
                                               
sample_data(West.Coast.zooplankton.cleaned.rare)$Non.Native <- sample_data(West.Coast.zooplankton.cleaned.rare)$Decid.Hard + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Dep.Grass + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Gorse +
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Mixed.Exo.Shrub + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Orchard + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Short.rot.Crop

sample_data(West.Coast.zooplankton.cleaned.rare)$Water <- sample_data(West.Coast.zooplankton.cleaned.rare)$Estu.Wat + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$River

sample_data(West.Coast.zooplankton.cleaned.rare)$Other <- sample_data(West.Coast.zooplankton.cleaned.rare)$Gravel + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Herb.FW.Veg + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Herb.Sal.Veg +
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Landslide + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Per.Snow + 
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Sand +
                                               sample_data(West.Coast.zooplankton.cleaned.rare)$Mine
```

```{r}

zooplankton.sd <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare))

zooplankton.sd.filtered <- zooplankton.sd %>% select(BulS.TOC, BulS.TN, BulS.Phosphorus, WCS.Chla, Secchi.Disk, maxDepthTemp, maxDepthDO, 
                                                      Altitude, Max.Depth, Forestry, Native, Non.Native, Other, Low.Prod.Grass)

zooplankton.sd.filtered$BulS.TOC   <- as.numeric(zooplankton.sd.filtered$BulS.TOC)
zooplankton.sd.filtered$BulS.TN   <- as.numeric(zooplankton.sd.filtered$BulS.TN)
zooplankton.sd.filtered$BulS.Phosphorus   <- as.numeric(zooplankton.sd.filtered$BulS.Phosphorus)
zooplankton.sd.filtered$WCS.Chla   <- as.numeric(zooplankton.sd.filtered$WCS.Chla)
zooplankton.sd.filtered$Secchi.Disk   <- as.numeric(zooplankton.sd.filtered$Secchi.Disk)
zooplankton.sd.filtered$maxDepthTemp   <- as.numeric(zooplankton.sd.filtered$maxDepthTemp)
zooplankton.sd.filtered$maxDepthDO   <- as.numeric(zooplankton.sd.filtered$maxDepthDO)
zooplankton.sd.filtered$Altitude   <- as.numeric(zooplankton.sd.filtered$Altitude)
zooplankton.sd.filtered$Max.Depth   <- as.numeric(zooplankton.sd.filtered$Max.Depth)
zooplankton.sd.filtered$Forestry   <- as.numeric(zooplankton.sd.filtered$Forestry)
zooplankton.sd.filtered$Native   <- as.numeric(zooplankton.sd.filtered$Native)
zooplankton.sd.filtered$Non.Native   <- as.numeric(zooplankton.sd.filtered$Non.Native)
zooplankton.sd.filtered$Other   <- as.numeric(zooplankton.sd.filtered$Other)
zooplankton.sd.filtered$Low.Prod.Grass   <- as.numeric(zooplankton.sd.filtered$Low.Prod.Grass)

zooplankton.sd.filtered2 <-
  preProcess(zooplankton.sd.filtered,
             method = c("center", "scale", "YeoJohnson", "corr", "bagImpute"))

zooplankton.sd.filtered3 <- predict(zooplankton.sd.filtered2, zooplankton.sd.filtered)
```

```{r}
West.Coast.zooplankton.cleaned.rare.comp.df <- data.frame(t(otu_table(West.Coast.zooplankton.cleaned.rare)), check.names = FALSE)
West.Coast.zooplankton.cleaned.rare.comp.mat <- as.matrix(t(West.Coast.zooplankton.cleaned.rare.comp.df))

West.Coast.zooplankton.cleaned.rare.bray <- vegdist(West.Coast.zooplankton.cleaned.rare.comp.mat, method="bray")
```

```{r}
zooplankton.sd.filtered.mat <- as.matrix(zooplankton.sd.filtered3)

zooplankton.sd.filtered.dist <- vegdist(zooplankton.sd.filtered.mat, method="euclidean")
```

```{r}
West.Coast.zooplankton.cleaned.rare.bray_simmat <- 1-(as.matrix(West.Coast.zooplankton.cleaned.rare.bray))

zooplankton_bray_simmat_melt <- reshape2::melt(West.Coast.zooplankton.cleaned.rare.bray_simmat)[reshape2::melt(upper.tri(West.Coast.zooplankton.cleaned.rare.bray_simmat))$value,]
names(zooplankton_bray_simmat_melt) <- c("A", "B", "similarity")


env_dist_melt <- reshape2::melt(as.matrix(zooplankton.sd.filtered.dist))[reshape2::melt(upper.tri(as.matrix(zooplankton.sd.filtered.dist)))$value,]
names(env_dist_melt) <- c("A", "B", "distance")

zooplankton_bray_simmat_melt["distance"] <- NA
zooplankton_bray_simmat_melt$distance <- env_dist_melt$distance

zooplankton_bray_simmat_melt["Group"] <- "zooplankton"
```

```{r}
zooplankton_bray_simmat_plot <- ggplot(zooplankton_bray_simmat_melt, aes(x=distance, y=similarity)) + 
  geom_point(shape=20, size=0.5) + 
  geom_smooth(method = "lm",
              se = T,
              formula = y ~ x) +
 ggpmisc::stat_fit_glance(
    method = "lm",
    label.y = "top",
    label.x = "left",
    method.args = list(formula = y ~ x),
    mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"="~%.2g',
                                  after_stat(r.squared), after_stat(p.value))),
                    parse = TRUE) +
 # geom_smooth(method=lm, se=FALSE, formula=y ~ poly(x, 3, raw=TRUE)) +
  theme(panel.grid.minor = element_blank(), 
        panel.grid.major = element_blank(),
        panel.background = element_blank(), 
        plot.background = element_blank()) + 
  theme_bw() + 
  theme(plot.title=element_text(size=14)) + 
  theme(axis.text=element_text(size=10, color="black")) + 
  theme(legend.background=element_rect(color="black")) + 
  theme(legend.title=element_text(size=10)) + 
  theme (axis.title=element_text(size=10)) + 
  theme(legend.text=element_text(size=10)) + 
  xlab("") + 
  ylab("") +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Large planktonic eukaryotes")

index.zooplankton = lm(similarity ~ distance, zooplankton_bray_simmat_melt)

index.zooplankton

summary(index.zooplankton)
```

```{r}
sample_data(West.Coast.bacterioplankton.rare)$Forestry <- sample_data(West.Coast.bacterioplankton.rare)$For.Harv + sample_data(West.Coast.bacterioplankton.rare)$Exot.For
 
sample_data(West.Coast.bacterioplankton.rare)$Native <- sample_data(West.Coast.bacterioplankton.rare)$Alpine.Grass + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Indi.Hard + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Fernland +
                                               sample_data(West.Coast.bacterioplankton.rare)$Flaxland + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Indi.For + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Manuka +
                                               sample_data(West.Coast.bacterioplankton.rare)$Matagouri + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Sub.Alp.Shrub + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Tussock

sample_data(West.Coast.bacterioplankton.rare)$Urban <- sample_data(West.Coast.bacterioplankton.rare)$Built.up + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Trans.Infra + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Parkland 
                                               
sample_data(West.Coast.bacterioplankton.rare)$Non.Native <- sample_data(West.Coast.bacterioplankton.rare)$Decid.Hard + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Dep.Grass + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Gorse +
                                               sample_data(West.Coast.bacterioplankton.rare)$Mixed.Exo.Shrub + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Orchard + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Short.rot.Crop

sample_data(West.Coast.bacterioplankton.rare)$Water <- sample_data(West.Coast.bacterioplankton.rare)$Estu.Wat + 
                                               sample_data(West.Coast.bacterioplankton.rare)$River

sample_data(West.Coast.bacterioplankton.rare)$Other <- sample_data(West.Coast.bacterioplankton.rare)$Gravel + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Herb.FW.Veg + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Herb.Sal.Veg +
                                               sample_data(West.Coast.bacterioplankton.rare)$Landslide + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Per.Snow + 
                                               sample_data(West.Coast.bacterioplankton.rare)$Sand +
                                               sample_data(West.Coast.bacterioplankton.rare)$Mine
```

```{r}
bacterioplankton.sd <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

bacterioplankton.sd.filtered <- bacterioplankton.sd %>% select(BulS.TOC, BulS.TN, BulS.Phosphorus, WCS.Chla, Secchi.Disk, maxDepthTemp, maxDepthDO, 
                                                      Altitude, Max.Depth, Forestry, Native, Non.Native, Other, Low.Prod.Grass)

bacterioplankton.sd.filtered$BulS.TOC   <- as.numeric(bacterioplankton.sd.filtered$BulS.TOC)
bacterioplankton.sd.filtered$BulS.TN   <- as.numeric(bacterioplankton.sd.filtered$BulS.TN)
bacterioplankton.sd.filtered$BulS.Phosphorus   <- as.numeric(bacterioplankton.sd.filtered$BulS.Phosphorus)
bacterioplankton.sd.filtered$WCS.Chla   <- as.numeric(bacterioplankton.sd.filtered$WCS.Chla)
bacterioplankton.sd.filtered$Secchi.Disk   <- as.numeric(bacterioplankton.sd.filtered$Secchi.Disk)
bacterioplankton.sd.filtered$maxDepthTemp   <- as.numeric(bacterioplankton.sd.filtered$maxDepthTemp)
bacterioplankton.sd.filtered$maxDepthDO   <- as.numeric(bacterioplankton.sd.filtered$maxDepthDO)
bacterioplankton.sd.filtered$Altitude   <- as.numeric(bacterioplankton.sd.filtered$Altitude)
bacterioplankton.sd.filtered$Max.Depth   <- as.numeric(bacterioplankton.sd.filtered$Max.Depth)
bacterioplankton.sd.filtered$Forestry   <- as.numeric(bacterioplankton.sd.filtered$Forestry)
bacterioplankton.sd.filtered$Native   <- as.numeric(bacterioplankton.sd.filtered$Native)
bacterioplankton.sd.filtered$Non.Native   <- as.numeric(bacterioplankton.sd.filtered$Non.Native)
bacterioplankton.sd.filtered$Other   <- as.numeric(bacterioplankton.sd.filtered$Other)
bacterioplankton.sd.filtered$Low.Prod.Grass   <- as.numeric(bacterioplankton.sd.filtered$Low.Prod.Grass)

bacterioplankton.sd.filtered2 <-
  preProcess(bacterioplankton.sd.filtered,
             method = c("center", "scale", "YeoJohnson", "corr", "bagImpute"))

bacterioplankton.sd.filtered3 <- predict(bacterioplankton.sd.filtered2, bacterioplankton.sd.filtered)
```

```{r}
West.Coast.bacterioplankton.rare.comp.df <- data.frame(t(otu_table(West.Coast.bacterioplankton.rare)), check.names = FALSE)
West.Coast.bacterioplankton.rare.comp.mat <- as.matrix(t(West.Coast.bacterioplankton.rare.comp.df))

West.Coast.bacterioplankton.rare.bray <- vegdist(West.Coast.bacterioplankton.rare.comp.mat, method="bray")
```

```{r}
bacterioplankton.sd.filtered.mat <- as.matrix(bacterioplankton.sd.filtered3)

bacterioplankton.sd.filtered.dist <- vegdist(bacterioplankton.sd.filtered.mat, method="euclidean")
```

```{r}
West.Coast.bacterioplankton.rare.bray_simmat <- 1-(as.matrix(West.Coast.bacterioplankton.rare.bray))

bacterioplankton_bray_simmat_melt <- reshape2::melt(West.Coast.bacterioplankton.rare.bray_simmat)[reshape2::melt(upper.tri(West.Coast.bacterioplankton.rare.bray_simmat))$value,]
names(bacterioplankton_bray_simmat_melt) <- c("A", "B", "similarity")

env_dist_melt <- reshape2::melt(as.matrix(bacterioplankton.sd.filtered.dist))[reshape2::melt(upper.tri(as.matrix(bacterioplankton.sd.filtered.dist)))$value,]
names(env_dist_melt) <- c("A", "B", "distance")

bacterioplankton_bray_simmat_melt["distance"] <- NA
bacterioplankton_bray_simmat_melt$distance <- env_dist_melt$distance

bacterioplankton_bray_simmat_melt["Group"] <- "bacterioplankton"
```

```{r}
bacterioplankton_bray_simmat_plot <- ggplot(bacterioplankton_bray_simmat_melt, aes(x=distance, y=similarity)) + 
  geom_point(shape=20, size=0.5) + 
  geom_smooth(method = "lm",
              se = T,
              formula = y ~ x) +
 ggpmisc::stat_fit_glance(
    method = "lm",
    label.y = "top",
    label.x = "left",
    method.args = list(formula = y ~ x),
    #mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"="~%.2g',
    mapping = aes(label = sprintf('r^2~"="~%.3f~~italic(p)~"<0.001"',
                                  after_stat(r.squared), after_stat(p.value))),
                    parse = TRUE) +
 # geom_smooth(method=lm, se=FALSE, formula=y ~ poly(x, 3, raw=TRUE)) +
  theme(panel.grid.minor = element_blank(), 
        panel.grid.major = element_blank(),
        panel.background = element_blank(), 
        plot.background = element_blank()) + 
  theme_bw() + 
  theme(plot.title=element_text(size=14)) + 
  theme(axis.text=element_text(size=10, color="black")) + 
  theme(legend.background=element_rect(color="black")) + 
  theme(legend.title=element_text(size=10)) + 
  theme (axis.title=element_text(size=10)) + 
  theme(legend.text=element_text(size=10)) + 
  xlab("") + 
  ylab("Community Similarity (1 - Bray)") +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("Bacterioplankton")

index.bacterioplankton = lm(similarity ~ distance, bacterioplankton_bray_simmat_melt)

index.bacterioplankton

summary(index.bacterioplankton)
```

```{r}
cairo_ps(file="West.Coast.envdist.eps", width=10, height=8, bg="transparent")
gl<-grid.layout(nrow=2, ncol=3, heights=unit(c(20,20,20), "null"), widths=unit(c(20,20,20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=2, layout.pos.row=1)
vp.3<-viewport(layout.pos.col=3, layout.pos.row=1)
vp.4<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.5<-viewport(layout.pos.col=2, layout.pos.row=2)
vp.6<-viewport(layout.pos.col=3, layout.pos.row=2)
pushViewport(viewport(layout=gl))
print(bacterioplankton_bray_simmat_plot + theme(legend.position=""), vp=vp.1)
print(zooplankton_bray_simmat_plot + theme(legend.position=""), vp=vp.2)
print(phytoplankton_bray_simmat_plot + theme(legend.position=""), vp=vp.3)
print(ss_euks_bray_simmat_plot + theme(legend.position=""), vp=vp.4)
print(SS.Bact_bray_simmat_plot + theme(legend.position=""), vp=vp.5)
dev.off()
```

#Mulitvariate - Access Pressure

```{r}
access <- read.csv("WestCoastAccess.csv", h=T)

access <- access %>% select(Lake, Code, No.access, Access.Pressure, Access.Pressure2, AccessPressure3)

access$No.access[access$No.access == 1] <- "No"
access$No.access[access$No.access == 0] <- "Yes"
access$Code <- as.character(access$Code)
```

```{r}
sd.bact.mv <- data.frame(sample_data(West.Coast.bacterioplankton.rare))
sd.bact.mv$Code <- rownames(sd.bact.mv)

sd.bact.mv$Code <- as.character(sd.bact.mv$Code)

sd.bact.mv <- left_join(sd.bact.mv, access, by="Code")

rownames(sd.bact.mv) <- sd.bact.mv$Code
sample_data(West.Coast.bacterioplankton.rare) <- sd.bact.mv

West.Coast.bacterioplankton.rare.perc = transform_sample_counts(West.Coast.bacterioplankton.rare, function(OTU) OTU/sum(OTU)*100)

West.Coast.bacterioplankton.rare.perc_bray <- phyloseq::distance(West.Coast.bacterioplankton.rare.perc, method="bray")

West.Coast.bacterioplankton.rare.perc_ord <- ordinate(West.Coast.bacterioplankton.rare.perc, method="NMDS", distance=West.Coast.bacterioplankton.rare.perc_bray)

sample_data(West.Coast.bacterioplankton.rare.perc)$AccessPressure3<- factor(sample_data(West.Coast.bacterioplankton.rare.perc)$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.bacterioplankton.rare.perc_MVplot <- plot_ordination(West.Coast.bacterioplankton.rare.perc, West.Coast.bacterioplankton.rare.perc_ord, "samples", color="AccessPressure3")
West.Coast.bacterioplankton.rare.perc_MVplot_final <- West.Coast.bacterioplankton.rare.perc_MVplot  + 
  theme_classic() + 
  theme(plot.title = element_text(size=14, colour="black")) + 
  theme(legend.title=element_text(size=12, colour="black")) + 
  guides(col = guide_legend(title.position="top", ncol = 1, title = "Ease of \nhuman access")) + 
  theme(axis.title.y=element_text(size=12, colour="black")) + 
  theme(axis.title.x=element_text(size=12, colour="black")) + 
  theme(legend.text=element_text(size=11, colour="black")) + 
  theme(axis.text=element_text(size=11, colour="black")) + 
  geom_point(size = 3) + 
  theme(legend.position="right") + 
  scale_color_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  theme(axis.line.x = element_line(colour="black", size = 0.5),
        axis.line.y = element_line(colour="black", size = 0.5))  +
  ggtitle("Bacterioplankton") +
  xlab("")

sd <- data.frame(sample_data(West.Coast.bacterioplankton.rare.perc))

attach(sd)

bray.adonis <- adonis2(West.Coast.bacterioplankton.rare.perc_bray ~ AccessPressure3, data=sd, permutations=999)

bray.adonis

pairwise.adonis(West.Coast.bacterioplankton.rare.perc_bray, sd$AccessPressure3, p.adjust.m='BH')
```

```{r}
sd.zoo.mv <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare))

sd.zoo.mv$Code <- as.character(rownames(sd.zoo.mv))

sd.zoo.mv <- left_join(sd.zoo.mv, access, by=c("Code" = "Code"))

rownames(sd.zoo.mv) <- sd.zoo.mv$Code
sample_data(West.Coast.zooplankton.cleaned.rare) <- sd.zoo.mv

West.Coast.zooplankton.cleaned.rare.perc = transform_sample_counts(West.Coast.zooplankton.cleaned.rare, function(OTU) OTU/sum(OTU)*100)

West.Coast.zooplankton.cleaned.rare.perc_bray <- phyloseq::distance(West.Coast.zooplankton.cleaned.rare.perc, method="bray")

West.Coast.zooplankton.cleaned.rare.perc_ord <- ordinate(West.Coast.zooplankton.cleaned.rare.perc, method="NMDS", distance=West.Coast.zooplankton.cleaned.rare.perc_bray)

sample_data(West.Coast.zooplankton.cleaned.rare.perc)$AccessPressure3 <- factor(sample_data(West.Coast.zooplankton.cleaned.rare.perc)$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.zooplankton.cleaned.rare.perc_MVplot <- plot_ordination(West.Coast.zooplankton.cleaned.rare.perc, West.Coast.zooplankton.cleaned.rare.perc_ord, "samples", color="AccessPressure3")
West.Coast.zooplankton.cleaned.rare.perc_MVplot_final <- West.Coast.zooplankton.cleaned.rare.perc_MVplot  + 
  theme_classic() + 
  theme(plot.title = element_text(size=14, colour="black")) + 
  theme(legend.title=element_text(size=12, colour="black")) + 
  guides(col = guide_legend(title.position="top", ncol = 1, title = "Ease of \nhuman access")) + 
  theme(axis.title.y=element_text(size=12, colour="black")) + 
  theme(axis.title.x=element_text(size=12, colour="black")) + 
  theme(legend.text=element_text(size=11, colour="black")) + 
  theme(axis.text=element_text(size=11, colour="black")) + 
  geom_point(size = 3) + 
  theme(legend.position="right") + 
  scale_color_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  theme(axis.line.x = element_line(colour="black", size = 0.5),
        axis.line.y = element_line(colour="black", size = 0.5))  +
  ggtitle("Large planktonic eukaryotes")  +
  xlab("") +
  ylab("")

sd <- data.frame(sample_data(West.Coast.zooplankton.cleaned.rare.perc))

attach(sd)

bray.adonis <- adonis2(West.Coast.zooplankton.cleaned.rare.perc_bray ~ AccessPressure3, data=sd, permutations=999)

bray.adonis

pairwise.adonis(West.Coast.zooplankton.cleaned.rare.perc_bray, sd$AccessPressure3, p.adjust.m='BH')
```

```{r}
sd.phyto.mv <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare))

sd.phyto.mv$Code <- as.character(rownames(sd.phyto.mv))

sd.phyto.mv <- left_join(sd.phyto.mv, access, by=c("Code" = "Code"))

rownames(sd.phyto.mv) <- sd.phyto.mv$Code
sample_data(West.Coast.Phytoplankton.cleaned.rare) <- sd.phyto.mv

West.Coast.Phytoplankton.cleaned.rare.perc = transform_sample_counts(West.Coast.Phytoplankton.cleaned.rare, function(OTU) OTU/sum(OTU)*100)

West.Coast.Phytoplankton.cleaned.rare.perc_bray <- phyloseq::distance(West.Coast.Phytoplankton.cleaned.rare.perc, method="bray")

West.Coast.Phytoplankton.cleaned.rare.perc_ord <- ordinate(West.Coast.Phytoplankton.cleaned.rare.perc, method="NMDS", distance=West.Coast.Phytoplankton.cleaned.rare.perc_bray)

sample_data(West.Coast.Phytoplankton.cleaned.rare.perc)$AccessPressure3 <- factor(sample_data(West.Coast.Phytoplankton.cleaned.rare.perc)$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.Phytoplankton.cleaned.rare.perc_MVplot <- plot_ordination(West.Coast.Phytoplankton.cleaned.rare.perc, West.Coast.Phytoplankton.cleaned.rare.perc_ord, "samples", color="AccessPressure3")
West.Coast.Phytoplankton.cleaned.rare.perc_MVplot_final <- West.Coast.Phytoplankton.cleaned.rare.perc_MVplot  + 
  theme_classic() + 
  theme(plot.title = element_text(size=14, colour="black")) + 
  theme(legend.title=element_text(size=12, colour="black")) + 
  guides(col = guide_legend(title.position="top", ncol = 1, title = "Ease of \nhuman access")) + 
  theme(axis.title.y=element_text(size=12, colour="black")) + 
  theme(axis.title.x=element_text(size=12, colour="black")) + 
  theme(legend.text=element_text(size=11, colour="black")) + 
  theme(axis.text=element_text(size=11, colour="black")) + 
  geom_point(size = 3) + 
  theme(legend.position="right") + 
  scale_color_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  theme(axis.line.x = element_line(colour="black", size = 0.5),
        axis.line.y = element_line(colour="black", size = 0.5)) +
  ggtitle("Small planktonic eukaryotes") +
  xlab("") +
  ylab("")

sd <- data.frame(sample_data(West.Coast.Phytoplankton.cleaned.rare.perc))

attach(sd)

bray.adonis <- adonis2(West.Coast.Phytoplankton.cleaned.rare.perc_bray ~ AccessPressure3, data=sd, permutations=999)

bray.adonis

pairwise.adonis(West.Coast.Phytoplankton.cleaned.rare.perc_bray, sd$AccessPressure3, p.adjust.m='BH')
```

```{r}
sd.ss.euks.mv <- data.frame(sample_data(West.Coast.ss.euks.rare))
sd.ss.euks.mv$Code <- rownames(sd.ss.euks.mv)
sd.ss.euks.mv$Code <- as.character(sd.ss.euks.mv$Code)

sd.ss.euks.mv <- left_join(sd.ss.euks.mv, access, by=c("Code" = "Code"))

rownames(sd.ss.euks.mv) <- sd.ss.euks.mv$Code
sample_data(West.Coast.ss.euks.rare) <- sd.ss.euks.mv

West.Coast.ss.euks.rare.perc = transform_sample_counts(West.Coast.ss.euks.rare, function(OTU) OTU/sum(OTU)*100)

West.Coast.ss.euks.rare.perc_bray <- phyloseq::distance(West.Coast.ss.euks.rare.perc, method="bray")

West.Coast.ss.euks.rare.perc_ord <- ordinate(West.Coast.ss.euks.rare.perc, method="NMDS", distance=West.Coast.ss.euks.rare.perc_bray)

sample_data(West.Coast.ss.euks.rare.perc)$AccessPressure3 <- factor(sample_data(West.Coast.ss.euks.rare.perc)$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.ss.euks.rare.perc_MVplot <- plot_ordination(West.Coast.ss.euks.rare.perc, West.Coast.ss.euks.rare.perc_ord, "samples", color="AccessPressure3")
West.Coast.ss.euks.rare.perc_MVplot_final <- West.Coast.ss.euks.rare.perc_MVplot  + 
  theme_classic() + 
  theme(plot.title = element_text(size=14, colour="black")) + 
  theme(legend.title=element_text(size=12, colour="black")) + 
  guides(col = guide_legend(title.position="top", ncol = 1, title = "Ease of \nhuman access")) + 
  theme(axis.title.y=element_text(size=12, colour="black")) + 
  theme(axis.title.x=element_text(size=12, colour="black")) + 
  theme(legend.text=element_text(size=11, colour="black")) + 
  theme(axis.text=element_text(size=11, colour="black")) + 
  geom_point(size = 3) + 
  theme(legend.position="right") + 
  scale_color_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  theme(axis.line.x = element_line(colour="black", size = 0.5),
        axis.line.y = element_line(colour="black", size = 0.5)) +
  ggtitle("Surface sediment eukaryotes") +
  xlab("")

sd <- data.frame(sample_data(West.Coast.ss.euks.rare.perc))

attach(sd)

bray.adonis <- adonis2(West.Coast.ss.euks.rare.perc_bray ~ AccessPressure3, data=sd, permutations=999)
bray.adonis

pairwise.adonis(West.Coast.ss.euks.rare.perc_bray, sd$AccessPressure3, p.adjust.m='BH')
```

```{r}
sd.ss.bact.mv <- data.frame(sample_data(West.Coast.ss.bact.rare))

sd.ss.bact.mv$Code <- rownames(sd.ss.bact.mv)

sd.ss.bact.mv$Code <- as.character(sd.ss.bact.mv$Code)

sd.ss.bact.mv <- left_join(sd.ss.bact.mv, access, by=c("Code" = "Code"))

rownames(sd.ss.bact.mv) <- sd.ss.bact.mv$Code
sample_data(West.Coast.ss.bact.rare) <- sd.ss.bact.mv

West.Coast.ss.bact.rare.perc = transform_sample_counts(West.Coast.ss.bact.rare, function(OTU) OTU/sum(OTU)*100)

West.Coast.ss.bact.rare.perc_bray <- phyloseq::distance(West.Coast.ss.bact.rare.perc, method="bray")

West.Coast.ss.bact.rare.perc_ord <- ordinate(West.Coast.ss.bact.rare.perc, method="NMDS", distance=West.Coast.ss.bact.rare.perc_bray)

sample_data(West.Coast.ss.bact.rare.perc)$AccessPressure3 <- factor(sample_data(West.Coast.ss.bact.rare.perc)$AccessPressure3,
                                                                             levels=c("No Access", "Walking", "Private", "Public"))

West.Coast.ss.bact.rare.perc_MVplot <- plot_ordination(West.Coast.ss.bact.rare.perc, West.Coast.ss.bact.rare.perc_ord, "samples", color="AccessPressure3")
West.Coast.ss.bact.rare.perc_MVplot_final <- West.Coast.ss.bact.rare.perc_MVplot  + 
  theme_classic() + 
  theme(plot.title = element_text(size=14, colour="black")) + 
  theme(legend.title=element_text(size=12, colour="black")) + 
  guides(col = guide_legend(title.position="top", ncol = 1, title = "Ease of \nhuman access")) + 
  theme(axis.title.y=element_text(size=12, colour="black")) + 
  theme(axis.title.x=element_text(size=12, colour="black")) + 
  theme(legend.text=element_text(size=11, colour="black")) + 
  theme(axis.text=element_text(size=11, colour="black")) + 
  geom_point(size = 3) + 
  theme(legend.position="right") + 
  scale_color_manual(values = c("#05017f", "#25c6e2","#fec33a", "#d35b18", "#f8260d")) +
  theme(axis.line.x = element_line(colour="black", size = 0.5),
        axis.line.y = element_line(colour="black", size = 0.5)) +
  ggtitle("Surface sediment bacteria")  +
  ylab("")

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend <-g_legend(West.Coast.ss.bact.rare.perc_MVplot_final)

sd <- data.frame(sample_data(West.Coast.ss.bact.rare.perc))

attach(sd)

bray.adonis <- adonis2(West.Coast.ss.bact.rare.perc_bray ~ AccessPressure3, data=sd, permutations=999)
bray.adonis

pairwise.adonis(West.Coast.ss.bact.rare.perc_bray, sd$AccessPressure3, p.adjust.m='BH')
```

```{r}
cairo_ps(file="West.Coast.MV.eps", width=10, height=8, bg="transparent")
gl<-grid.layout(nrow=2, ncol=3, heights=unit(c(20,20,20), "null"), widths=unit(c(20,20,20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=2, layout.pos.row=1)
vp.3<-viewport(layout.pos.col=3, layout.pos.row=1)
vp.4<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.5<-viewport(layout.pos.col=2, layout.pos.row=2)
vp.6<-viewport(layout.pos.col=3, layout.pos.row=2)
pushViewport(viewport(layout=gl))
print(West.Coast.bacterioplankton.rare.perc_MVplot_final + theme(legend.position=""), vp=vp.1)
print(West.Coast.zooplankton.cleaned.rare.perc_MVplot_final + theme(legend.position=""), vp=vp.2)
print(West.Coast.Phytoplankton.cleaned.rare.perc_MVplot_final + theme(legend.position=""), vp=vp.3)
print(West.Coast.ss.euks.rare.perc_MVplot_final + theme(legend.position=""), vp=vp.4)
print(West.Coast.ss.bact.rare.perc_MVplot_final + theme(legend.position=""), vp=vp.5)
pushViewport(vp.6)
grid.draw(mylegend)
popViewport()
dev.off()
```

#Daphnia pulex

```{r}

zooplankton.access.merged.perc = transform_sample_counts(West.Coast.zooplankton.cleaned.rare, function(OTU) OTU/sum(OTU)*100)

zooplankton.access.merged.perc.Daphnia <- subset_taxa(zooplankton.access.merged.perc, Species == "Daphnia_pulex")

Daphnia.taxtable <- data.frame((tax_table(zooplankton.access.merged.perc.Daphnia)))

Daphnia.taxtable <- rownames_to_column(Daphnia.taxtable)

Daphnia.taxtable <- data.table(Daphnia.taxtable)
Daphnia.taxtable <- Daphnia.taxtable[Class == "Unclassified_Phylum_Arthropoda" , Class := "Branchiopoda"]
Daphnia.taxtable <- Daphnia.taxtable[Order == "Unclassified_Phylum_Arthropoda" , Order := "Diplostraca"]
Daphnia.taxtable <- Daphnia.taxtable[Family == "Unclassified_Phylum_Arthropoda" , Family := "Daphniidae"]
Daphnia.taxtable <- Daphnia.taxtable[Genus == "Unclassified_Phylum_Arthropoda" , Genus := "Daphnia"]

Daphnia.taxtable <- column_to_rownames(Daphnia.taxtable, var = "rowname")

tax_table(zooplankton.access.merged.perc.Daphnia) <- as.matrix(Daphnia.taxtable)

zooplankton.access.merged.Species <- tax_glom(zooplankton.access.merged.perc.Daphnia, "Species")

zooplankton.access.merged.perc.Daphnia_df <- psmelt(zooplankton.access.merged.Species)

compositionplot.ss.zooplankton.Daphnia <- zooplankton.access.merged.perc.Daphnia_df %>%
  ggplot(aes(x=AccessPressure3, y=log(Abundance+1))) + 
  geom_point() + 
  theme_bw()

Daphnia_pulex_summary <- zooplankton.access.merged.perc.Daphnia_df %>% group_by(AccessPressure3) %>% summarise(mean = mean(Abundance),
                                                                                                                sd = sd(Abundance))

kruskal.test(Abundance ~ AccessPressure3, data = zooplankton.access.merged.perc.Daphnia_df)

dunnTest(Abundance ~ AccessPressure3, data = zooplankton.access.merged.perc.Daphnia_df, method="bh") 

```

#SAR11

```{r}
bacterioplankton.access.merged.perc = transform_sample_counts(West.Coast.bacterioplankton.rare, function(OTU) OTU/sum(OTU)*100)

bacterioplankton.access.merged.perc.SAR11 <- subset_taxa(bacterioplankton.access.merged.perc, Order == "SAR11_clade")
bacterioplankton.access.merged.perc.SAR11 <- subset_taxa(bacterioplankton.access.merged.perc, Family == "Clade_III")

bacterioplankton.access.merged.Family <- tax_glom(bacterioplankton.access.merged.perc.SAR11, "Family")

bacterioplankton.access.merged.perc.SAR11_df <- psmelt(bacterioplankton.access.merged.Family)

compositionplot.ss.bacterioplankton.SAR11 <- bacterioplankton.access.merged.perc.SAR11_df %>%
  ggplot(aes(x=AccessPressure3, y=log(Abundance+1))) + 
  geom_point() + 
  theme_bw()

SAR11_pulex_summary <- bacterioplankton.access.merged.perc.SAR11_df %>% group_by(AccessPressure3) %>% summarise(mean = mean(Abundance),
                                                                                                                sd = sd(Abundance))

kruskal.test(Abundance ~ AccessPressure3, data = bacterioplankton.access.merged.perc.SAR11_df)

dunnTest(Abundance ~ AccessPressure3, data = bacterioplankton.access.merged.perc.SAR11_df, method="bh") 

```

#Mantel - biological

```{r}
sample_data(West.Coast.ss.bact.rare)$Code <- sample_names(West.Coast.ss.bact.rare)

West.Coast.ss.bact.rare.comp <- West.Coast.ss.bact.rare %>%
  subset_samples(Code != "47252") %>%
  subset_samples(Code != "46775")

sample_data(West.Coast.ss.euks.rare)$Code <- sample_names(West.Coast.ss.euks.rare)

West.Coast.ss.euks.rare.comp <- West.Coast.ss.euks.rare %>%
  subset_samples(Code != "47252") %>%
  subset_samples(Code != "46775")

West.Coast.Phytoplankton.cleaned.rare1 <- merge_samples(West.Coast.Phytoplankton.cleaned.rare, "Code")

West.Coast.Phytoplankton.cleaned.rare.comp <- West.Coast.Phytoplankton.cleaned.rare1 %>%
  subset_samples(Code != "47252") %>%
  subset_samples(Code != "46775")

West.Coast.zooplankton.cleaned.rare1 <- merge_samples(West.Coast.zooplankton.cleaned.rare, "Code")

West.Coast.zooplankton.cleaned.rare.comp <- West.Coast.zooplankton.cleaned.rare1 %>%
  subset_samples(Code != "47252") %>%
  subset_samples(Code != "46775")


sample_data(West.Coast.bacterioplankton.rare)$Code <- sample_names(West.Coast.bacterioplankton.rare)

West.Coast.bacterioplankton.rare.comp <- West.Coast.bacterioplankton.rare %>%
  subset_samples(Code != "47252") %>%
  subset_samples(Code != "46775")
```

```{r}
West.Coast.bacterioplankton.rare.comp.df <- data.frame(t(otu_table(West.Coast.bacterioplankton.rare.comp)), check.names = FALSE)

West.Coast.bacterioplankton.rare.comp.mat <- as.matrix(t(West.Coast.bacterioplankton.rare.comp.df))


West.Coast.zooplankton.cleaned.rare.comp.df <- data.frame(t(otu_table(West.Coast.zooplankton.cleaned.rare.comp)), check.names = FALSE)


West.Coast.zooplankton.cleaned.rare.comp.df <- West.Coast.zooplankton.cleaned.rare.comp.df %>% dplyr::select("38421", "38664", "38665", "38955", "39050", "39191", "39225", "39621", 
                                                                                                             "41679", "45434", "45472", "45686", "45688", "46138", "46212", "46484", 
                                                                                                             "46493", "46494", "46496", "46723", "46725", "46774", "46954", "46959", 
                                                                                                             "47193", "47198", "47225", "47491", "54177", "54655", "54720", "8138")

West.Coast.zooplankton.cleaned.rare.comp.mat <- as.matrix(t(West.Coast.zooplankton.cleaned.rare.comp.df))

West.Coast.Phytoplankton.cleaned.rare.comp.df <- data.frame(t(otu_table(West.Coast.Phytoplankton.cleaned.rare.comp)), check.names = FALSE)

West.Coast.Phytoplankton.cleaned.rare.comp.df <- West.Coast.Phytoplankton.cleaned.rare.comp.df %>% dplyr::select("38421", "38664", "38665", "38955", "39050", "39191", "39225", "39621", 
                                                                                                             "41679", "45434", "45472", "45686", "45688", "46138", "46212", "46484", 
                                                                                                             "46493", "46494", "46496", "46723", "46725", "46774", "46954", "46959", 
                                                                                                             "47193", "47198", "47225", "47491", "54177", "54655", "54720", "8138")

West.Coast.Phytoplankton.cleaned.rare.comp.mat <- as.matrix(t(West.Coast.Phytoplankton.cleaned.rare.comp.df))

West.Coast.ss.euks.rare.comp.df <- data.frame(t(otu_table(West.Coast.ss.euks.rare.comp)), check.names = FALSE)
West.Coast.ss.euks.rare.comp.mat <- as.matrix(t(West.Coast.ss.euks.rare.comp.df))

West.Coast.ss.bact.rare.comp.df <- data.frame(t(otu_table(West.Coast.ss.bact.rare.comp)), check.names = FALSE)
West.Coast.ss.bact.rare.comp.mat <- as.matrix(t(West.Coast.ss.bact.rare.comp.df))
```

```{r}
West.Coast.bacterioplankton.rare.bray <- vegdist(West.Coast.bacterioplankton.rare.comp.mat, method="bray")
West.Coast.zooplankton.cleaned.ps.pruned.rare.bray <- vegdist(West.Coast.zooplankton.cleaned.rare.comp.mat, method="bray")
West.Coast.Phytoplankton.cleaned.rare.bray <- vegdist(West.Coast.Phytoplankton.cleaned.rare.comp.mat, method="bray")
West.Coast.ss.euks.rare.bray <- vegdist(West.Coast.ss.euks.rare.comp.mat, method="bray")
West.Coast.ss.bact.rare.bray <- vegdist(West.Coast.ss.bact.rare.comp.mat, method="bray")
```

```{r}
vegan::mantel(West.Coast.ss.bact.rare.bray, West.Coast.ss.euks.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.bact.rare.bray, West.Coast.Phytoplankton.cleaned.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.bact.rare.bray, West.Coast.zooplankton.cleaned.ps.pruned.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.bact.rare.bray, West.Coast.bacterioplankton.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.euks.rare.bray, West.Coast.Phytoplankton.cleaned.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.euks.rare.bray, West.Coast.zooplankton.cleaned.ps.pruned.rare.bray, permutations=999) 
vegan::mantel(West.Coast.ss.euks.rare.bray, West.Coast.bacterioplankton.rare.bray, permutations=999) 
vegan::mantel(West.Coast.Phytoplankton.cleaned.rare.bray, West.Coast.zooplankton.cleaned.ps.pruned.rare.bray, permutations=999) 
vegan::mantel(West.Coast.Phytoplankton.cleaned.rare.bray, West.Coast.bacterioplankton.rare.bray, permutations=999) 
vegan::mantel(West.Coast.zooplankton.cleaned.ps.pruned.rare.bray, West.Coast.bacterioplankton.rare.bray, permutations=999) 
```

#Env variables

```{r}
sd <- data.frame(sample_data(West.Coast.bacterioplankton.rare))

sd.filtered <- sd %>% select(Max.Depth, Secchi.Disk, Altitude, WCS.Bul.TN, WCS.Bul.Nitrite, WCS.Bul.Nitrate, WCS.Bul.TKN, WCS.Bul.TP,
                             WCS.Bul.TSS, WCS.Bul.VSS, WCS.DOC, WCS.TOC, WCS.DIN.Nitrite, WCS.DIN.Nitrate, WCS.DIN.Ammonia, WCS.DIN.DRP, 
                             WCS.Arsenic, WCS.Cadmium, WCS.Chromium, WCS.Copper, WCS.Lead, WCS.Mercury, WCS.Nickel, WCS.Zinc, WCS.Chla,
                             minDepthTemp, maxDepthTemp, minDepthDO, maxDepthDO, Forestry, Native, Non.Native, Water, Other, High.Prod.Exotic.Grass, Low.Prod.Grass)

nzv <- nearZeroVar(sd.filtered, saveMetrics= TRUE, uniqueCut = 25)
nzv # Water, Urban and catPeat near zero variance

sd.filtered <- sd.filtered %>% select(-WCS.Mercury, -Water, -WCS.DIN.DRP, -WCS.Bul.VSS)

sd.filtered$minDepthTemp  <- as.numeric(sd.filtered$minDepthTemp)
sd.filtered$maxDepthTemp  <- as.numeric(sd.filtered$maxDepthTemp)
sd.filtered$minDepthDO  <- as.numeric(sd.filtered$minDepthDO)
sd.filtered$maxDepthDO <- as.numeric(sd.filtered$maxDepthDO)

p_mat <- cor_pmat(sd.filtered)

corr.plot <- ggcorrplot(cor(sd.filtered, use = 'pairwise.complete.obs'),
  tl.cex = 9,
  p.mat = p_mat,
  hc.order = TRUE,
  type = "lower",
  insig = "blank",
  lab = T,
  lab_size = 3
)

corr.barplot <- corr_cross(sd.filtered, # name of dataset
           max_pvalue = 0.1, 
           top = 30 
)

cairo_ps(file="~/Documents/Bioinformatics/Lakes380/DNA/Pred3800//corr.barplot.eps", width=5, height=6)
corr.barplot
dev.off()

sd.filtered.final <- sd.filtered %>% select(Max.Depth, Secchi.Disk, Altitude, WCS.Bul.TN, WCS.Bul.TP,
                             WCS.DOC, WCS.DIN.Nitrite, WCS.DIN.Nitrate, WCS.Chla, 
                             WCS.Arsenic, WCS.Cadmium, WCS.Chromium, maxDepthTemp, maxDepthDO, Forestry, Native, Non.Native, 
                             Other, Low.Prod.Grass)

```

